<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>DE analysis pipeline — de_analysis • mastR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="DE analysis pipeline — de_analysis"><meta property="og:description" content="Standard DE analysis by using edgeR and limma::voom pipeline"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mastR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/mastR_Demo.html">mastR_Demo</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/DavisLaboratory/mastR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DE analysis pipeline</h1>
    <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/mastR/blob/HEAD/R/degs_generic.R" class="external-link"><code>R/degs_generic.R</code></a></small>
    <div class="hidden name"><code>de_analysis.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Standard DE analysis by using edgeR and limma::voom pipeline</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">de_analysis</span><span class="op">(</span></span>
<span>  <span class="va">dge</span>,</span>
<span>  <span class="va">group_col</span>,</span>
<span>  <span class="va">target_group</span>,</span>
<span>  normalize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  group <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  lfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  p <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  batch <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  summary <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dge</dt>
<dd><p>DGEList object for DE analysis, including expr and samples info</p></dd>


<dt>group_col</dt>
<dd><p>character, column name of coldata to specify the DE comparisons</p></dd>


<dt>target_group</dt>
<dd><p>pattern, specify the group of interest, e.g. NK</p></dd>


<dt>normalize</dt>
<dd><p>logical, if the expr in data is raw counts needs to be normalized</p></dd>


<dt>group</dt>
<dd><p>logical, TRUE to separate samples into only 2 groups:
`target_group`` and 'Others'; FALSE to set each level as a group</p></dd>


<dt>filter</dt>
<dd><p>a vector of 2 numbers, filter condition to remove low expression
genes, the 1st for min.counts (if normalize = TRUE) or CPM/TPM
(if normalize = FALSE), the 2nd for samples size 'large.n'</p></dd>


<dt>plot</dt>
<dd><p>logical, if to make plots to show QC before and after filtration</p></dd>


<dt>lfc</dt>
<dd><p>num, cutoff of logFC for DE analysis</p></dd>


<dt>p</dt>
<dd><p>num, cutoff of p value for DE analysis and permutation test if
feature_selection = "rankproduct"</p></dd>


<dt>markers</dt>
<dd><p>vector, a vector of gene names, listed the gene symbols to be
kept anyway after filtration. Default 'NULL' means no special
genes need to be kept.</p></dd>


<dt>gene_id</dt>
<dd><p>character, specify the gene ID target_group of rownames of expression data
when markers is not NULL, could be one of 'ENSEMBL', 'SYMBOL',
'ENTREZ'..., default 'SYMBOL'</p></dd>


<dt>slot</dt>
<dd><p>character, specify which slot to use for DGEList,
default 'counts'</p></dd>


<dt>batch</dt>
<dd><p>vector of character, column name(s) of coldata to be treated as
batch effect factor, default NULL</p></dd>


<dt>summary</dt>
<dd><p>logical, if to show the summary of DE analysis</p></dd>


<dt>...</dt>
<dd><p>omitted</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>MArrayLM object generated by <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="fu">mastR</span><span class="fu">::</span><span class="va"><a href="im_data_6.html">im_data_6</a></span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      samples <span class="op">=</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="fu">mastR</span><span class="fu">::</span><span class="va"><a href="im_data_6.html">im_data_6</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">de_analysis</span><span class="op">(</span><span class="va">dge</span>, group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, target_group <span class="op">=</span> <span class="st">"NK"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No group or design set. Assuming all samples belong to one group.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Down             2530         2201       1673   1494   1099</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NotSig            737         1711       2738   2978   3727</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Up               2805         2160       1661   1600   1246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class "DGEList"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479438 GSM1479439 GSM1479440 GSM1479441 GSM1479442</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         41         18         12         14         20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         32          9         24         19         16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938       2804       1711        105          1         56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084          2         34         14         20         24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167        160         74         75         57         67</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479443 GSM1479499 GSM1479500 GSM1479501 GSM1479502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         15         48         23         16         24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         14         34          8         20         24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938        560       2085       1722        156          1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084         30          2         35         17         21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167         48        200         80         54         51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479503 GSM1479504 GSM1479506 GSM1479507 GSM1479508</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         20         18         39         24         18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         17         16         40          9         18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938         43        657       2406       1672        150</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084         20         38          2         35         19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167         56         52        194         72         65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479509 GSM1479510 GSM1479511 GSM1479520 GSM1479521</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         19         25         21         45         19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         18         12         15         33          7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938         22        176        573       2963       1529</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084         20         24         36          4         28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167         51         54         59        197         65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479522 GSM1479523 GSM1479524 GSM1479525</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         20         20         30         21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         17         18         14         15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938        217          2         63        494</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084         17         15         19         30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167         62         58         63         53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  group lib.size norm.factors  title geo_accession</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Neutrophils  1421890    0.5840488 lib226    GSM1479438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439   Monocytes   982216    0.9428708 lib227    GSM1479439</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440     B.cells   778374    1.0757535 lib228    GSM1479440</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441         CD4   715624    1.1080012 lib229    GSM1479441</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442         CD8   802285    1.0720010 lib230    GSM1479442</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           status submission_date last_update_date type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            channel_count source_name_ch1 organism_ch1 characteristics_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.1 characteristics_ch1.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438   cellcount: 14737500 celltype: Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439    cellcount: 2000000   celltype: Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440    cellcount: 1012333     celltype: B-cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441    cellcount: 1071990         celltype: CD4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442    cellcount: 1011154         celltype: CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   characteristics_ch1.3          characteristics_ch1.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.5 characteristics_ch1.6 characteristics_ch1.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438           donorid: 44             gender: F              index: 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439           donorid: 44             gender: F             index: 23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440           donorid: 44             gender: F              index: 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441           donorid: 44             gender: F             index: 25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442           donorid: 44             gender: F             index: 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.8      characteristics_ch1.9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438        race: Hispanic samplename: 44_Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439        race: Hispanic   samplename: 44_Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440        race: Hispanic      samplename: 44_Bcells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441        race: Hispanic        samplename: 44_CD4T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442        race: Hispanic        samplename: 44_CD8T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.10             characteristics_ch1.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 characteristics_ch1.12             characteristics_ch1.13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               characteristics_ch1.14 molecule_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     extract_protocol_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            taxid_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                 data_processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                       data_processing.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                              data_processing.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                              data_processing.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             data_processing.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                    data_processing.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            platform_id    contact_name                  contact_email</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_department           contact_institute contact_address</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_city contact_state contact_zip.postal_code contact_country</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            data_row_count  instrument_model library_selection library_source</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            library_strategy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                  relation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990322</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990323</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990324</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990331</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                      relation.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680546</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680547</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680548</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680549</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680550</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            supplementary_file_1 age.ch1 cellcount.ch1 celltype.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                 NONE      32      14737500  Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                 NONE      32       2000000    Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                 NONE      32       1012333      B-cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                 NONE      32       1071990          CD4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                 NONE      32       1011154          CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            collectiondate.ch1 diseasestatus.ch1 donorid.ch1 gender.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            index.ch1 race.ch1 samplename.ch1 smoker.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438         5 Hispanic 44_Neutrophils         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439        23 Hispanic   44_Monocytes         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440         6 Hispanic      44_Bcells         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441        25 Hispanic        44_CD4T         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442        12 Hispanic        44_CD8T         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            time.since.last.flare..months..ch1 time.since.steroid.dose.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            time.since.symptom.onset..yrs..ch1 years.since.diagnosis.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $original_counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479438 GSM1479439 GSM1479440 GSM1479441 GSM1479442</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000003          0          0          0          2          1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000005          0          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         41         18         12         14         20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         32          9         24         19         16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000460          4          5          9          4          6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479443 GSM1479499 GSM1479500 GSM1479501 GSM1479502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000003          0          0          0          0          3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000005          0          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         15         48         23         16         24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         14         34          8         20         24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000460          6          4          6         12         10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479503 GSM1479504 GSM1479506 GSM1479507 GSM1479508</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000003          2          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000005          0          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         20         18         39         24         18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         17         16         40          9         18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000460         12         10          7          4         12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479509 GSM1479510 GSM1479511 GSM1479520 GSM1479521</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000003          2          1          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000005          0          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         19         25         21         45         19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         18         12         15         33          7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000460          7          7          7          3          5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479522 GSM1479523 GSM1479524 GSM1479525</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000003          0          2          1          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000005          0          0          0          0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419         20         20         30         21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457         17         18         14         15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000460          9          7          7          6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50040 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $vfit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $targets</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  group lib.size norm.factors  title geo_accession</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Neutrophils 830453.1    0.5840488 lib226    GSM1479438</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439   Monocytes 926102.8    0.9428708 lib227    GSM1479439</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440     B.cells 837338.6    1.0757535 lib228    GSM1479440</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441         CD4 792912.3    1.1080012 lib229    GSM1479441</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442         CD8 860050.3    1.0720010 lib230    GSM1479442</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           status submission_date last_update_date type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Public on Jan 06 2015     Aug 14 2014      May 15 2019  SRA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            channel_count source_name_ch1 organism_ch1 characteristics_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442             1     Whole Blood Homo sapiens             age: 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.1 characteristics_ch1.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438   cellcount: 14737500 celltype: Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439    cellcount: 2000000   celltype: Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440    cellcount: 1012333     celltype: B-cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441    cellcount: 1071990         celltype: CD4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442    cellcount: 1011154         celltype: CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   characteristics_ch1.3          characteristics_ch1.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 collectiondate: June 26 2012 diseasestatus: Healthy Control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.5 characteristics_ch1.6 characteristics_ch1.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438           donorid: 44             gender: F              index: 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439           donorid: 44             gender: F             index: 23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440           donorid: 44             gender: F              index: 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441           donorid: 44             gender: F             index: 25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442           donorid: 44             gender: F             index: 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.8      characteristics_ch1.9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438        race: Hispanic samplename: 44_Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439        race: Hispanic   samplename: 44_Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440        race: Hispanic      samplename: 44_Bcells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441        race: Hispanic        samplename: 44_CD4T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442        race: Hispanic        samplename: 44_CD8T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            characteristics_ch1.10             characteristics_ch1.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442             smoker: -- time since last flare (months): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 characteristics_ch1.12             characteristics_ch1.13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 time since steroid dose: -- time since symptom onset (yrs): --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               characteristics_ch1.14 molecule_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 years since diagnosis: --    total RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     extract_protocol_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Fresh blood samples were collected from healthy subjects and subjects diagnosed type 1 diabetes, amyotrophic lateral sclerosis, and sepsis, as well as multiple sclerosis patients before and 24 hours after the first treatment with IFN-beta. At the time of blood draw, an aliquot of whole blood was collected into a Tempus tube (Invitrogen), while the remainder of the primary fresh blood sample was processed to highly pure populations of neutrophils, monocytes, B cells, CD4 T cells, CD8 T cells, and natural killer cells. RNA was extracted from each of these cell subsets, as well as the whole blood samples, and processed into RNA sequencing (RNAseq) libraries (Illumina TruSeq). Sequencing libraries were analyzed on an Illumina HiScan, with a target read depth of ~20M reads. Reads were demultiplexed, mapped to human gene models (ENSEMBL), and tabulated using HTSeq. Read count data were normalized by the TMM procedure (edgeR package).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            taxid_ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442      9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                 data_processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 fastq files were created from .bcl files via Illumina's CASAVA 1.8.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                       data_processing.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 reads were aligned to Hg19 via Omicsoft sequence aligner (OSA) version 2.0.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                              data_processing.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 gene counts were generated by HTSeq version 0.5.4p3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                              data_processing.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 TMMs were generated with Bioconductor package EdgeR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             data_processing.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Genome_build: Hg19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                    data_processing.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Supplementary_files_format_and_content: tab delimited file contains TMM normalized counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            platform_id    contact_name                  contact_email</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442    GPL15456 Scott,,Presnell SPresnell@benaroyaresearch.org</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_department           contact_institute contact_address</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 Systems Immunology Benaroya Research Institute 1201 Ninth Ave.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            contact_city contact_state contact_zip.postal_code contact_country</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442      Seattle            WA                   98101             USA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            data_row_count  instrument_model library_selection library_source</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442              0 Illumina HiScanSQ              cDNA transcriptomic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            library_strategy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442          RNA-Seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                  relation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990322</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990323</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990324</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 BioSample: https://www.ncbi.nlm.nih.gov/biosample/SAMN02990331</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                      relation.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680546</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680547</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680548</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680549</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442 SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX680550</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            supplementary_file_1 age.ch1 cellcount.ch1 celltype.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                 NONE      32      14737500  Neutrophils</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                 NONE      32       2000000    Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                 NONE      32       1012333      B-cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                 NONE      32       1071990          CD4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                 NONE      32       1011154          CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            collectiondate.ch1 diseasestatus.ch1 donorid.ch1 gender.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442       June 26 2012   Healthy Control          44          F</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            index.ch1 race.ch1 samplename.ch1 smoker.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438         5 Hispanic 44_Neutrophils         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439        23 Hispanic   44_Monocytes         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440         6 Hispanic      44_Bcells         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441        25 Hispanic        44_CD4T         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442        12 Hispanic        44_CD8T         --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            time.since.last.flare..months..ch1 time.since.steroid.dose.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                                 --                          --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            time.since.symptom.onset..yrs..ch1 years.since.diagnosis.ch1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442                                 --                        --</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $E</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479438 GSM1479439 GSM1479440 GSM1479441 GSM1479442</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419   5.643067   4.320208   3.899971  4.1927460   4.575057</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457   5.290395   3.358682   4.870825  4.6201673   4.261899</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938  11.721555  10.851800   6.977214  0.9197275   6.037684</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084   1.589956   5.219279   4.114096  4.6923171   4.832215</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167   7.594457   6.329923   6.494520  6.1802551   6.294321</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479443 GSM1479499 GSM1479500 GSM1479501 GSM1479502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419   4.344363   5.806401   4.720269   4.348242  4.8657561</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457   4.248148   5.315013   3.253143   4.661400  4.8657561</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938   9.520737  11.232666  10.915969   7.593866  0.8360088</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084   5.320904   1.528416   5.315428   4.433131  4.6773110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167   5.990080   7.853947   6.496597   6.072032  5.9375468</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479503 GSM1479504 GSM1479506 GSM1479507 GSM1479508</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419   4.712787   4.477929   5.432015   4.741275   4.414937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457   4.484518   4.312870   5.468084   3.374493   4.414937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938   5.798178   9.629323  11.360954  10.834356   7.439103</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084   4.712787   5.535263   1.450162   5.276312   4.490886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167   6.175414   5.982721   7.731860   6.306474   6.238907</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479509 GSM1479510 GSM1479511 GSM1479520 GSM1479521</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419   4.611609   4.965249   4.620666   5.641953   4.427209</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457   4.535660   3.936680   4.148597   5.200247   3.048697</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938   4.818060   7.756348   9.358051  11.667245  10.720651</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084   4.683758   4.907534   5.384226   2.304083   4.974696</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167   6.012707   6.061008   6.089219   7.759867   6.175229</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 GSM1479522 GSM1479523 GSM1479524 GSM1479525</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419   4.632546   4.692640   5.208811   4.684634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000457   4.404277   4.544541   4.136054   4.212566</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000938   8.039865   1.657016   6.266758   9.208196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001084   4.404277   4.289284   4.563476   5.189107</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000001167   6.240778   6.205453   6.266758   5.999836</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $weights</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          [,1]     [,2]     [,3]     [,4]     [,5]     [,6]     [,7]     [,8]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,] 23.52843 18.48717 16.13976 17.30505 19.65489 16.58688 23.87302 18.14619</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 21.81077 11.50456 17.79285 17.63470 15.67051 14.85962 22.16243 11.33712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,] 15.99580 17.33898 28.92827 11.33712 28.15291 22.83036 15.84055 17.49110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,] 11.33712 22.01323 16.35964 17.27292 19.05162 21.33154 11.33712 21.69631</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,] 28.45129 28.03019 27.06651 25.71295 26.94764 25.02412 28.27501 27.79304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          [,9]    [,10]    [,11]    [,12]    [,13]    [,14]    [,15]    [,16]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,] 15.85911 17.84322 18.90439 17.35320 24.32060 18.39060 16.44432 17.36002</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 17.48626 18.17182 14.90869 15.52740 22.60091 11.43514 18.11891 17.68869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,] 29.00207 11.33712 27.56602 22.36070 15.64512 17.38263 28.83397 11.33712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,] 16.07418 17.81118 18.22607 22.03148 11.33712 21.92405 16.66949 17.32775</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,] 26.82649 26.17149 26.23093 25.70079 28.02760 27.96524 27.31005 25.75954</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [,17]    [,18]    [,19]    [,20]    [,21]    [,22]    [,23]    [,24]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,] 19.25672 17.82953 24.28634 18.29810 16.02801 17.30298 19.34119 17.41836</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 15.24790 15.95287 22.56914 11.36871 17.67316 17.63267 15.32948 15.58457</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3,] 27.83933 22.07950 15.65981 17.42484 28.96007 11.33712 27.90424 22.32211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4,] 18.60834 22.45537 11.33712 21.83542 16.24599 17.27085 18.69976 22.08878</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5,] 26.56539 26.10726 28.04630 27.89798 26.97126 25.71120 26.64520 25.75578</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $design</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            NK Neutrophils Monocytes B.cells CD4 CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438  0           1         0       0   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439  0           0         1       0   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440  0           0         0       1   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441  0           0         0       0   1   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442  0           0         0       0   0   1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $tfit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coefficients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NK-Neutrophils NK-Monocytes NK-B.cells      NK-CD4     NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000419      -1.095206  -0.01680197  0.2110171 -0.05809146 -0.3303203</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000457      -1.088967   0.97055237 -0.3572824 -0.41337101  0.0262364</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000938      -2.066211  -1.40087125  1.9171064  7.37217240  2.9648155</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001084       3.639531   0.16131804  0.9968626  0.77122334  0.6032395</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001167      -1.718854  -0.31097227 -0.2460316 -0.06739113 -0.1839269</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $stdev.unscaled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NK-Neutrophils NK-Monocytes NK-B.cells    NK-CD4    NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000419      0.1576995    0.1676062  0.1731017 0.1696399 0.1655715</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000457      0.1654286    0.1950812  0.1738364 0.1738042 0.1802781</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000938      0.1643141    0.1597551  0.1407226 0.1822446 0.1418917</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001084      0.1828308    0.1510240  0.1633341 0.1603945 0.1574262</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001167      0.1364293    0.1367530  0.1378117 0.1393668 0.1383734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $sigma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1.0207698 0.6732638 3.3887698 0.8461538 0.6360073</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more elements ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $df.residual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 18 18 18 18 18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more elements ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $cov.coefficients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Contrasts        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK-Neutrophils           0.50         0.25       0.25   0.25   0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK-Monocytes             0.25         0.50       0.25   0.25   0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK-B.cells               0.25         0.25       0.50   0.25   0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK-CD4                   0.25         0.25       0.25   0.50   0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK-CD8                   0.25         0.25       0.25   0.25   0.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $pivot</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1 2 3 4 5 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $Amean</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENSG00000000419 ENSG00000000457 ENSG00000000938 ENSG00000001084 ENSG00000001167 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        4.749181        4.373652        7.965056        4.328704        6.437080 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more elements ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $method</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ls"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $design</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            NK Neutrophils Monocytes B.cells CD4 CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479438  0           1         0       0   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479439  0           0         1       0   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479440  0           0         0       1   0   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479441  0           0         0       0   1   0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GSM1479442  0           0         0       0   0   1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Levels        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   NK                       1            1          1      1      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Neutrophils             -1            0          0      0      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Monocytes                0           -1          0      0      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   B.cells                  0            0         -1      0      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   CD4                      0            0          0     -1      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   CD8                      0            0          0      0     -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $df.prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 4.585584</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $s2.prior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.8251309</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $s2.post</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.9979457 0.5287807 9.3197235 0.7381381 0.4899055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more elements ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $df.total</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 22.58558 22.58558 22.58558 22.58558 22.58558</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more elements ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $t</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NK-Neutrophils NK-Monocytes NK-B.cells     NK-CD4     NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000419      -6.952034   -0.1003498   1.220290 -0.3427920 -1.9970831</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000457      -9.052453    6.8417284  -2.826396 -3.2707098  0.2001352</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000938      -4.119061   -2.8723805   4.462527 13.2507168  6.8444597</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001084      23.170066    1.2432774   7.103783  5.5965687  4.4600928</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001167     -18.000107   -3.2488434  -2.550636 -0.6908555 -1.8990510</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $p.value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   NK-Neutrophils NK-Monocytes   NK-B.cells       NK-CD4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000419   4.830559e-07 9.209521e-01 2.349437e-01 7.349249e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000457   5.673866e-09 6.199504e-07 9.674042e-03 3.413958e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000938   4.314141e-04 8.703347e-03 1.845372e-04 3.916390e-12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001084   3.070262e-17 2.265173e-01 3.435909e-07 1.146976e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001167   6.927190e-15 3.596825e-03 1.801611e-02 4.966960e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  Contrasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         NK-CD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000419 5.800431e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000457 8.431659e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000000938 6.161205e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001084 1.856517e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSG00000001167 7.040557e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6067 more rows ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $treat.lfc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jinjin Chen.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

