<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ mastR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mastR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/mastR_Demo.html">mastR: Markers Automated Screening Tool in R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/DavisLaboratory/mastR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="DEGs_Group.html">DEGs_Group()</a></code> </p>
        </td>
        <td><p>return DEGs UP and DOWN list based on intersection or union of comparisons</p></td>
      </tr><tr><td>
          <p><code><a href="DEGs_RP.html">DEGs_RP()</a></code> </p>
        </td>
        <td><p>return DEGs UP and DOWN list based on Rank Product</p></td>
      </tr><tr><td>
          <p><code><a href="ccle_2_wide.html">ccle_2_wide()</a></code> </p>
        </td>
        <td><p>Convert CCLE data from long data to wide data.</p></td>
      </tr><tr><td>
          <p><code><a href="ccle_crc_5.html">ccle_crc_5</a></code> </p>
        </td>
        <td><p>RNA-seq TPM data of 5 CRC cell line samples from CCLE.</p></td>
      </tr><tr><td>
          <p><code><a href="de_analysis.html">de_analysis()</a></code> </p>
        </td>
        <td><p>DE analysis pipeline</p></td>
      </tr><tr><td>
          <p><code><a href="filter_subset_sig.html">filter_subset_sig()</a></code> </p>
        </td>
        <td><p>Filter specific cell type signature genes against other subsets.</p></td>
      </tr><tr><td>
          <p><code><a href="get_de_table.html">get_de_table()</a></code> </p>
        </td>
        <td><p>Get DE analysis result table(s) with statistics</p></td>
      </tr><tr><td>
          <p><code><a href="get_degs.html">get_degs()</a></code> </p>
        </td>
        <td><p>Get differentially expressed genes by comparing specified groups</p></td>
      </tr><tr><td>
          <p><code><a href="get_gsc_sig.html">get_gsc_sig()</a></code> </p>
        </td>
        <td><p>Collect genes from MSigDB or provided GeneSetCollection.</p></td>
      </tr><tr><td>
          <p><code><a href="get_lm_sig.html">get_lm_sig()</a></code> </p>
        </td>
        <td><p>Extract specific subset markers from LM7 or/and LM22</p></td>
      </tr><tr><td>
          <p><code><a href="get_panglao_sig.html">get_panglao_sig()</a></code> </p>
        </td>
        <td><p>Extract immune subset markers from PanglaoDB website.</p></td>
      </tr><tr><td>
          <p><code><a href="gls2gsc.html">gls2gsc()</a></code> </p>
        </td>
        <td><p>Convert gene-set list into GeneSetCollection</p></td>
      </tr><tr><td>
          <p><code><a href="gsc_plot.html">gsc_plot()</a></code> </p>
        </td>
        <td><p>Make upset plot for given gene sets</p></td>
      </tr><tr><td>
          <p><code><a href="im_data_6.html">im_data_6</a></code> </p>
        </td>
        <td><p>RNA-seq TMM normalized counts data of 6 sorted immune subsets.</p></td>
      </tr><tr><td>
          <p><code><a href="list_panglao_organs.html">list_panglao_organs()</a></code> </p>
        </td>
        <td><p>Show the summary info of available organs in PanglaoDB.</p></td>
      </tr><tr><td>
          <p><code><a href="list_panglao_types.html">list_panglao_types()</a></code> </p>
        </td>
        <td><p>Show the summary info of available cell types in PanglaoDB.</p></td>
      </tr><tr><td>
          <p><code><a href="lm22.html">lm22</a></code> </p>
        </td>
        <td><p>LM22 matrix for CIBERSORT.</p></td>
      </tr><tr><td>
          <p><code><a href="lm7.html">lm7</a></code> </p>
        </td>
        <td><p>LM7 matrix for CIBERSORT.</p></td>
      </tr><tr><td>
          <p><code><a href="merge_markers.html">merge_markers()</a></code> </p>
        </td>
        <td><p>Merge markers list into one.</p></td>
      </tr><tr><td>
          <p><code><a href="msigdb_gobp_nk.html">msigdb_gobp_nk</a></code> </p>
        </td>
        <td><p>Sub-collection of MSigDB gene sets.</p></td>
      </tr><tr><td>
          <p><code><a href="nk_markers.html">nk_markers</a></code> </p>
        </td>
        <td><p>NK cell markers combination.</p></td>
      </tr><tr><td>
          <p><code><a href="pca_matrix_plot.html">pca_matrix_plot()</a></code> </p>
        </td>
        <td><p>Make a matrix plot of PCA with top PCs</p></td>
      </tr><tr><td>
          <p><code><a href="pca_matrix_plot_init.html">pca_matrix_plot_init()</a></code> </p>
        </td>
        <td><p>Make a matrix plot of PCA with top PCs</p></td>
      </tr><tr><td>
          <p><code><a href="plotPCAbiplot.html">plotPCAbiplot()</a></code> </p>
        </td>
        <td><p>Single PCA plot function</p></td>
      </tr><tr><td>
          <p><code><a href="plot_diagnostics.html">plot_diagnostics()</a></code> </p>
        </td>
        <td><p>plot diagnostics before and after <code><a href="../reference/process_data.html">process_data()</a></code></p></td>
      </tr><tr><td>
          <p><code><a href="plot_mean_var.html">plot_mean_var()</a></code> </p>
        </td>
        <td><p>plot Mean-variance trend after voom and after final linear fit</p></td>
      </tr><tr><td>
          <p><code><a href="process_data.html">process_data()</a></code> </p>
        </td>
        <td><p>process data</p></td>
      </tr><tr><td>
          <p><code><a href="pseudo_sample_list.html">pseudo_sample_list()</a></code> </p>
        </td>
        <td><p>Split cells according to specific factors</p></td>
      </tr><tr><td>
          <p><code><a href="pseudo_samples.html">pseudo_samples()</a></code> </p>
        </td>
        <td><p>Aggregate single cells to pseudo-samples according to specific factors</p></td>
      </tr><tr><td>
          <p><code><a href="remove_bg_exp.html">remove_bg_exp()</a></code> </p>
        </td>
        <td><p>Remove markers with high signal in background data.</p></td>
      </tr><tr><td>
          <p><code><a href="remove_bg_exp_mat.html">remove_bg_exp_mat()</a></code> </p>
        </td>
        <td><p>Remove genes show high signal in the background expression data from markers.</p></td>
      </tr><tr><td>
          <p><code><a href="select_sig.html">select_sig()</a></code> </p>
        </td>
        <td><p>select DEGs from multiple comparisons</p></td>
      </tr><tr><td>
          <p><code><a href="sig_boxplot.html">sig_boxplot()</a></code> </p>
        </td>
        <td><p>Boxplot of median expression or scores of signature</p></td>
      </tr><tr><td>
          <p><code><a href="sig_gseaplot.html">sig_gseaplot()</a></code> </p>
        </td>
        <td><p>Visualize GSEA result with input list of gene symbols.</p></td>
      </tr><tr><td>
          <p><code><a href="sig_heatmap.html">sig_heatmap()</a></code> </p>
        </td>
        <td><p>Heatmap original markers and screened signature</p></td>
      </tr><tr><td>
          <p><code><a href="sig_rankdensity_plot.html">sig_rankdensity_plot()</a></code> </p>
        </td>
        <td><p>Plot rank density</p></td>
      </tr><tr><td>
          <p><code><a href="sig_scatter_plot.html">sig_scatter_plot()</a></code> </p>
        </td>
        <td><p>Scatter plot of signature for specific subset vs others</p></td>
      </tr><tr><td>
          <p><code><a href="voom_fit_treat.html">voom_fit_treat()</a></code> </p>
        </td>
        <td><p>return DGEList containing vfit by limma::voom (if normalize = TRUE) and
tfit by limma::treat</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jinjin Chen, Ahmed Mohamed.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

