<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mastR_Demo • mastR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mastR_Demo">
<meta property="og:description" content="mastR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mastR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/mastR_Demo.html">mastR_Demo</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/mastR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mastR_Demo</h1>
                        <h4 data-toc-skip class="author">Jinjin
Chen</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical
Biology, University of Melbourne, Parkville, VIC 3010,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:chen.j@wehi.edu.au" class="email">chen.j@wehi.edu.au</a>
      </address>
                  
            <h4 data-toc-skip class="date">09 Mar 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/mastR/blob/HEAD/vignettes/mastR_Demo.Rmd" class="external-link"><code>vignettes/mastR_Demo.Rmd</code></a></small>
      <div class="hidden name"><code>mastR_Demo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr>
<ul>
<li><em>Why do we need imm sigs (cell type)</em></li>
</ul>
<p>Tumour immune microenvironment (TIME) is reported to be highly
associated with prognosis and various treatment response to many kinds
of cancers. It’s also proved that tumour infiltrating lymphocytes (TILs)
play an essential role in tumour progression, metastasis and treatment
response. Thus, to identify the extent or population of the specific
immune subset in specific tissue type, would be an interesting topic for
many immunologists.</p>
<p>But to annotate cell types, estimate cellular composition or identify
sample states (activities), refined signature is needed for the
specified cells, which always requires large amount of manual work of
experts. To save time and reduce manual labor work, mastR is developed
to automatically screen signature for users interest.</p>
<ul>
<li><em>What this package does</em></li>
</ul>
<p>This report demonstrates how to use the functions in R package
<code>mastR 0.1.0</code> to screen immune subset signatures in specific
tissue and visualize the effects of the curation.</p>
<p><code>mastR</code> is developed to screen immune cells signature in
cancer tissues. So there are several datasets imported into this package
for running by default. But users could specify other immune subsets and
other tissues by setting the parameters in the R functions. Next, this
report will show how the workflow runs to refine NK cells signatures
which are specific for CRC, and how to visualize the results effect.</p>
<ul>
<li><p>How does mastR screen the signature<br></p></li>
<li><p>step 1. markers pool generation<br>
Markers pool can be built from MSigDB, PanglaoDB or LM7/LM22 by our
imported functions, if users have any preliminary knowledge about the
target group type (cell type). All genes in the pool will be reserved
for DE analysis even failed the filtration.<br>
The final signature will only be the intersected genes with the markers
pool. The whole gene list of the data will be regarded as the markers
pool if no preliminary result or intereted genes are provided. But
<em>note</em> that <code>markers = NULL</code> won’t keep any special
genes if they fail the filtration by edgeR.</p></li>
<li>
<p>step 2. screen markers for group specificity</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>process data<br>
Data will be filtered by given cutoff, all gene with low expression will
be removed by edgeR. If the given data is raw count data, further
normalization ‘TMM’ and fit by <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code> will be
done.<br>
Then final linear model will be fitted and passed to
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code> to compute gene statistics for DE
analysis.<br>
</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>DE analysis<br>
</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>feature selection based on rank product<br>
</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>intersect with markers pool</li>
</ol></li>
</ul>
</li>
<li>
<p>step 3. screen markers for cancer/tissue specificity</p>
<ul>
<li>filter out cancer cell expressed genes<br>
Collect all interested cancer cell lines from CCLE (default), compute
the quantile (25% percentile default) expression of all non-0 expressed
genes. Only keep genes with median expression &lt; quantile cutoff in
cancer cells. Then keep the intersected genes with markers pool.<br>
By doing this, we can guarantee the activity of our final signature for
infiltrating immune cells won’t be affected by tumor purity when it’s
applied on tumor samples.<br>
</li>
<li>optional DE analysis to compare tumor infiltrating immune cells vs
normal immune cells (e.g. PBMC or normal tissue resident immune cells),
add DEGs into markers pool or combine it with the result of step 2. to
increase tissue/cancer specificity.</li>
</ul>
</li>
<li>
<p>step 4. combine results of step 2 &amp; 3</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>single dataset<br>
Keep the intersection of step 2 and 3 results.<br>
</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>multiple datasets After 4. a), choose one assemble method to
aggregate signature lists generated by different datasets. Robust Rank
Aggregation (“RRA”) is the default method to do this in mastR, which
detects genes that are ranked consistently better than expected under
null hypothesis of uncorrelated inputs and assigns a significance score
for each gene. It’s better than “RemoveBatchEffect” sometimes.</li>
</ol></li>
</ul>
</li>
<li>
<p>step 5. visualize signature performance</p>
<ul>
<li>Use multiple visualization ways to assess the performance of
screened signature.</li>
</ul>
</li>
<li>
<p>Applications</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>score samples</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>estimate cellular proportion</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>single cell annotation</li>
</ol></li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<hr>
<p>Install <code>mastR</code> R package from <a href="https://github.com/Gene233/mastR" class="external-link">GitHub</a>. The most updated
version of <code>mastR</code> is hosted on GitHub and can be easily
installed using <code><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">devtools::install_github()</a></code> function
provided by <a href="https://cran.r-project.org/package=devtools" class="external-link">devtools</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!requireNamespace("devtools", quietly = TRUE)) {</span></span>
<span><span class="co">#   install.packages("devtools")</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># if (!requireNamespace("mastR", quietly = TRUE)) {</span></span>
<span><span class="co">#   devtools::install_github("Gene233/mastR")</span></span>
<span><span class="co"># }</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"mastR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"mastR"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1--prepare-immune-markers-pool">Step 1. Prepare Immune Markers Pool<a class="anchor" aria-label="anchor" href="#step-1--prepare-immune-markers-pool"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="load-datasets">Load Datasets<a class="anchor" aria-label="anchor" href="#load-datasets"></a>
</h3>
<p>Here’s step 1. To screen the immune subset signatures, we have to
load the example datasets. The datasets used in this report have been
built within the package or can be accessed publicly. You can use the
following scripts to load them into your R environment.</p>
<p><code>NK_markers</code> dataset is a combination of CIBERSORT LM7,
LM22 and human orthologs in mice from <a href="https://cancerimmunolres.aacrjournals.org/content/7/7/1162.long" class="external-link">Huntington</a>.
It contains 114 genes in total.</p>
<p><code>LM7</code> and <code>LM22</code> are signature matrices, you
can get more details by <code><a href="../reference/LM7.html">?mastR::LM7</a></code> or
<code><a href="../reference/LM22.html">?mastR::LM22</a></code>.</p>
<p><code>im_data_6</code> is a eSet object, containing RNA-seq TMM
normalized counts data of 6 sorted immune subsets. More details in
<code><a href="../reference/im_data_6.html">?mastR::im_data_6</a></code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/mastR" class="external-link">mastR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; Loading required package: annotate</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationDbi</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: XML</span></span>
<span><span class="co">#&gt; Loading required package: graph</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'graph'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:XML':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     addNode</span></span></code></pre></div>
<div class="section level4">
<h4 id="nk-markers-from-cursons-publication">NK markers from <a href="https://aacrjournals.org/cancerimmunolres/article/7/7/1162/469488/A-Gene-Signature-Predicting-Natural-Killer-Cell" class="external-link">Curson’s
Publication</a><a class="anchor" aria-label="anchor" href="#nk-markers-from-cursons-publication"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show what NK_markers looks like:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"NK_markers"</span><span class="op">)</span></span>
<span><span class="va">NK_markers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 114 × 4</span></span></span>
<span><span class="co">#&gt;    HGNC_Symbol LM22  LM7   Huntington</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> APOBEC3G    TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> APOL6       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> AZU1        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> BPI         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CAMP        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CCL4        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CCL5        TRUE  -     TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CCND2       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CD160       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CD2         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 104 more rows</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="markers-from-lm">Markers from LM<a class="anchor" aria-label="anchor" href="#markers-from-lm"></a>
</h4>
<p>Users can extract markers for subsets matched to the given regex
pattern from imported data <code>LM7</code>/<code>LM22</code>.</p>
<p>The matched genes will be saved in ‘GeneSet’ class object, if both
pattern are provided, the output would be a ‘GeneSetCollection’ class
object with setName: LM7, LM22.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"LM7"</span>, <span class="st">"LM22"</span><span class="op">)</span></span>
<span><span class="co">## only retrieve LM7</span></span>
<span><span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm7.pattern <span class="op">=</span> <span class="st">"^NK"</span><span class="op">)</span></span>
<span><span class="co">#&gt; setName: LM7 </span></span>
<span><span class="co">#&gt; geneIds: CD244, FASLG, ..., XCL2 (total: 21)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Null </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span></span>
<span><span class="co">## only retrieve LM22</span></span>
<span><span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm22.pattern <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span><span class="co">#&gt; setName: LM22 </span></span>
<span><span class="co">#&gt; geneIds: APOBEC3G, APOL6, ..., ZNF135 (total: 79)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Null </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span></span>
<span><span class="co">## collect both LM7 and LM22</span></span>
<span><span class="va">LM</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm7.pattern <span class="op">=</span> <span class="st">"^NK"</span>, lm22.pattern <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: LM7, LM22 (2 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: CD244, FASLG, ..., ZNF135 (92 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span>
<span><span class="co">## show upset diagram</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">LM</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/LM%20markers-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="markers-from-msigdb">Markers from MSigDB<a class="anchor" aria-label="anchor" href="#markers-from-msigdb"></a>
</h4>
<p>The users can specify the <code>species</code>, <code>cat</code>,
<code>subcat</code> and <code>pattern</code> to decide which genesets to
be added.</p>
<p>All genesets with geneset name matched to the <code>pattern</code> in
the whole MSigDB would be extracted if <code>cat</code> and
<code>subcat</code> are not set, otherwise only the genesets under the
union of <code>cat</code> and <code>subcat</code> would be searched.</p>
<p>By setting <code>plot = TRUE</code>, an UpSetR plot across matched
genesets would be printed.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## collect all "natural killer mediated" relavent genesets from MSigDB</span></span>
<span><span class="co"># MSig &lt;- get_gsc_sig(gsc = "msigdb",</span></span>
<span><span class="co">#                     pattern = "NATURAL_KILLER_CELL_MEDIATED",</span></span>
<span><span class="co">#                     cat = "c5", subcat = "GO:BP",</span></span>
<span><span class="co">#                     version = '7.4',</span></span>
<span><span class="co">#                     species = "hs")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"msigdb_gobp_nk"</span><span class="op">)</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gsc_sig.html">get_gsc_sig</a></span><span class="op">(</span>gsc <span class="op">=</span> <span class="va">msigdb_gobp_nk</span>,</span>
<span>                    pattern <span class="op">=</span> <span class="st">"NATURAL_KILLER_CELL_MEDIATED"</span><span class="op">)</span></span>
<span><span class="va">MSig</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY, GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET, ..., GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY (18 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: AP1G1, ARRB2, ..., KLRC4-KLRK1 (67 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: BroadCollection (1 total)</span></span>
<span><span class="co">## show upset diagram of collected gene-sets</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/MSigDB%20genesets-1.png" width="1152"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## merge all genesets into one</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">setName</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"MSigDB"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="markers-from-panglaodb">Markers from PanglaoDB<a class="anchor" aria-label="anchor" href="#markers-from-panglaodb"></a>
</h4>
<p>Users can use <code><a href="../reference/list_panglao_organs.html">list_panglao_organs()</a></code> and
<code><a href="../reference/list_panglao_types.html">list_panglao_types()</a></code> functions to list all available organs
and cell types on PanglaoDB website and use
<code><a href="../reference/get_panglao_sig.html">get_panglao_sig()</a></code> function to retrive them.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show availbable organs on PanglaoDB</span></span>
<span><span class="fu"><a href="../reference/list_panglao_organs.html">list_panglao_organs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "Adrenal glands"     "Blood"              "Bone"              </span></span>
<span><span class="co">#&gt;  [4] "Brain"              "Connective tissue"  "Embryo"            </span></span>
<span><span class="co">#&gt;  [7] "Epithelium"         "Eye"                "GI tract"          </span></span>
<span><span class="co">#&gt; [10] "Heart"              "Immune system"      "Kidney"            </span></span>
<span><span class="co">#&gt; [13] "Liver"              "Lungs"              "Mammary gland"     </span></span>
<span><span class="co">#&gt; [16] "Olfactory system"   "Oral cavity"        "Pancreas"          </span></span>
<span><span class="co">#&gt; [19] "Parathyroid glands" "Placenta"           "Reproductive"      </span></span>
<span><span class="co">#&gt; [22] "Skeletal muscle"    "Skin"               "Smooth muscle"     </span></span>
<span><span class="co">#&gt; [25] "Thymus"             "Thyroid"            "Urinary bladder"   </span></span>
<span><span class="co">#&gt; [28] "Vasculature"        "Zygote"             "Other"</span></span>
<span></span>
<span><span class="co">## show available cell types of interest organ on PanglaoDB</span></span>
<span><span class="co">## Number in the bracket represents the number of markers for each cell type (in both Homo and Mus).</span></span>
<span><span class="fu"><a href="../reference/list_panglao_types.html">list_panglao_types</a></span><span class="op">(</span>organ <span class="op">=</span> <span class="st">"Immune system"</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "B cells (110)"                        </span></span>
<span><span class="co">#&gt;  [2] "B cells memory (66)"                  </span></span>
<span><span class="co">#&gt;  [3] "B cells naive (69)"                   </span></span>
<span><span class="co">#&gt;  [4] "Basophils (82)"                       </span></span>
<span><span class="co">#&gt;  [5] "Dendritic cells (133)"                </span></span>
<span><span class="co">#&gt;  [6] "Eosinophils (30)"                     </span></span>
<span><span class="co">#&gt;  [7] "Gamma delta T cells (66)"             </span></span>
<span><span class="co">#&gt;  [8] "Macrophages (153)"                    </span></span>
<span><span class="co">#&gt;  [9] "Mast cells (162)"                     </span></span>
<span><span class="co">#&gt; [10] "Megakaryocytes (48)"                  </span></span>
<span><span class="co">#&gt; [11] "Monocytes (102)"                      </span></span>
<span><span class="co">#&gt; [12] "Myeloid-derived suppressor cells (17)"</span></span>
<span><span class="co">#&gt; [13] "Natural killer T cells (24)"          </span></span>
<span><span class="co">#&gt; [14] "Neutrophils (80)"                     </span></span>
<span><span class="co">#&gt; [15] "NK cells (98)"                        </span></span>
<span><span class="co">#&gt; [16] "Nuocytes (12)"                        </span></span>
<span><span class="co">#&gt; [17] "Plasma cells (86)"                    </span></span>
<span><span class="co">#&gt; [18] "Plasmacytoid dendritic cells (58)"    </span></span>
<span><span class="co">#&gt; [19] "Red pulp macrophages (12)"            </span></span>
<span><span class="co">#&gt; [20] "T cells (107)"                        </span></span>
<span><span class="co">#&gt; [21] "T cytotoxic cells (8)"                </span></span>
<span><span class="co">#&gt; [22] "T follicular helper cells (13)"       </span></span>
<span><span class="co">#&gt; [23] "T helper cells (62)"                  </span></span>
<span><span class="co">#&gt; [24] "T memory cells (63)"                  </span></span>
<span><span class="co">#&gt; [25] "T regulatory cells (21)"</span></span>
<span></span>
<span><span class="co">## collect all "NK cells" markers from PanglaoDB website</span></span>
<span><span class="va">Panglao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_panglao_sig.html">get_panglao_sig</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Panglao</span></span>
<span><span class="co">#&gt; setName: PanglaoDB </span></span>
<span><span class="co">#&gt; geneIds: ITGAM, ITGAX, ..., CD8A (total: 80)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Null </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span><span class="co">## number differs from 'NK cells' under list_panglao_types(organ = "Immune system"), because we only keep 'Hs' markers.</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="integrate-markers-pool">Integrate Markers Pool<a class="anchor" aria-label="anchor" href="#integrate-markers-pool"></a>
</h3>
<p>Finally, all markers can be merged into one ‘GeneSet’ object by using
function <code><a href="../reference/merge_markers.html">merge_markers()</a></code>.</p>
<p>The input can be a list of vectors of genes, or a list of GeneSet
objects, or a GeneSetCollection object. <code>plot = TRUE</code>
requires the list to have &gt; 1 elements.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nk_m</span> <span class="op">&lt;-</span> <span class="va">NK_markers</span><span class="op">$</span><span class="va">HGNC_Symbol</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html" class="external-link">GeneSet</a></span><span class="op">(</span>geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span>, setName <span class="op">=</span> <span class="st">"NK_markers"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gsc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html" class="external-link">GeneSetCollection</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">nk_m</span>, <span class="va">LM</span>, <span class="va">MSig</span>, <span class="va">Panglao</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## upset plot</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/integrate%20markers-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Markers</span></span>
<span><span class="co">#&gt; setName: merged_markers_pool </span></span>
<span><span class="co">#&gt; geneIds: APOBEC3G, APOL6, ..., CD8A (total: 200)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Computed </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span></span>
<span><span class="co">## to show the table summary of merged list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/jsonlite/man/fromJSON.html" class="external-link">fromJSON</a></span><span class="op">(</span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">longDescription</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       Gene NK_markers LM7 LM22 MSigDB PanglaoDB</span></span>
<span><span class="co">#&gt; 1 ADAMTS14          -   -    -      -      TRUE</span></span>
<span><span class="co">#&gt; 2      AHR          -   -    -      -      TRUE</span></span>
<span><span class="co">#&gt; 3    AP1G1          -   -    -   TRUE         -</span></span>
<span><span class="co">#&gt; 4 APOBEC3G       TRUE   - TRUE      -         -</span></span>
<span><span class="co">#&gt; 5    APOL6       TRUE   - TRUE      -         -</span></span>
<span><span class="co">#&gt; 6    ARRB2          -   -    -   TRUE         -</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="screen-immune-subset-signatures">Screen Immune Subset Signatures<a class="anchor" aria-label="anchor" href="#screen-immune-subset-signatures"></a>
</h2>
<hr>
<div class="section level3">
<h3 class="tabset tabset-fade tabset-pills" id="step-2--specify-markers-against-other-immune-subsets">Step 2. Specify Markers
Against Other Immune Subsets<a class="anchor" aria-label="anchor" href="#step-2--specify-markers-against-other-immune-subsets"></a>
</h3>
<p>You can use <code>celldex::DatabaseImmuneCellExpressionData()</code>
to load a large collection of 1561 bulk RNA-seq samples generated by
DICE from pure populations of human immune cells. It contains 5 main
cell types and 15 fine cell types. Here to save time, we use our example
data <code>im_data_6</code> as input.</p>
<p>The example uses <code>im_data_6</code> from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60424" class="external-link">GSE60424</a>.</p>
<ul>
<li>
<code>im_data_6</code> consists 6 immune subsets * 4 samples, the
data type is ‘TMM normalized counts data’.</li>
</ul>
<p>Only samples from healthy individuals are kept, and ‘Whole Blood’ or
‘PBMC’ cells are removed in dataset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"im_data_6"</span><span class="op">)</span></span>
<span><span class="va">im_data_6</span></span>
<span><span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">#&gt; assayData: 50045 features, 24 samples </span></span>
<span><span class="co">#&gt;   element names: exprs </span></span>
<span><span class="co">#&gt; protocolData: none</span></span>
<span><span class="co">#&gt; phenoData</span></span>
<span><span class="co">#&gt;   sampleNames: GSM1479438 GSM1479439 ... GSM1479525 (24 total)</span></span>
<span><span class="co">#&gt;   varLabels: title geo_accession ... years since diagnosis:ch1 (66</span></span>
<span><span class="co">#&gt;     total)</span></span>
<span><span class="co">#&gt;   varMetadata: labelDescription</span></span>
<span><span class="co">#&gt; featureData: none</span></span>
<span><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">#&gt;   pubMedIds: 25314013 </span></span>
<span><span class="co">#&gt; Annotation: GPL15456</span></span>
<span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"lib.size"</span>, <span class="st">"donorid.ch1"</span>, <span class="st">"smoker.ch1"</span>,</span>
<span>  <span class="st">"cellcount.ch1"</span>, <span class="st">"collectiondate.ch1"</span>,</span>
<span>  <span class="st">"celltype.ch1"</span>, <span class="st">"gender.ch1"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">im_data_6</span><span class="op">)</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">im_data_6</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">[[</span><span class="st">"samples"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">idx</span><span class="op">]</span></span>
<span><span class="co">#&gt;            lib.size donorid.ch1 smoker.ch1 cellcount.ch1 collectiondate.ch1</span></span>
<span><span class="co">#&gt; GSM1479438  1800344          44         --      14737500       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479439  1198738          44         --       2000000       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479440   917827          44         --       1012333       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479441   815853          44         --       1071990       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479442   911138          44         --       1011154       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479443   743059          44         --        379236       June 26 2012</span></span>
<span><span class="co">#&gt; GSM1479499  1985823          20         --      42250000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479500  1113175          20         --       2000000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479501   823759          20         --       2000000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479502   802268          20         --       2000000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479503   761037          20         --       1500000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479504   761050          20         --        285000    January 25 2012</span></span>
<span><span class="co">#&gt; GSM1479506  1888834          21         --       3700000    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479507  1137698          21         --       2000000    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479508   907916          21         --        363469    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479509   789451          21         --       2000000    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479510   799980          21         --       2000000    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479511   828570          21         --        147461    February 2 2012</span></span>
<span><span class="co">#&gt; GSM1479520  1876949          53         --      17887500     August 21 2012</span></span>
<span><span class="co">#&gt; GSM1479521  1157574          53         --       2000000     August 21 2012</span></span>
<span><span class="co">#&gt; GSM1479522   895130          53         --        682371     August 21 2012</span></span>
<span><span class="co">#&gt; GSM1479523   801045          53         --       2000000     August 21 2012</span></span>
<span><span class="co">#&gt; GSM1479524   880511          53         --       2000000     August 21 2012</span></span>
<span><span class="co">#&gt; GSM1479525   818890          53         --        140773     August 21 2012</span></span>
<span><span class="co">#&gt;            celltype.ch1 gender.ch1</span></span>
<span><span class="co">#&gt; GSM1479438  Neutrophils          F</span></span>
<span><span class="co">#&gt; GSM1479439    Monocytes          F</span></span>
<span><span class="co">#&gt; GSM1479440      B-cells          F</span></span>
<span><span class="co">#&gt; GSM1479441          CD4          F</span></span>
<span><span class="co">#&gt; GSM1479442          CD8          F</span></span>
<span><span class="co">#&gt; GSM1479443           NK          F</span></span>
<span><span class="co">#&gt; GSM1479499  Neutrophils          F</span></span>
<span><span class="co">#&gt; GSM1479500    Monocytes          F</span></span>
<span><span class="co">#&gt; GSM1479501      B-cells          F</span></span>
<span><span class="co">#&gt; GSM1479502          CD4          F</span></span>
<span><span class="co">#&gt; GSM1479503          CD8          F</span></span>
<span><span class="co">#&gt; GSM1479504           NK          F</span></span>
<span><span class="co">#&gt; GSM1479506  Neutrophils          F</span></span>
<span><span class="co">#&gt; GSM1479507    Monocytes          F</span></span>
<span><span class="co">#&gt; GSM1479508      B-cells          F</span></span>
<span><span class="co">#&gt; GSM1479509          CD4          F</span></span>
<span><span class="co">#&gt; GSM1479510          CD8          F</span></span>
<span><span class="co">#&gt; GSM1479511           NK          F</span></span>
<span><span class="co">#&gt; GSM1479520  Neutrophils          F</span></span>
<span><span class="co">#&gt; GSM1479521    Monocytes          F</span></span>
<span><span class="co">#&gt; GSM1479522      B-cells          F</span></span>
<span><span class="co">#&gt; GSM1479523          CD4          F</span></span>
<span><span class="co">#&gt; GSM1479524          CD8          F</span></span>
<span><span class="co">#&gt; GSM1479525           NK          F</span></span></code></pre></div>
<div class="section level4">
<h4 id="process-data">process data<a class="anchor" aria-label="anchor" href="#process-data"></a>
</h4>
<p>To find the passing genes, low expression genes filtration needs to
be done first. <code><a href="../reference/process_data.html">process_data()</a></code> can help to do this, when
<code>normalize = TRUE</code>, further normalization and voom fit will
be done after filtration. The data will be returned with filtered counts
(unfiltered counts data will be saved under “original”), vfit after
<code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code> and tfit after <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">limma::lmFit()</a></code>
and <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code>.</p>
<p><code><a href="../reference/get_degs.html">get_degs()</a></code> function can filter low expression genes out
from immune dataset, normalize and fit voom model to data if
<code>normalize = TRUE</code>, then a processed data
<code>proc_data</code> object and a differentially expressed
GeneSetCollection (‘UP’ and ‘DOWN’) will be returned as output.</p>
<ul>
<li>
<code>data</code> param can be a bulk RNA-seq expression object with
subset labels, can be matrix, eSet, DGEList, …<br>
</li>
<li>
<code>group_col</code> param can specify the column name of groups
(labels), can be a vector when <code>data</code> is a matrix.<br>
</li>
<li>
<code>target_group</code> param can specify the cell target_group
name of interest. Users can choose one name in <code>group_col</code>
vector.<br>
</li>
<li>
<code>normalize</code> param indicates if the input data is raw
counts data. If <code>normalize = FALSE</code>, the function would
directly use the expression data <em>per se</em> to filter out genes
instead of calculating CPM.<br>
</li>
<li>
<code>feature_selection</code> param to choose whether to use rank
product (‘rankproduct’) or ‘none’ to select DEGs from multiple
comparisons of DE analysis, default ‘auto’ uses ‘rankproduct’ but change
to ‘none’ if final genes &lt; 5 for both UP and DOWN.<br>
</li>
<li>
<code>plot = TRUE</code> would visualize the QC results before and
after filtration. It would make log hist plot, RLE plot, MDS plot and SA
plot.<br>
</li>
<li>
<code>markers</code> param could specify a geneset to be kept no
matter if they pass the filtration. Setting it to <code>NULL</code> can
skip this.<br>
</li>
<li>
<code>gene_id</code> specify the gene ID type of rownames of data
when markers is not NULL, could be one of ‘ENSEMBL’, ‘SYMBOL’,
‘ENTREZ’…, default ‘SYMBOL’.</li>
</ul>
<p>Let’s see what cell types are in <code>im_data_6</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">im_data_6</span><span class="op">$</span><span class="va">`celltype:ch1`</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Neutrophils" "Monocytes"   "B-cells"     "CD4"         "CD8"        </span></span>
<span><span class="co">#&gt; [6] "NK"</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proc_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/process_data.html">process_data</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,  </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>, <span class="co">## rownames of im_data_6 is ENSEMBL ID</span></span>
<span>  summary <span class="op">=</span> <span class="cn">TRUE</span> <span class="co">## if to show the summary of tfit result</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; No group or design set. Assuming all samples belong to one group.</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             2554         2256       1740   1581   1187</span></span>
<span><span class="co">#&gt; NotSig            768         1675       2701   2923   3684</span></span>
<span><span class="co">#&gt; Up               2896         2287       1777   1714   1347</span></span>
<span><span class="co"># summary(limma::decideTests(proc_data$tfit))</span></span></code></pre></div>
<p>mastR also provides visualizations for data process, to compare the
data before and after <code><a href="../reference/process_data.html">process_data()</a></code> and see how well the
QC is done.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr1</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">$</span><span class="va">original_counts</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">expr2</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span></span>
<span><span class="co">## plot expression distribution, RLE and MDS</span></span>
<span><span class="fu"><a href="../reference/plot_diagnostics.html">plot_diagnostics</a></span><span class="op">(</span></span>
<span>  <span class="va">expr1</span>,</span>
<span>  <span class="va">expr2</span>,</span>
<span>  group_col <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-1.png" width="768"><img src="mastR_Demo_files/figure-html/plot%20diagnostics-2.png" width="768"><img src="mastR_Demo_files/figure-html/plot%20diagnostics-3.png" width="768"></p>
<pre><code><span><span class="co">#&gt; NULL</span></span>
<span><span class="co">## plot mean-variance trend</span></span>
<span><span class="fu"><a href="../reference/plot_mean_var.html">plot_mean_var</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">)</span></span></code></pre>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-4.png" width="768"></p>
<pre><code><span><span class="co">#&gt; NULL</span></span>
<span><span class="co">## plot PCA</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">expr1</span>, group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">expr2</span>, group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span><span class="op">)</span></span></code></pre>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-5.png" width="768"></p>
</div>
<div class="section level4">
<h4 id="select-degs">select DEGs<a class="anchor" aria-label="anchor" href="#select-degs"></a>
</h4>
<p>Then proc_data can be passed into <code><a href="../reference/select_sig.html">select_sig()</a></code> function
for further genes selection based on rank product scoring. Cell type
specific signature can be obtained by this.</p>
<ul>
<li>
<code>feature_selection</code> param to choose whether to use rank
product (‘rankproduct’) or ‘none’ to select DEGs from multiple
comparisons of DE analysis. Default “auto” will perform “rankproduct”
and switch to “none” if the number of final “UP” and “DOWN” genes are
&lt; 5.</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## get the same result as there's permutation test for rank product</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_sig.html">select_sig</a></span><span class="op">(</span>tfit <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">tfit</span><span class="op">)</span></span></code></pre></div>
<p>All above steps for screening subset signature can be done by one
integrated function <code><a href="../reference/get_degs.html">get_degs()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG_6</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_degs.html">get_degs</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,  </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  plot <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co">## show QC plots after filtration</span></span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>  <span class="co">## convert ensembl IDs of rownames into symbols if gene_id != "SYMBOL"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## `proc_data` object would be saved</span></span>
<span><span class="va">proc_data</span> <span class="op">&lt;-</span> <span class="va">DEG_6</span><span class="op">$</span><span class="va">proc_data</span></span>
<span></span>
<span><span class="va">DEG_6</span><span class="op">$</span><span class="va">DEGs</span></span>
<span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">DEG_6</span><span class="op">$</span><span class="va">DEGs</span><span class="op">)</span></span>
<span><span class="va">DEG_6</span><span class="op">$</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">design</span> <span class="co">## show the design matrix</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="intersect-with-markers-pool">intersect with markers pool<a class="anchor" aria-label="anchor" href="#intersect-with-markers-pool"></a>
</h4>
<p>Then we will only keep those UP regulated genes which are also in our
markers pool <code>Markers</code>.</p>
<p>We can see PCA shows clearer separation between NK cells and other
cells by using screened UP DEGs. And intersected genes explain more
variance in PC1 compared with non-intersected DEGs.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## convert ensembl IDs into symbols to match markers pool</span></span>
<span><span class="va">deg_up</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">sig_ct</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                 <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">## markers specific for NK cells</span></span>
<span><span class="va">m_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>, <span class="va">deg_up</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">deg_up</span><span class="op">)</span><span class="op">]</span>  <span class="co">## set ensembl ID as names for downstream visualization</span></span>
<span></span>
<span><span class="va">m_ct</span></span>
<span><span class="co">#&gt; ENSG00000117281 ENSG00000122223 ENSG00000198821 ENSG00000172543 ENSG00000145649 </span></span>
<span><span class="co">#&gt;         "CD160"         "CD244"         "CD247"          "CTSW"          "GZMA" </span></span>
<span><span class="co">#&gt; ENSG00000197540 ENSG00000081985 ENSG00000100385 ENSG00000183542 ENSG00000150045 </span></span>
<span><span class="co">#&gt;          "GZMM"       "IL12RB2"         "IL2RB"         "KLRC4"         "KLRF1" </span></span>
<span><span class="co">#&gt; ENSG00000168060 ENSG00000149294 ENSG00000189430 ENSG00000168229 ENSG00000041353 </span></span>
<span><span class="co">#&gt;      "NAALADL1"         "NCAM1"          "NCR1"         "PTGDR"        "RAB27B" </span></span>
<span><span class="co">#&gt; ENSG00000164483 ENSG00000198574 ENSG00000074966 ENSG00000115085 ENSG00000134545 </span></span>
<span><span class="co">#&gt;         "SAMD3"        "SH2D1B"           "TXK"         "ZAP70"         "KLRC1" </span></span>
<span><span class="co">#&gt; ENSG00000150637 ENSG00000135077 ENSG00000027869 ENSG00000171476 </span></span>
<span><span class="co">#&gt;         "CD226"        "HAVCR2"        "SH2D2A"          "HOPX"</span></span>
<span></span>
<span><span class="co">## PCA shows clear separation of NK cells</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>                group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>                  group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/markers%20for%20cell%20type-1.png" width="768"></p>
<p>There’s a wrapper function <code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> which can
directly do the whole in one step. The result would be the same as
running the workflow step by step</p>
<p><code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> can also be helpful when you have
multiple datasets. It can output final signature after aggregating
signature lists from all datasets by using “RRA”.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run by wrapped function in one-step</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="step-3--specify-markers-against-tissue">Step 3. Specify Markers Against Tissue<a class="anchor" aria-label="anchor" href="#step-3--specify-markers-against-tissue"></a>
</h3>
<p>The example uses <code>CCLE_tpm</code> from <a href="https://depmap.org/portal/download/" class="external-link">DepMap CCLE</a>, containing
RSEM quantified TPM data of 1392 cancer cell lines (until 2022.04).</p>
<p>To specify NK markers for CRC, use <code><a href="../reference/filter_non_tissue.html">filter_non_tissue()</a></code>
function to remove failing genes which are also highly expressed by CRC
adherent cell lines <em>per se</em>.</p>
<ul>
<li>
<code>data</code> can be either “CCLE” or expression dataset,
e.g. matrix, eSet, DGEList, Seurat and so on.<br>
</li>
<li>
<code>group_col</code> param is the group vector of input samples,
or the name of group column.<br>
</li>
<li>
<code>target_group</code> param is an expression pattern to specify
the tissue of interest, here we choose ‘colorectal’ to select all
colorectal adherent cell lines.<br>
</li>
<li>
<code>ignore.case</code> means whether to ignore the capital or
not.<br>
</li>
<li>
<code>log = TURE</code> would do log transformation to the
data.<br>
</li>
<li>
<code>q</code> param is the threshold to remove failing genes. 0.25
means removing markers with median expression &gt;= 25% percentile of
non-zero expression of all genes.<br>
</li>
<li>
<code>markers</code> param is the updated markers pool.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CCLE</span> <span class="op">&lt;-</span> <span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/depmap/man/TPM.html" class="external-link">depmap_TPM</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">CCLE_meta</span> <span class="op">&lt;-</span> <span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/depmap/man/metadata.html" class="external-link">depmap_metadata</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_non_tissue.html">filter_non_tissue</a></span><span class="op">(</span>data <span class="op">=</span> <span class="st">"CCLE"</span>,</span>
<span>                           group_col <span class="op">=</span> <span class="st">"primary_disease"</span>,</span>
<span>                           target_group <span class="op">=</span> <span class="st">"colorectal"</span>,</span>
<span>                           markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>                           ccle_tpm <span class="op">=</span> <span class="va">CCLE</span>,</span>
<span>                           ccle_meta <span class="op">=</span> <span class="va">CCLE_meta</span><span class="op">)</span></span>
<span><span class="co">## or without loading CCLE yourself</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_non_tissue.html">filter_non_tissue</a></span><span class="op">(</span>group_col <span class="op">=</span> <span class="st">"primary_disease"</span>,</span>
<span>                           target_group <span class="op">=</span> <span class="st">"colorectal"</span>,</span>
<span>                           markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>For fast test, here we only selected 5 CRC cell line samples as a
small test dataset <code>ccle_crc_5</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ccle_crc_5"</span><span class="op">)</span></span>
<span><span class="va">ccle_crc_5</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span>
<span><span class="co">#&gt; An object of class "DGEList"</span></span>
<span><span class="co">#&gt; $counts</span></span>
<span><span class="co">#&gt;          SNU283_LARGE_INTESTINE TGBC18TKB_LARGE_INTESTINE SW837_LARGE_INTESTINE</span></span>
<span><span class="co">#&gt; TSPAN6               4.91408610                  6.045923             5.2570106</span></span>
<span><span class="co">#&gt; TNMD                 0.17632277                  0.000000             0.3334237</span></span>
<span><span class="co">#&gt; DPM1                 6.94684781                  6.724105             6.9242187</span></span>
<span><span class="co">#&gt; SCYL3                2.57773093                  2.260026             2.9653225</span></span>
<span><span class="co">#&gt; C1orf112             3.85399565                  3.835924             4.7618171</span></span>
<span><span class="co">#&gt; FGR                  0.00000000                  0.000000             0.1890338</span></span>
<span><span class="co">#&gt; CFH                  0.08406426                  4.115200             0.2509616</span></span>
<span><span class="co">#&gt; FUCA2                4.85549144                  6.263034             6.9090530</span></span>
<span><span class="co">#&gt; GCLC                 4.93404465                  4.398487             5.8750428</span></span>
<span><span class="co">#&gt; NFYA                 3.68144927                  3.727920             4.0703893</span></span>
<span><span class="co">#&gt;          SNU1040_LARGE_INTESTINE HT29_LARGE_INTESTINE</span></span>
<span><span class="co">#&gt; TSPAN6                5.16188768           4.39985467</span></span>
<span><span class="co">#&gt; TNMD                  0.46466827           0.00000000</span></span>
<span><span class="co">#&gt; DPM1                  6.67849451           6.71836163</span></span>
<span><span class="co">#&gt; SCYL3                 2.48284828           2.78450398</span></span>
<span><span class="co">#&gt; C1orf112              3.32624970           3.46858332</span></span>
<span><span class="co">#&gt; FGR                   1.48542683           0.01435529</span></span>
<span><span class="co">#&gt; CFH                   0.05658353           4.08151007</span></span>
<span><span class="co">#&gt; FUCA2                 5.96809075           5.35508676</span></span>
<span><span class="co">#&gt; GCLC                  5.15583017           5.10097765</span></span>
<span><span class="co">#&gt; NFYA                  4.81249823           4.32481060</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $samples</span></span>
<span><span class="co">#&gt;                           group lib.size norm.factors cancer</span></span>
<span><span class="co">#&gt; SNU283_LARGE_INTESTINE        1 49889.02            1    CRC</span></span>
<span><span class="co">#&gt; TGBC18TKB_LARGE_INTESTINE     1 49489.64            1    CRC</span></span>
<span><span class="co">#&gt; SW837_LARGE_INTESTINE         1 56426.42            1    CRC</span></span>
<span><span class="co">#&gt; SNU1040_LARGE_INTESTINE       1 54366.12            1    CRC</span></span>
<span><span class="co">#&gt; HT29_LARGE_INTESTINE          1 49232.06            1    CRC</span></span>
<span></span>
<span><span class="co">## markers against for CRC cell lines</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_non_tissue.html">filter_non_tissue</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ccle_crc_5</span>,</span>
<span>                           group_col <span class="op">=</span> <span class="st">"cancer"</span>,</span>
<span>                           target_group <span class="op">=</span> <span class="st">"CRC"</span>,</span>
<span>                           markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## as CCLE is default setting, this can also be easily run without loading CCLE</span></span>
<span><span class="co"># m_ccl &lt;- filter_non_tissue(target_group = "colorectal",</span></span>
<span><span class="co">#                            markers = geneIds(Markers))</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "KIR2DL2"  "KIR2DL5A" "KIR2DS2"  "KIR2DS4"  "KIR2DS5"  "KLRA1P"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-4--final-nk-cells-signature-for-crc">Step 4. Final NK Cells Signature for CRC<a class="anchor" aria-label="anchor" href="#step-4--final-nk-cells-signature-for-crc"></a>
</h3>
<p>Select the Signature which pass both criteria: cell type and cancer
cell line.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_NK_CRC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="va">sig_NK_CRC</span></span>
<span><span class="co">#&gt;  [1] "CD160"    "CD244"    "CD247"    "CTSW"     "GZMA"     "GZMM"    </span></span>
<span><span class="co">#&gt;  [7] "IL12RB2"  "IL2RB"    "KLRC4"    "KLRF1"    "NAALADL1" "NCAM1"   </span></span>
<span><span class="co">#&gt; [13] "NCR1"     "PTGDR"    "SAMD3"    "SH2D1B"   "ZAP70"    "KLRC1"   </span></span>
<span><span class="co">#&gt; [19] "CD226"    "HAVCR2"   "HOPX"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="step-5--visulize-the-curation-effect">Step 5. Visulize The
Curation Effect<a class="anchor" aria-label="anchor" href="#step-5--visulize-the-curation-effect"></a>
</h2>
<hr>
<div class="section level3">
<h3 id="pca-matrix-plot">PCA matrix plot<a class="anchor" aria-label="anchor" href="#pca-matrix-plot"></a>
</h3>
<p>To see if curated NK cells signatures have more distinguishable
expression pattern from other immune subsets, we can use PCA plot again
here to show it.</p>
<p>Loading can be drawn to show which genes are relevant to NK cells,
and result shows most of signature genes are indicating NK cells.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## as proc_data shows better discrimination, use it for visualization</span></span>
<span><span class="va">proc_data</span><span class="op">$</span><span class="va">voomE</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  features <span class="op">=</span> <span class="va">sig_NK_CRC</span>, </span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/pca-1.png" width="768"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## loadings can be shown by setting loading = TRUE</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  features <span class="op">=</span> <span class="va">sig_NK_CRC</span>, </span>
<span>  loading <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 3 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 4 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span>
<span><span class="co">#&gt; ggrepel: 4 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/pca-2.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p><code>result_heatmap()</code> function can plot the original NK cells
markers and curated signatures expression pattern heatmaps in immune
datasets.</p>
<ul>
<li>
<code>sigs</code> param is the final curated signature symbols,
while <code>markers</code> param is the original markers pool.<br>
</li>
<li>
<code>scale</code> could specify the <code>column</code> or
<code>row</code> to be scaled or no scaling by <code>none</code>.<br>
</li>
<li>
<code>min_max</code> is used to decide if to apply min-max
normalization to each row of expression data.</li>
<li>
<code>normalize</code> means whether to normalize and calculate
logCPM or not.<br>
</li>
<li>Users are able to specify the color of heatmap by setting
<code>color</code> param.</li>
</ul>
<p>We can see the expression pattern of cutrated signature is more
distinguishable in NK cells.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_heatmap.html">sig_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/heatmap-1.png" width="960"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## for multiple datasets</span></span>
<span><span class="co"># sig_heatmap(</span></span>
<span><span class="co">#   data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#               proc_data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"), </span></span>
<span><span class="co">#   markers = geneIds(Markers),</span></span>
<span><span class="co">#   normalize = TRUE,</span></span>
<span><span class="co">#   gene_id = "ENSEMBL",</span></span>
<span><span class="co">#   # col = colorRampPalette(c(rgb(1, 1, 0), "red"))(256)  ## can customize heatmap color</span></span>
<span><span class="co"># ) + patchwork::plot_layout(guides = "collect", ncol = 1)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="rank-density-plot">Rank Density Plot<a class="anchor" aria-label="anchor" href="#rank-density-plot"></a>
</h3>
<p>Use <code><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot()</a></code> function to make rank density
plot for all subsets in the data.</p>
<p>By setting <code>aggregate = TRUE</code>, only one plot for each cell
type with mean expression would be shown.</p>
<p>We might want to see the rank density plot in more than 1 datasets
for comparison, this can be done by simply passing a list of datasets to
function.</p>
<p>We can find, processed data shows more digstinguishable rankdensity
distribution of screened NK signature genes.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ## visualization on both original and processed data</span></span>
<span><span class="co"># sig_rankdensity_plot(</span></span>
<span><span class="co">#   data = list(im_data_6 = im_data_6,</span></span>
<span><span class="co">#               process_Data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   normalize = c(TRUE, FALSE),</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"), </span></span>
<span><span class="co">#   gene_id = "ENSEMBL",</span></span>
<span><span class="co">#   slot = "voomE"</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co">## visualization on aggregated data</span></span>
<span><span class="fu"><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  aggregate <span class="op">=</span>  <span class="cn">TRUE</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/rankdensity%20plot-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="nk-score-boxplot">NK Score Boxplot<a class="anchor" aria-label="anchor" href="#nk-score-boxplot"></a>
</h3>
<p>To see if the curated NK cells signatures have evident higher
abundance in NK cells than other immune subsets, use
<code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code> function to make boxplot of the NK scores for
all subsets in the data.</p>
<p>NK score is calculated by <code>singscore</code> package.</p>
<p>Obviously, our signature always shows significantly higher score in
NK cells.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  normalize <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/score%20boxplot-1.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## To make boxplots for more than 1 immune datasets at once:</span></span>
<span><span class="co"># result_score_boxplot(data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#                                  proc_data = proc_data), </span></span>
<span><span class="co">#                      sigs = sig_NK_CRC,</span></span>
<span><span class="co">#                      group_col = c("celltype:ch1", "celltype.ch1"),</span></span>
<span><span class="co">#                      target_group = "NK",</span></span>
<span><span class="co">#                      normalize = TRUE,</span></span>
<span><span class="co">#                      gene_id = "ENSEMBL") *</span></span>
<span><span class="co">#   guides(col = "none")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="nk-signature-abundance-boxplot">NK Signature Abundance Boxplot<a class="anchor" aria-label="anchor" href="#nk-signature-abundance-boxplot"></a>
</h3>
<p><code>result_exp_boxplot()</code> function can also generate boxplot
of the NK signature abundance for all subsets in the data by setting
<code>plot.score = FALSE</code>.</p>
<p>It’s clear the score can increase the difference between NK cells and
others.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  plot.score <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20boxplot-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="nk-signature-abundance-scatter-plot">NK Signature Abundance Scatter Plot<a class="anchor" aria-label="anchor" href="#nk-signature-abundance-scatter-plot"></a>
</h3>
<p>To see if the curated NK cells signature shows higher specificity to
NK cells than other immune subsets. Use <code><a href="../reference/sig_scatter_plot.html">sig_scatter_plot()</a></code>
function to make scatter plot of the NK signature abundance for all
subsets in the data.</p>
<p>Based on sactter plot result, we can actually do further filtration
to improve signature specificity.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  xint <span class="op">=</span> <span class="fl">4</span>, yint <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot-1.png" width="768"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## To make scatter plot for more than 1 immune datasets at once:</span></span>
<span><span class="co"># sig_scatter_plot(</span></span>
<span><span class="co">#   data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#               proc_data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"),</span></span>
<span><span class="co">#   target_group = "NK", </span></span>
<span><span class="co">#   xint = 4, yint = 4,</span></span>
<span><span class="co">#   normalize = TRUE, </span></span>
<span><span class="co">#   gene_id = "ENSEMBL")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="nk-signature-gsea-plot">NK Signature GSEA plot<a class="anchor" aria-label="anchor" href="#nk-signature-gsea-plot"></a>
</h3>
<p>To see if the curated NK cells signature is enriched in NK cells than
other immune subsets. Use <code><a href="../reference/sig_gseaplot.html">sig_gseaplot()</a></code> function to make
GSEAplot.</p>
<p>It’s clear that our screened signature is enriched in all
comparisons.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ## gseaplot</span></span>
<span><span class="co"># sig_gseaplot(</span></span>
<span><span class="co">#   data = proc_data, </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = "celltype.ch1", </span></span>
<span><span class="co">#   target_group = "NK",</span></span>
<span><span class="co">#   gene_id = "ENSEMBL",</span></span>
<span><span class="co">#   slot = "voomE"</span></span>
<span><span class="co"># )</span></span>
<span><span class="co">## dotplot</span></span>
<span><span class="fu"><a href="../reference/sig_gseaplot.html">sig_gseaplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"dotplot"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; No group or design set. Assuming all samples belong to one group.</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             2526         2221       1704   1539   1160</span></span>
<span><span class="co">#&gt; NotSig            735         1651       2677   2891   3630</span></span>
<span><span class="co">#&gt; Up               2815         2204       1695   1646   1286</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gseaplot-1.png" width="768"></p>
</div>
</div>
<div class="section level2">
<h2 id="refine-signature-specificity">Refine Signature Specificity<a class="anchor" aria-label="anchor" href="#refine-signature-specificity"></a>
</h2>
<hr>
<p>According to the visualization result, we can see our NK signature
shows good performance in separating NK cells with all other cells. But
CD8+ T cells still show similar abundance to NK cells in some signature
genes expression, which might impair the ability of signature to
distinguish NK cells from CD8+ T cells.</p>
<p>But we know those genes with similar abundance across NK and CD8+ T
are importance and distinguishable for separating NK cells from other
non-CD8+ T cells. So we don’t recommend to simply delete those genes
based on our abundance scatter plot.</p>
<p>Instead, here we implemented a better way to increase the difference
between the ambiguous cell types.</p>
<p>By setting <code>keep.top</code> and <code>keep.group</code>, we can
easily keep the top n DEGs of specified DE comparisons, DEGs are ranked
based on <code>Rank</code> param. For instance, here we want keep more
distinguishable genes between NK and CD8+ T cells, we can set
<code>keep.group = "CD8"</code> or <code>keep.group = "NK-CD8"</code>.
<code>keep.top</code> can specify how many top genes in that comparison
should be kept.</p>
<p>We only get 2 more new genes when keeping top 200 DEGs in ‘NK-CD8’,
that’s because majority of those genes are not in our markers pool
<code>Markers</code> and can’t pass the screening. So it would be better
to create a large enough pool for screening in case we miss any useful
DEGs.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## keep top 200 DEGs in 'NK-CD8'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_refine</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  keep.top <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  keep.group <span class="op">=</span> <span class="st">"CD8"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; No group or design set. Assuming all samples belong to one group.</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             2554         2256       1740   1581   1187</span></span>
<span><span class="co">#&gt; NotSig            768         1675       2701   2923   3684</span></span>
<span><span class="co">#&gt; Up               2896         2287       1777   1714   1347</span></span>
<span></span>
<span><span class="co">## get 8 more new signature genes</span></span>
<span><span class="va">sig_NK_CRC_refine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">m_ct_refine</span>, <span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">sig_NK_CRC_refine</span>, <span class="va">sig_NK_CRC</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "CDHR1"   "CLNK"    "FASLG"   "FCGR3A"  "IL18R1"  "IL18RAP" "KIR2DL1"</span></span>
<span><span class="co">#&gt;  [8] "KIR2DL4" "KIR2DS4" "KLRD1"   "MGAM"    "NME8"    "S1PR5"   "XCL2"   </span></span>
<span><span class="co">#&gt; [15] "ITGAM"   "ITGAX"   "SIGLEC7"</span></span>
<span></span>
<span><span class="co">## most NK-CD8 top DEGs are not in the markers pool</span></span>
<span><span class="va">deg_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_de_table.html">get_de_table</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>, </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; No group or design set. Assuming all samples belong to one group.</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             2530         2201       1673   1494   1099</span></span>
<span><span class="co">#&gt; NotSig            737         1711       2738   2978   3727</span></span>
<span><span class="co">#&gt; Up               2805         2160       1661   1600   1246</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu">mapIds</span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">deg_tables</span><span class="op">$</span><span class="va">`NK-CD8`</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span>,</span>
<span>                 <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>,</span>
<span>          <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span>  <span class="co">## only 19 out of top 200 DEGs are in the pool</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; [1] "LTB"</span></span></code></pre></div>
<div class="section level3">
<h3 id="visualization-after-refinement">Visualization after refinement<a class="anchor" aria-label="anchor" href="#visualization-after-refinement"></a>
</h3>
<p>We can see the difference of NK vs CD4 and CD8 T cells increased, but
the difference vs monocytes and neutrophils dropped a bit.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC_refine</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature after Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature before Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span></span>
<span><span class="op">(</span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span><span class="op">)</span><span class="op">*</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>,<span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/score%20boxplot%20refine-1.png" width="768"></p>
<p>Abundance scatter plot also shows more specific genes to NK compared
with CD4/8 T cells.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC_refine</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  xint <span class="op">=</span> <span class="fl">4</span>, yint <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature after Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  xint <span class="op">=</span> <span class="fl">4</span>, yint <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature before Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span></span>
<span><span class="op">(</span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot%20refine-1.png" width="768"></p>
</div>
</div>
<div class="section level2">
<h2 id="screen-on-multiple-datasets">Screen on Multiple Datasets<a class="anchor" aria-label="anchor" href="#screen-on-multiple-datasets"></a>
</h2>
<hr>
<p>Our package can be applied on multiple datasets. Instead of
integrating datasets into one larger dataset to fit into one linear
model, we screen signature in each dataset respectively and aggregating
ranked lists via <code>RobustRankAggreg</code> method (as our output of
signature by screening function is ordered based on the given
<code>Rank</code>).</p>
<p>The advantage is that we can avoid over-normalizing or mis-correcting
when carrying out data integration. And we can get more robust and
conserved signature across datasets.</p>
<p>Of course, easy way such as “union” and “intersect” is also supported
besides “RRA” (Robust Rank Aggreg), by setting <code>comb = union</code>
or <code>comb = intersect</code>.</p>
<p>It’s better to use union when you get only a few genes screened for
each dataset, while “RRA” is better to pick up significant genes from
large DEG lists, “intersect” is proper for highly overlapping gene
lists.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## As a test, we just repeatly use im_data_6 as a show case</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## we will get exactly the same list</span></span>
<span><span class="co">## if we choose 'union' or 'intersect' as combination</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setequal</a></span><span class="op">(</span><span class="va">m_ct_m</span>, <span class="va">m_ct</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## but we will only get the genes appear at top rank across gene lists</span></span>
<span><span class="co">## if we choose 'RRA', s_thres is to determine the threshold for ranking score</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  comb <span class="op">=</span> <span class="st">"RRA"</span>,  <span class="co">## change this to use different strategy, default is "union"</span></span>
<span>  s_thres <span class="op">=</span> <span class="fl">0.5</span>  <span class="co">## only work when comb = "RRA", set a threshold for ranking score</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## we can get only 8 signature genes this time</span></span>
<span><span class="va">m_ct_m</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="screen-on-scrna-data">Screen on scRNA data<a class="anchor" aria-label="anchor" href="#screen-on-scrna-data"></a>
</h2>
<hr>
<p>We also provide pseudo-sample functions in this package to help users
convert scRNA data into pseudo-bulk data, which can then be used in our
above workflow.</p>
<p><code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> can help aggregate cells into
pseudo-sample according to the given factor.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a test scRNA object of 100 genes x 100 cells</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>,</span>
<span>  level <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, each <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">counts</span>, by <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pb</span>, group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span><span class="va">pb</span>, group_col <span class="op">=</span> <span class="st">"group"</span>, target_group <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Seurat or SCE object are also accepted</span></span>
<span><span class="co"># scRNA &lt;- Seurat::CreateSeuratObject(counts = counts, meta.data = meta)</span></span>
<span><span class="co"># pseudo_samples(scRNA, by = c("subset","level"))</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="application-on-simulated-data">Application on Simulated
data<a class="anchor" aria-label="anchor" href="#application-on-simulated-data"></a>
</h2>
<p>Here we will simulate a scRNA data using <code>splatter</code>. Then
randomly pseudo-bulking it to generate pseudo bulk data for scoring and
deconvolution.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">splatter</span><span class="op">)</span></span>
<span><span class="co">## set seed for reproduce as there's permutation inside</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim_params</span> <span class="op">&lt;-</span> <span class="fu">newSplatParams</span><span class="op">(</span></span>
<span>  nGenes <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  batchCells <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  group.prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.4</span>, length.out <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>  de.prob <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  <span class="co"># de.downProb = 0,  ## only set up-regulated genes for each group</span></span>
<span>  de.facLoc <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  de.facScale <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">splatSimulate</span><span class="op">(</span><span class="va">sim_params</span>, method <span class="op">=</span> <span class="st">"groups"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">markers_list</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"DEFacGroup"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## aggregate into pseudo-bulk samples</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>,</span>
<span>                     by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"Group"</span><span class="op">)</span>,</span>
<span>                     min.cells <span class="op">=</span> <span class="fl">50</span>, max.cells <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">dge</span>,</span>
<span>    markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    group_col <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>    target_group <span class="op">=</span> <span class="va">x</span>,</span>
<span>    lfc <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    assemble <span class="op">=</span> <span class="st">"intersect"</span>  <span class="co">## get intersected DEGs across all comparisons</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, \<span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu">ggvenn</span><span class="fu">::</span><span class="fu">ggvenn</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">sig_ls</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                                          marker <span class="op">=</span> <span class="va">markers_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                     show_percentage <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## randomly generate aggregating idx</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">rand_idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## aggregate into pseudo-bulk samples based on rand_idx</span></span>
<span><span class="va">pb_r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb_r</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## append cellular composition</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">@</span><span class="va">colData</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">rand_idx</span>, <span class="va">Group</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>count <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">Group</span>, values_from <span class="op">=</span> <span class="va">count</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>rand_idx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rand_idx</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span>, <span class="va">tmp</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span> <span class="op">=</span> <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## data process</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">filterByExpr</span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="va">dge_r</span><span class="op">[</span><span class="va">keep</span>,, keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">calcNormFactors</span><span class="op">(</span><span class="va">dge_r</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="score">Score<a class="anchor" aria-label="anchor" href="#score"></a>
</h3>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rank_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/rankGenes.html" class="external-link">rankGenes</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge_r</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span>, <span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>,<span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Prop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prop</span>, y <span class="op">=</span> <span class="va">Score</span>, col <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="deconvolution">Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h3>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_matrix</span> <span class="op">&lt;-</span> <span class="fu">DWLS</span><span class="fu">::</span><span class="fu">buildSignatureMatrixMAST</span><span class="op">(</span></span>
<span>  scdata <span class="op">=</span> <span class="va">data_sim</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>  id <span class="op">=</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span>,</span>
<span>  path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotation">Annotation<a class="anchor" aria-label="anchor" href="#annotation"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GSEABase_1.60.0      graph_1.76.0         annotate_1.76.0     </span></span>
<span><span class="co">#&gt;  [4] XML_3.99-0.13        AnnotationDbi_1.60.0 IRanges_2.32.0      </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.36.2     Biobase_2.58.0       BiocGenerics_0.44.0 </span></span>
<span><span class="co">#&gt; [10] ggplot2_3.4.1        mastR_0.9.0         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] scattermore_0.8             SeuratObject_4.1.3         </span></span>
<span><span class="co">#&gt;   [3] ragg_1.2.5                  tidyr_1.3.0                </span></span>
<span><span class="co">#&gt;   [5] bit64_4.0.5                 knitr_1.42                 </span></span>
<span><span class="co">#&gt;   [7] irlba_2.3.5.1               DelayedArray_0.24.0        </span></span>
<span><span class="co">#&gt;   [9] data.table_1.14.8           KEGGREST_1.38.0            </span></span>
<span><span class="co">#&gt;  [11] RCurl_1.98-1.10             generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [13] cowplot_1.1.1               RSQLite_2.3.0              </span></span>
<span><span class="co">#&gt;  [15] shadowtext_0.1.2            RANN_2.6.1                 </span></span>
<span><span class="co">#&gt;  [17] future_1.32.0               bit_4.0.5                  </span></span>
<span><span class="co">#&gt;  [19] enrichplot_1.18.3           spatstat.data_3.0-0        </span></span>
<span><span class="co">#&gt;  [21] xml2_1.3.3                  httpuv_1.6.9               </span></span>
<span><span class="co">#&gt;  [23] SummarizedExperiment_1.28.0 viridis_0.6.2              </span></span>
<span><span class="co">#&gt;  [25] xfun_0.37                   jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [27] evaluate_0.20               promises_1.2.0.1           </span></span>
<span><span class="co">#&gt;  [29] fansi_1.0.4                 igraph_1.4.1               </span></span>
<span><span class="co">#&gt;  [31] DBI_1.1.3                   htmlwidgets_1.6.1          </span></span>
<span><span class="co">#&gt;  [33] spatstat.geom_3.0-6         purrr_1.0.1                </span></span>
<span><span class="co">#&gt;  [35] ellipsis_0.3.2              selectr_0.4-2              </span></span>
<span><span class="co">#&gt;  [37] dplyr_1.1.0                 ggpubr_0.6.0               </span></span>
<span><span class="co">#&gt;  [39] backports_1.4.1             deldir_1.0-6               </span></span>
<span><span class="co">#&gt;  [41] MatrixGenerics_1.10.0       vctrs_0.5.2                </span></span>
<span><span class="co">#&gt;  [43] SingleCellExperiment_1.20.0 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [45] abind_1.4-5                 cachem_1.0.7               </span></span>
<span><span class="co">#&gt;  [47] withr_2.5.0                 ggforce_0.4.1              </span></span>
<span><span class="co">#&gt;  [49] HDO.db_0.99.1               progressr_0.13.0           </span></span>
<span><span class="co">#&gt;  [51] sctransform_0.3.5           treeio_1.22.0              </span></span>
<span><span class="co">#&gt;  [53] goftest_1.2-3               cluster_2.1.4              </span></span>
<span><span class="co">#&gt;  [55] DOSE_3.24.2                 ape_5.7                    </span></span>
<span><span class="co">#&gt;  [57] lazyeval_0.2.2              crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [59] spatstat.explore_3.0-6      edgeR_3.40.2               </span></span>
<span><span class="co">#&gt;  [61] pkgconfig_2.0.3             labeling_0.4.2             </span></span>
<span><span class="co">#&gt;  [63] tweenr_2.0.2                GenomeInfoDb_1.34.9        </span></span>
<span><span class="co">#&gt;  [65] nlme_3.1-162                rlang_1.0.6                </span></span>
<span><span class="co">#&gt;  [67] globals_0.16.2              lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt;  [69] miniUI_0.1.1.1              downloader_0.4             </span></span>
<span><span class="co">#&gt;  [71] rprojroot_2.0.3             polyclip_1.10-4            </span></span>
<span><span class="co">#&gt;  [73] matrixStats_0.63.0          lmtest_0.9-40              </span></span>
<span><span class="co">#&gt;  [75] Matrix_1.5-3                aplot_0.1.9                </span></span>
<span><span class="co">#&gt;  [77] carData_3.0-5               singscore_1.18.0           </span></span>
<span><span class="co">#&gt;  [79] zoo_1.8-11                  ggridges_0.5.4             </span></span>
<span><span class="co">#&gt;  [81] pheatmap_1.0.12             png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [83] viridisLite_0.4.1           bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [85] msigdb_1.6.0                gson_0.1.0                 </span></span>
<span><span class="co">#&gt;  [87] KernSmooth_2.23-20          Biostrings_2.66.0          </span></span>
<span><span class="co">#&gt;  [89] blob_1.2.3                  stringr_1.5.0              </span></span>
<span><span class="co">#&gt;  [91] qvalue_2.30.0               parallelly_1.34.0          </span></span>
<span><span class="co">#&gt;  [93] spatstat.random_3.1-3       rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [95] gridGraphics_0.5-1          ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt;  [97] scales_1.2.1                memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [99] magrittr_2.0.3              plyr_1.8.8                 </span></span>
<span><span class="co">#&gt; [101] ica_1.0-3                   zlibbioc_1.44.0            </span></span>
<span><span class="co">#&gt; [103] scatterpie_0.1.8            compiler_4.2.2             </span></span>
<span><span class="co">#&gt; [105] RColorBrewer_1.1-3          fitdistrplus_1.1-8         </span></span>
<span><span class="co">#&gt; [107] cli_3.6.0                   XVector_0.38.0             </span></span>
<span><span class="co">#&gt; [109] listenv_0.9.0               patchwork_1.1.2            </span></span>
<span><span class="co">#&gt; [111] pbapply_1.7-0               MASS_7.3-58.3              </span></span>
<span><span class="co">#&gt; [113] tidyselect_1.2.0            stringi_1.7.12             </span></span>
<span><span class="co">#&gt; [115] textshaping_0.3.6           highr_0.10                 </span></span>
<span><span class="co">#&gt; [117] yaml_2.3.7                  GOSemSim_2.24.0            </span></span>
<span><span class="co">#&gt; [119] locfit_1.5-9.7              ggrepel_0.9.3              </span></span>
<span><span class="co">#&gt; [121] grid_4.2.2                  sass_0.4.5                 </span></span>
<span><span class="co">#&gt; [123] fastmatch_1.1-3             tools_4.2.2                </span></span>
<span><span class="co">#&gt; [125] future.apply_1.10.0         parallel_4.2.2             </span></span>
<span><span class="co">#&gt; [127] gridExtra_2.3               farver_2.1.1               </span></span>
<span><span class="co">#&gt; [129] Rtsne_0.16                  ggraph_2.1.0               </span></span>
<span><span class="co">#&gt; [131] digest_0.6.31               shiny_1.7.4                </span></span>
<span><span class="co">#&gt; [133] Rcpp_1.0.10                 GenomicRanges_1.50.2       </span></span>
<span><span class="co">#&gt; [135] car_3.1-1                   broom_1.0.3                </span></span>
<span><span class="co">#&gt; [137] later_1.3.0                 RcppAnnoy_0.0.20           </span></span>
<span><span class="co">#&gt; [139] org.Hs.eg.db_3.16.0         httr_1.4.5                 </span></span>
<span><span class="co">#&gt; [141] colorspace_2.1-0            rvest_1.0.3                </span></span>
<span><span class="co">#&gt; [143] fs_1.6.1                    tensor_1.5                 </span></span>
<span><span class="co">#&gt; [145] reticulate_1.28             splines_4.2.2              </span></span>
<span><span class="co">#&gt; [147] uwot_0.1.14                 yulab.utils_0.0.6          </span></span>
<span><span class="co">#&gt; [149] tidytree_0.4.2              spatstat.utils_3.0-1       </span></span>
<span><span class="co">#&gt; [151] pkgdown_2.0.7               graphlayouts_0.8.4         </span></span>
<span><span class="co">#&gt; [153] sp_1.6-0                    ggplotify_0.1.0            </span></span>
<span><span class="co">#&gt; [155] plotly_4.10.1               systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt; [157] xtable_1.8-4                jsonlite_1.8.4             </span></span>
<span><span class="co">#&gt; [159] ggtree_3.6.2                tidygraph_1.2.3            </span></span>
<span><span class="co">#&gt; [161] UpSetR_1.4.0                ggfun_0.0.9                </span></span>
<span><span class="co">#&gt; [163] R6_2.5.1                    pillar_1.8.1               </span></span>
<span><span class="co">#&gt; [165] htmltools_0.5.4             mime_0.12                  </span></span>
<span><span class="co">#&gt; [167] glue_1.6.2                  fastmap_1.1.1              </span></span>
<span><span class="co">#&gt; [169] clusterProfiler_4.6.2       BiocParallel_1.32.5        </span></span>
<span><span class="co">#&gt; [171] codetools_0.2-19            fgsea_1.24.0               </span></span>
<span><span class="co">#&gt; [173] utf8_1.2.3                  lattice_0.20-45            </span></span>
<span><span class="co">#&gt; [175] bslib_0.4.2                 spatstat.sparse_3.0-0      </span></span>
<span><span class="co">#&gt; [177] tibble_3.1.8                curl_5.0.0                 </span></span>
<span><span class="co">#&gt; [179] leiden_0.4.3                GO.db_3.16.0               </span></span>
<span><span class="co">#&gt; [181] survival_3.5-3              limma_3.54.2               </span></span>
<span><span class="co">#&gt; [183] rmarkdown_2.20              desc_1.4.2                 </span></span>
<span><span class="co">#&gt; [185] munsell_0.5.0               GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">#&gt; [187] reshape2_1.4.4              gtable_0.3.1               </span></span>
<span><span class="co">#&gt; [189] Seurat_4.3.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr>
<blockquote>
<p>Cursons J, Souza-Fonseca-Guimaraes F, Foroutan M, Anderson A,
Hollande F, Hediyeh-Zadeh S, Behren A, Huntington ND, Davis MJ. A Gene
Signature Predicting Natural Killer Cell Infiltration and Improved
Survival in Melanoma Patients. Cancer Immunol Res. 2019
Jul;7(7):1162-1174. doi: 10.1158/2326-6066.CIR-18-0500. Epub 2019 May
14. PMID: 31088844.</p>
</blockquote>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jinjin Chen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
