<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mastR: an R package for automatical signature screening • mastR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mastR: an R package for automatical signature screening">
<meta property="og:description" content="mastR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mastR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/mastR_Demo.html">mastR: an R package for automatical signature screening</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/mastR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mastR: an R package for automatical signature
screening</h1>
                        <h4 data-toc-skip class="author">Jinjin
Chen</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical
Biology, University of Melbourne, Parkville, VIC 3010,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:chen.j@wehi.edu.au" class="email">chen.j@wehi.edu.au</a>
      </address>
                  
            <h4 data-toc-skip class="date">18 Mar 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/mastR/blob/HEAD/vignettes/mastR_Demo.Rmd" class="external-link"><code>vignettes/mastR_Demo.Rmd</code></a></small>
      <div class="hidden name"><code>mastR_Demo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr>
<p><strong>Why do we need group markers</strong></p>
<p>Identifying marker genes for specific groups is crucial in various
biological and medical applications, as it allows us to distinguish
between different cell types or disease states.</p>
<p>For example, in cancer research, identifying marker genes that are
differentially expressed between cancer cells and normal cells can help
diagnose and monitor the progression of the disease, as well as identify
potential therapeutic targets. Similarly, in developmental biology,
identifying marker genes that are specific to certain cell types or
stages can help us understand the underlying mechanisms of
differentiation and development.</p>
<p>Furthermore, marker genes can also be used in diagnostic assays to
detect specific diseases or monitor treatment responses. For instance,
the presence of certain marker genes in a patient’s blood or tissue
sample can indicate the presence or progression of a disease.</p>
<p>Overall, identifying marker genes for specific groups is crucial for
understanding biological processes, diagnosing diseases, and developing
targeted therapies.</p>
<p><strong>Why do we need immune signature (cell type)</strong></p>
<p>The tumor microenvironment (TME) is made up of a diverse range of
cell types (fibroblasts, epithelial cells, endothelial cells, and immune
cells) as well as various extracellular components (collagens, growth
factors, hormones, cytokines, etc.). TIME is reported to be highly
associated with prognosis and various treatment response to many kinds
of cancers.</p>
<p>Recent studies have highlighted the role of immune components in the
TME in modulating tumor progression, making them attractive therapeutic
targets. These components make up the tumor immune microenvironment
(TIME), which is a subset of the TME. Subsequently, it’s proved that
tumour infiltrating lymphocytes (TILs) play an essential role in tumour
progression, metastasis and treatment response.</p>
<p>This drives TILs to be a strong prognostic indicator for better
precision therapy of cancer patients. And the identification of these
TILs are the subject of keen research interest due to the roles of
specific subset of immune cells acting on different tissues types.</p>
<p>Identification of these cell types are based on flow cytometry in the
past, which is limited in the granularity. But now with the advance of
single cell RNA-seq and spatial transcriptomics technologies, more
detailed and novel cell types or subtypes are being identified. One of
the key paradygm of scRNAseq and spatial is to further investigate cell
types in the TIME and understand cellullar heterogeniety in the TME.</p>
<p>To quantify TILs infiltration (in bulk), estimate cellular
composition (in bulk), annotate single cell types (in scRNA) or identify
sample states/activities, many computational methods like cell
deconvolution (CIBERSORTx), marker based annotation (CelliD) and sample
scoring (singscore) are developed. But to distinguish between closely
related cell types, estimate cell composition and states, a refined
signature is required which typiclaly requires manual curation by domain
expert. With the increasing large amount of data and cell types, this is
gradually getting to be intenable and will require a different approach
to automatically screen such signatures.</p>
<p>mastR is such a softare package designed to automatically screen
signature for researchers interest, which can save a lot of time and
manual labor work.</p>
<p><strong>What this package does</strong></p>
<p><em>mastR</em>, MArkers Screening Tool in R, is a package to
automatically derive signature for specific group of interest in
specific tissue.</p>
<p>With mastR, users can simply input their expression data containing
the groups of interest, along with group labels, to obtain a list of
marker genes (signature) for the target group.</p>
<p>Although there’re a lot of tools developed to generate cell type
specific markers, they are all designed for scRNA-seq data. And most of
them utilize machine learning to select features contributing to the
classification result, which is less robust and consistent across
datasets when compared with statistical method, like empirical Bayesian
in limma. And some of them will always return a signature even if the
data doesn’t have any.</p>
<p>Although DE analysis can also be done on scRNA data, like
<code><a href="https://satijalab.org/seurat/reference/FindMarkers.html" class="external-link">Seurat::FindMarkers()</a></code>, it’s reported that DE analysis done
on peudo-bulk scRNA data is more robust and reliable than directly done
on scRNA data.</p>
<p>Thus, our mastR is designed to generate a more refined list of
siganture genes from multiple group comparisons based on
<code>edgeR</code> and <code>limma</code> DE analysis result. The final
signature is selected by rank product score test for picking up genes
with high ranks in the most of comparisons. The rank can be ordered by
any gene statistic generated by limma analysis. Signature can be further
refined by keeping top n DEGs in specified comparison(s), which can help
to improve the discrimination between similar cell types.</p>
<p>Another novel point of mastR is that all of the current markers
generation tools only consider the genes contribution to the
classification, but ignore the background noise of tissue-specificity.
mastR offers a background noise removal function, designed to eliminate
parts of the marker genes that are highly expressed in specific tissues
or cancer cells, thereby avoiding potential confusion caused by
background or sample purity in specific tissue applications.</p>
<p>mastR also allows people to build a markers pool before signature
screening, which might contain all of potential markers or interesting
genes. The final signature will be integrated with this pool (by
intersection), which can help to constrain the final signature within
the interested pathway-related genes or functional gene-sets. People can
borrow the published knowledge to build this.</p>
<p>The need for this package arises from the importance of identifying
specific genes that are differentially expressed in different groups or
tissues, as these genes can serve as biomarkers for diagnosis,
prognosis, and therapeutic targeting. However, identifying marker genes
can be a challenging and time-consuming task, and the presence of
background noise can lead to erroneous results. Our package simplifies
and streamlines this process, allowing researchers to focus on their
analyses and interpretations without the burden of manual marker gene
selection and background noise removal.</p>
<p>This report demonstrates the main functions of mastR 0.9.0.</p>
<p>As a simple demo, here the report will specifically demonstrate the
signature screening workflow of NK cells in colorectal cancer (CRC),
then assess the final result by using some visualization functions.</p>
<p><strong>How does mastR screen the signature</strong></p>
<ul>
<li>step 1. generate markers pool<br>
</li>
<li>step 2. screen markers from the pool<br>
</li>
<li>step 3. visualize signature performance</li>
</ul>
<p><strong>Applications</strong></p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>score samples<br>
</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>estimate cellular proportion<br>
</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>single cell annotation</li>
</ol></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<hr>
<p><code>mastR</code> R package can be installed from Bioconductor or <a href="https://github.com/Gene233/mastR" class="external-link">GitHub</a>.</p>
<p>The most updated version of <code>mastR</code> is hosted on GitHub
and can be installed using <code><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">devtools::install_github()</a></code>
function provided by <a href="https://cran.r-project.org/package=devtools" class="external-link">devtools</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!requireNamespace("devtools", quietly = TRUE)) {</span></span>
<span><span class="co">#   install.packages("devtools")</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># if (!requireNamespace("mastR", quietly = TRUE)) {</span></span>
<span><span class="co">#   devtools::install_github("Gene233/mastR")</span></span>
<span><span class="co"># }</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"mastR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"mastR"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/mastR" class="external-link">mastR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1--build-immune-markers-pool">Step 1. Build Immune Markers Pool<a class="anchor" aria-label="anchor" href="#step-1--build-immune-markers-pool"></a>
</h2>
<hr>
<p>The first step is to define the original markers pool this analysis
will be based on.</p>
<p>The final signature will only be the intersected genes with the
markers pool. The whole gene list of the data will be regarded as the
markers pool if no preliminary result or intereted genes are provided.
But <em>note</em> that <code>markers = NULL</code> won’t keep any
special genes if they fail the filtration by edgeR.</p>
<p>Users can build the markers pool of interest from the available
datasets or build from MSigDB, PanglaoDB or LM7/LM22 by our imported
functions, if users have any preliminary knowledge about the target
group type (cell type). All genes in the pool will be reserved for DE
analysis even failed the filtration.</p>
<p>The standard pool building consists of:</p>
<ul>
<li>
<ol style="list-style-type: decimal">
<li>generate from sources</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>generate from LM7/22 signature matrix for CIBERSORT</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>generate from MSigDB</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>generate from PanglaoDB</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>generate from customized gene list</li>
</ol></li>
</ul>
</li>
<li><ol start="2" style="list-style-type: decimal">
<li>merge gene-sets from all sources together</li>
</ol></li>
</ul>
<p>mastR allows specific markers to be conveniently loaded as
follows:</p>
<ul>
<li><p><code>NK_markers</code> dataset is a combination of CIBERSORT
LM7, LM22 and human orthologs in mice from <a href="https://cancerimmunolres.aacrjournals.org/content/7/7/1162.long" class="external-link">Huntington</a>.
It contains 114 genes in total.</p></li>
<li><p><code>LM7</code> and <code>LM22</code> are signature matrices,
you can get more details by <code><a href="../reference/LM7.html">?mastR::LM7</a></code> or
<code><a href="../reference/LM22.html">?mastR::LM22</a></code>.</p></li>
<li><p><code>msigdb_gobp_nk</code> is a ‘GeneSetCollection’ object,
contains genesets with gene-set name matched to ‘NATURAL_KILLER’ from <a href="GO:BP" class="uri">GO:BP</a> MSigDB v7.4 database. More details
for <code><a href="../reference/msigdb_gobp_nk.html">?mastR::msigdb_gobp_nk</a></code>.</p></li>
</ul>
<p>The output of markers generation functions is either ‘GeneSet’ or
‘GeneSetCollection’ object.</p>
<div class="section level3">
<h3 id="generate-from-sources">Generate from sources<a class="anchor" aria-label="anchor" href="#generate-from-sources"></a>
</h3>
<p>To screen the immune cell subset signatures, we need to generate a
markers pool first. mastR allows people to generate markers pool from
multiple resources as metinoned before.</p>
<p>In this demo, we will load some example datasets to show how it
works. The datasets used in this report have been built within the
package or can be accessed publicly. You can use the following scripts
to load them into your R environment.</p>
<div class="section level4">
<h4 id="markers-from-lm">Markers from LM<a class="anchor" aria-label="anchor" href="#markers-from-lm"></a>
</h4>
<p>Markers can also be generated from LM7/22 signature matrix to get
immune cells markers using function <code><a href="../reference/get_lm_sig.html">get_lm_sig()</a></code>.</p>
<ul>
<li>
<code>LM7/22</code> are signature matrix from CIBERSORT, contains
7/22 immune cell subsets markers lists.</li>
</ul>
<p>Users can extract markers for cell subsets matched to the given regex
pattern from imported data <code>LM7</code>/<code>LM22</code>.</p>
<p>By using <code><a href="../reference/gsc_plot.html">gsc_plot()</a></code>, an UpSetR plot across all gene-sets
in the input would be printed. The input can be either ‘GeneSet’ or
‘GeneSetCollection’ objects. Using <code><a href="../reference/gsc_plot.html">gsc_plot()</a></code> we can see
how different gene-sets intersect with each other.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"LM7"</span>, <span class="st">"LM22"</span><span class="op">)</span></span>
<span><span class="co">## only retrieve LM7</span></span>
<span><span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm7.pattern <span class="op">=</span> <span class="st">"^NK"</span><span class="op">)</span></span>
<span><span class="co">#&gt; setName: LM7 </span></span>
<span><span class="co">#&gt; geneIds: CD244, FASLG, ..., XCL2 (total: 21)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Null </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span></span>
<span><span class="co">## only retrieve LM22</span></span>
<span><span class="co"># get_lm_sig(lm22.pattern = "NK cells")</span></span>
<span></span>
<span><span class="co">## collect both LM7 and LM22</span></span>
<span><span class="va">LM</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm7.pattern <span class="op">=</span> <span class="st">"^NK"</span>, lm22.pattern <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: LM7, LM22 (2 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: CD244, FASLG, ..., ZNF135 (92 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span>
<span><span class="co">## show upset diagram</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">LM</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/LM%20markers-1.png" width="768"></p>
</div>
<div class="section level4">
<h4 id="markers-from-msigdb">Markers from MSigDB<a class="anchor" aria-label="anchor" href="#markers-from-msigdb"></a>
</h4>
<p>Gene-sets from MSigDB can also be searched and collected by function
<code><a href="../reference/get_gsc_sig.html">get_gsc_sig()</a></code>.</p>
<ul>
<li>The Molecular Signatures Database (MSigDB) is a resource of tens of
thousands of annotated gene sets for use with GSEA software, divided
into Human and Mouse collections.</li>
</ul>
<p>Users can use the <code>species</code>, <code>cat</code>,
<code>subcat</code> and <code>pattern</code> to specify which gene-sets
to be added.</p>
<p>All gene-sets with gene-set name matched to the regex
<code>pattern</code> in the whole MSigDB would be extracted if
<code>cat</code> and <code>subcat</code> are not set, otherwise only the
genesets under the union of <code>cat</code> and <code>subcat</code>
would be searched.</p>
<p>In this demo, we’ll use a small collection of MSigDB
<code>msigdb_gobp_nk</code> to show how it works.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"msigdb_gobp_nk"</span><span class="op">)</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gsc_sig.html">get_gsc_sig</a></span><span class="op">(</span>gsc <span class="op">=</span> <span class="va">msigdb_gobp_nk</span>,</span>
<span>                    pattern <span class="op">=</span> <span class="st">"NATURAL_KILLER_CELL_MEDIATED"</span><span class="op">)</span></span>
<span><span class="va">MSig</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY, GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET, ..., GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY (18 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: AP1G1, ARRB2, ..., KLRC4-KLRK1 (67 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: BroadCollection (1 total)</span></span>
<span><span class="co">## cut geneset name within 11 characters</span></span>
<span><span class="va">gsn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span>, <span class="va">LETTERS</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">setName</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="va">i</span><span class="op">]</span></span>
<span><span class="op">}</span>   </span>
<span><span class="co">## show upset diagram of collected gene-sets</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/MSigDB%20genesets-1.png" width="768"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsn</span>  <span class="co">## show gene-set names</span></span>
<span><span class="co">#&gt;                                                                                                          A </span></span>
<span><span class="co">#&gt;                                                               "GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY" </span></span>
<span><span class="co">#&gt;                                                                                                          B </span></span>
<span><span class="co">#&gt;                        "GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET" </span></span>
<span><span class="co">#&gt;                                                                                                          C </span></span>
<span><span class="co">#&gt;                                          "GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNE_RESPONSE_TO_TUMOR_CELL" </span></span>
<span><span class="co">#&gt;                                                                                                          D </span></span>
<span><span class="co">#&gt;                                                 "GOBP_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY" </span></span>
<span><span class="co">#&gt;                                                                                                          E </span></span>
<span><span class="co">#&gt;                                        "GOBP_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY" </span></span>
<span><span class="co">#&gt;                                                                                                          F </span></span>
<span><span class="co">#&gt;                                        "GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY" </span></span>
<span><span class="co">#&gt;                                                                                                          G </span></span>
<span><span class="co">#&gt;                            "GOBP_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNE_RESPONSE_TO_TUMOR_CELL" </span></span>
<span><span class="co">#&gt;                                                                                                          H </span></span>
<span><span class="co">#&gt;                   "GOBP_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNE_RESPONSE_TO_TUMOR_CELL" </span></span>
<span><span class="co">#&gt;                                                                                                          I </span></span>
<span><span class="co">#&gt;                   "GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNE_RESPONSE_TO_TUMOR_CELL" </span></span>
<span><span class="co">#&gt;                                                                                                          J </span></span>
<span><span class="co">#&gt;          "GOBP_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET" </span></span>
<span><span class="co">#&gt;                                                                                                          K </span></span>
<span><span class="co">#&gt; "GOBP_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET" </span></span>
<span><span class="co">#&gt;                                                                                                          L </span></span>
<span><span class="co">#&gt; "GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET" </span></span>
<span><span class="co">#&gt;                                                                                                          M </span></span>
<span><span class="co">#&gt;                                                           "GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                                                                          N </span></span>
<span><span class="co">#&gt;                                             "GOBP_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                                                                          O </span></span>
<span><span class="co">#&gt;                                           "GOBP_PROTECTION_FROM_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                                                                          P </span></span>
<span><span class="co">#&gt;                                         "GOBP_SUSCEPTIBILITY_TO_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                                                                          Q </span></span>
<span><span class="co">#&gt;                                    "GOBP_NEGATIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                                                                          R </span></span>
<span><span class="co">#&gt;                                    "GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY"</span></span></code></pre></div>
<p>As we can see, there’re 18 gene-sets in MSig, which is too many for
visualization. Thus, we can use another function
<code><a href="../reference/merge_markers.html">merge_markers()</a></code> to merge all gene-sets into one ‘GeneSet’
object.</p>
<p>The input of <code><a href="../reference/merge_markers.html">merge_markers()</a></code> can be a list of vectors of
genes, or a list of GeneSet objects, or a GeneSetCollection object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## merge all genesets into one</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">setName</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"MSigDB"</span></span></code></pre></div>
<p>It’s also available to search on the whole MSigDB without loading it
yourself by setting param <code>data = "msigdb"</code> in the function,
specific species or version can also be set as optional params.</p>
</div>
<div class="section level4">
<h4 id="markers-from-panglaodb">Markers from PanglaoDB<a class="anchor" aria-label="anchor" href="#markers-from-panglaodb"></a>
</h4>
<p>Markers from PanlaoDB can also be obtained to build the markers pool
by using function <code><a href="../reference/get_panglao_sig.html">get_panglao_sig()</a></code>.</p>
<ul>
<li>PanglaoDB is a database for the scientific community interested in
exploration of single cell RNA sequencing experiments from mouse and
human. It collects and integrates data from multiple studies and present
them through a unified framework.</li>
</ul>
<p>Users can use <code><a href="../reference/list_panglao_organs.html">list_panglao_organs()</a></code> and
<code><a href="../reference/list_panglao_types.html">list_panglao_types()</a></code> functions to list all available organs
and cell types on PanglaoDB website and use
<code><a href="../reference/get_panglao_sig.html">get_panglao_sig()</a></code> function to retrive them.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show availbable organs on PanglaoDB</span></span>
<span><span class="fu"><a href="../reference/list_panglao_organs.html">list_panglao_organs</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## show available cell types of interest organ on PanglaoDB</span></span>
<span><span class="co">## Number in the bracket represents the number of markers for each cell type (in both Homo and Mus).</span></span>
<span><span class="fu"><a href="../reference/list_panglao_types.html">list_panglao_types</a></span><span class="op">(</span>organ <span class="op">=</span> <span class="st">"Immune system"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## collect all "NK cells" markers from PanglaoDB website</span></span>
<span><span class="va">Panglao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_panglao_sig.html">get_panglao_sig</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Panglao</span></span>
<span><span class="co">## number differs from 'NK cells' under list_panglao_types(organ = "Immune system"), because we only keep 'Hs' markers.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="markers-from-customized-gene-list">Markers from customized gene list<a class="anchor" aria-label="anchor" href="#markers-from-customized-gene-list"></a>
</h4>
<p>Markers can also be generated from customized gene list. Here we use
a list of natural killer (NK) cell markers from Curson et al 2019 from
<a href="https://aacrjournals.org/cancerimmunolres/article/7/7/1162/469488/A-Gene-Signature-Predicting-Natural-Killer-Cell" class="external-link">Curson’s
Publication</a> as a customized gene list.</p>
<ul>
<li>
<code>NK_markers</code> dataset is a combination of CIBERSORT LM7,
LM22 and human orthologs in mice from Huntington. It contains 114 genes
in total.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show what NK_markers looks like:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"NK_markers"</span><span class="op">)</span></span>
<span><span class="va">NK_markers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 114 × 4</span></span></span>
<span><span class="co">#&gt;    HGNC_Symbol LM22  LM7   Huntington</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> APOBEC3G    TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> APOL6       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> AZU1        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> BPI         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CAMP        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CCL4        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CCL5        TRUE  -     TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CCND2       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CD160       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CD2         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 104 more rows</span></span></span>
<span></span>
<span><span class="co">## convert NK markers into 'GeneSet' object</span></span>
<span><span class="va">nk_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html" class="external-link">GeneSet</a></span><span class="op">(</span><span class="va">NK_markers</span><span class="op">$</span><span class="va">HGNC_Symbol</span>,</span>
<span>                geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                setName <span class="op">=</span> <span class="st">"NK_markers"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="integrate-markers-pool">Integrate Markers Pool<a class="anchor" aria-label="anchor" href="#integrate-markers-pool"></a>
</h3>
<p>Now we have multiple lists of markers from different sources, to use
them in the subsequent analysis, we need to merge them together.</p>
<p>All markers can be merged into one ‘GeneSet’ object by using function
<code><a href="../reference/merge_markers.html">merge_markers()</a></code> we mentioned before. Each marker’s origin
will be saved as a data.frame under the ‘longDescription’ slot of the
output in json format.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html" class="external-link">GeneSetCollection</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">nk_m</span>, <span class="va">LM</span>, <span class="va">MSig</span><span class="op">)</span><span class="op">)</span> <span class="co">## add Panglao if you run it</span></span>
<span><span class="va">Markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## upset plot</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/integrate%20markers-1.png" width="768"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Markers</span></span>
<span><span class="co">#&gt; setName: merged_markers_pool </span></span>
<span><span class="co">#&gt; geneIds: APOBEC3G, APOL6, ..., KLRC4-KLRK1 (total: 167)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Computed </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span>
<span></span>
<span><span class="co">## to show the table summary of merged list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/jsonlite/man/fromJSON.html" class="external-link">fromJSON</a></span><span class="op">(</span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">longDescription</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       Gene NK_markers LM7 LM22 MSigDB</span></span>
<span><span class="co">#&gt; 1    AP1G1          -   -    -   TRUE</span></span>
<span><span class="co">#&gt; 2 APOBEC3G       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 3    APOL6       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 4    ARRB2          -   -    -   TRUE</span></span>
<span><span class="co">#&gt; 5     AZU1       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 6      BPI       TRUE   - TRUE      -</span></span></code></pre></div>
<p>Now we get a large markers pool for screening, next we need to screen
signature genes from the pool for both cell subset and tissue
specificity.</p>
</div>
</div>
<div class="section level2">
<h2 id="step-2--screen-markers-from-the-markers-pool">Step 2. Screen Markers from the Markers Pool<a class="anchor" aria-label="anchor" href="#step-2--screen-markers-from-the-markers-pool"></a>
</h2>
<div class="section level3">
<h3 id="screen-markers-for-group">Screen Markers for Group<a class="anchor" aria-label="anchor" href="#screen-markers-for-group"></a>
</h3>
<hr>
<div class="section level4">
<h4 id="screen-markers-against-other-groups">Screen Markers Against Other Groups<a class="anchor" aria-label="anchor" href="#screen-markers-against-other-groups"></a>
</h4>
<p>For group specificity, there’re 4 main steps:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>process data: filtration, normalization, sample weighting and linear
model fit</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>differential expression (DE) analysis<br>
</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>feature selection: pick differentially expressed genes with high
ranks in most of comparisons</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>integrate selected genes with markers pool</li>
</ol></li>
</ul>
<p>Customized external data is accepted in diverse formats: DGEList,
eSet, matrix… For example, a large collection of 1561 bulk RNA-seq
samples generated by DICE from pure populations of human immune cells,
can be used to as input data. It can be loaded by function
<code>celldex::DatabaseImmuneCellExpressionData()</code>, which contains
5 main cell types and 15 fine cell types.</p>
<p><strong><em>Notice</em></strong>: all input data must be raw counts
data or log-transformed expression data.</p>
<p>In this demo, we use our imported example data <code>im_data_6</code>
from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60424" class="external-link">GSE60424</a>
as input. Only samples from healthy individuals are kept, and ‘Whole
Blood’ or ‘PBMC’ cells are removed in dataset. It can also be obtained
by <code>GEOquery::getGEO()</code>.</p>
<ul>
<li>
<code>im_data_6</code> is a eSet object, containing RNA-seq TMM
normalized counts data of 6 sorted immune cell subsets * 4 samples. More
details in <code><a href="../reference/im_data_6.html">?mastR::im_data_6</a></code>.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"im_data_6"</span><span class="op">)</span></span>
<span><span class="va">im_data_6</span></span>
<span><span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">#&gt; assayData: 50045 features, 24 samples </span></span>
<span><span class="co">#&gt;   element names: exprs </span></span>
<span><span class="co">#&gt; protocolData: none</span></span>
<span><span class="co">#&gt; phenoData</span></span>
<span><span class="co">#&gt;   sampleNames: GSM1479438 GSM1479439 ... GSM1479525 (24 total)</span></span>
<span><span class="co">#&gt;   varLabels: title geo_accession ... years since diagnosis:ch1 (66</span></span>
<span><span class="co">#&gt;     total)</span></span>
<span><span class="co">#&gt;   varMetadata: labelDescription</span></span>
<span><span class="co">#&gt; featureData: none</span></span>
<span><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">#&gt;   pubMedIds: 25314013 </span></span>
<span><span class="co">#&gt; Annotation: GPL15456</span></span></code></pre></div>
<div class="section level5">
<h5 id="process">process data<a class="anchor" aria-label="anchor" href="#process"></a>
</h5>
<p>To screen signature for cell subset specificity, some processes need
to be done on the data first.</p>
<p><code><a href="../reference/process_data.html">process_data()</a></code> provided by mastR can do the genes
filtration, data normalization, samples weighting, linear model fit and
the computation of gene differential expression statistic for you.</p>
<p>Users need to provide an expression data with sample group labels
(containing interested target group), then it will return a
<code>DGEList</code> object with filtered counts (unfiltered counts data
will be saved under “original”), vfit (<code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code> is
done), tfit (<code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">limma::lmFit()</a></code> and <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code>
are done).</p>
<p>What this function does is:</p>
<ul>
<li><p>Data will be filtered by the given cutoff, all gene with low
expression will be removed by edgeR.</p></li>
<li><p>If the given data is raw count data
(<code>normalize = TRUE</code>), further normalization ‘TMM’ and
<code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code> fit will be done inside the process function,
otherwise the <code>trend</code> of limma will be applied on the given
data.</p></li>
<li><p>Final linear model will be fitted.</p></li>
<li><p>gene statistic for DE analysis are computed by
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code>.</p></li>
</ul>
<p>Let’s see what cell types are in <code>im_data_6</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">im_data_6</span><span class="op">$</span><span class="va">`celltype:ch1`</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     B-cells         CD4         CD8   Monocytes Neutrophils          NK </span></span>
<span><span class="co">#&gt;           4           4           4           4           4           4</span></span></code></pre></div>
<p>With below simple params, data processing can be easily done by
<code><a href="../reference/process_data.html">process_data()</a></code>:</p>
<ul>
<li>
<code>data</code> param can be a bulk RNA-seq expression object with
cell subset labels, can be matrix, eSet, DGEList, … (expression must be
either raw counts or logcounts).</li>
<li>
<code>group_col</code> param can specify the column name of groups
(labels), can be a vector when <code>data</code> is a matrix.<br>
</li>
<li>
<code>target_group</code> param can specify the cell target_group
name of interest. Users can choose one name in <code>group_col</code>
vector.<br>
</li>
<li>
<code>markers</code> param could specify a geneset to be kept no
matter if they pass the filtration. Setting it to <code>NULL</code> can
skip this. Must be gene SYMBOL ID.<br>
</li>
<li>
<code>gene_id</code> specify the gene ID type of rownames of data
when markers is not NULL, could be one of ‘ENSEMBL’, ‘SYMBOL’,
‘ENTREZ’…, default ‘SYMBOL’.</li>
</ul>
<p>More optional param details can be found in
<code>help(proc_data)</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proc_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/process_data.html">process_data</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,  </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span> <span class="co">## rownames of im_data_6 is ENSEMBL ID</span></span>
<span>  <span class="co"># summary = TRUE ## if to show the summary of tfit result</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             4012         3949       3146   2698   2155</span></span>
<span><span class="co">#&gt; NotSig           1492         2694       4429   5001   6201</span></span>
<span><span class="co">#&gt; Up               4945         3806       2874   2750   2093</span></span>
<span><span class="co"># summary(limma::decideTests(proc_data$tfit))</span></span>
<span></span>
<span><span class="co">## add voom fitted expression as a new list of proc_data for use</span></span>
<span><span class="va">proc_data</span><span class="op">$</span><span class="va">voomE</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="visualize-qc-before-and-after">visualize QC before and after<a class="anchor" aria-label="anchor" href="#visualize-qc-before-and-after"></a>
</h5>
<p>After data processing, we need to check whether the data has been
well filtered , normalized and fitted.</p>
<p>mastR provides visualization functions for this, to compare the data
before and after <code><a href="../reference/process_data.html">process_data()</a></code> and see how well the QC is
done.</p>
<p>To see if the low quality genes are removed,
<code><a href="../reference/plot_diagnostics.html">plot_diagnostics()</a></code> can be used to show the expression
distribution, RLE and MDS plots.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr1</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">im_data_6</span><span class="op">@</span><span class="va">assayData</span><span class="op">$</span><span class="va">exprs</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co">## or proc_data$original_counts</span></span>
<span><span class="va">expr2</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">voomE</span></span>
<span><span class="co">## plot expression distribution, RLE and MDS</span></span>
<span><span class="fu"><a href="../reference/plot_diagnostics.html">plot_diagnostics</a></span><span class="op">(</span></span>
<span>  <span class="va">expr1</span>,</span>
<span>  <span class="va">expr2</span>,</span>
<span>  group_col <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; $density</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-1.png" width="960"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $RLE</span></span></code></pre>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-2.png" width="960"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $MDS</span></span></code></pre>
<p><img src="mastR_Demo_files/figure-html/plot%20diagnostics-3.png" width="960"></p>
<p>Mean-variance trend can be plot by <code><a href="../reference/plot_mean_var.html">plot_mean_var()</a></code> to
show the linear fit result.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot mean-variance trend</span></span>
<span><span class="fu"><a href="../reference/plot_mean_var.html">plot_mean_var</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/plot%20mean-variance-1.png" width="960"></p>
<p>PCA can be plot by <code><a href="../reference/pca_matrix_plot.html">pca_matrix_plot()</a></code> to show the
normalization result and if the samples can be clustered by groups. You
can select how many top n PCs to be plot by param <code>n</code>.</p>
<p>The input for this function should be log-normalized.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot PCA</span></span>
<span><span class="co">## before process</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">expr1</span>, group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/plot%20PCA%20mat-1.png" width="768"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## after process</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">expr2</span>, group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/plot%20PCA%20mat-2.png" width="768"></p>
</div>
<div class="section level5">
<h5 id="selection">select signature genes<a class="anchor" aria-label="anchor" href="#selection"></a>
</h5>
<p>Now we get processed data with DE statistic for each gene in each
comparison, then the proc_data can be passed into
<code><a href="../reference/select_sig.html">select_sig()</a></code> function for further genes selection based on
rank product scoring. Cell type specific signature can be obtained by
this.</p>
<ul>
<li>
<code>feature_selection</code> param to choose whether to use rank
product (‘rankproduct’) or not (‘none’) to select DEGs from multiple
comparisons of DE analysis. Default “auto” will perform “rankproduct”
and switch to “none” if the number of final “UP” and “DOWN” genes are
&lt; 5.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## get the same result as there's permutation test for rank product</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_sig.html">select_sig</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sig_ct</span><span class="op">)</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: UP, DOWN (2 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ENSG00000149294, ENSG00000173068, ..., ENSG00000162591 (98 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: NullIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
<p><strong><em>Note</em></strong>: All above steps from @ref(process) to
@ref(selection) for screening group signature can be done by one
integrated function <code><a href="../reference/get_degs.html">get_degs()</a></code>.</p>
<p><code><a href="../reference/get_degs.html">get_degs()</a></code> function can filter low expression genes out
from immune dataset, normalize and fit voom model to data if
<code>normalize = TRUE</code>, then a processed data
<code>proc_data</code> object and a differentially expressed
GeneSetCollection (‘UP’ and ‘DOWN’) will be returned as output.</p>
<p>QC plots can also be shown at the same time by setting optional param
<code>plot = TRUE</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG_6</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_degs.html">get_degs</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,  </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>  <span class="co">## convert ensembl IDs of rownames into symbols if gene_id != "SYMBOL"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## `proc_data` object would be saved</span></span>
<span><span class="va">proc_data</span> <span class="op">&lt;-</span> <span class="va">DEG_6</span><span class="op">$</span><span class="va">proc_data</span></span>
<span></span>
<span><span class="va">DEG_6</span><span class="op">$</span><span class="va">DEGs</span></span>
<span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">DEG_6</span><span class="op">$</span><span class="va">DEGs</span><span class="op">)</span></span>
<span><span class="va">DEG_6</span><span class="op">$</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">design</span> <span class="co">## show the design matrix</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="intersect-pool">intersect with markers pool<a class="anchor" aria-label="anchor" href="#intersect-pool"></a>
</h5>
<p>After getting specific signature genes for the target group, we will
only keep those UP regulated genes which are also present in our markers
pool <code>Markers</code>, to constrain the final signature wthin the
interested gene list.</p>
<p>PCA can be used to show clearer separation of NK cells from other
cells by using our screened UP DEGs. And the intersected genes with
markers pool explain more variance in PC1 compared with non-intersected
DEGs.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## convert ensembl IDs into symbols to match markers pool</span></span>
<span><span class="va">deg_up</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">sig_ct</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                 <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">## markers specific for NK cells</span></span>
<span><span class="va">m_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>, <span class="va">deg_up</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">deg_up</span><span class="op">)</span><span class="op">]</span>  <span class="co">## set ensembl ID as names for downstream visualization</span></span>
<span></span>
<span><span class="va">m_ct</span></span>
<span><span class="co">#&gt; ENSG00000122223 ENSG00000198821 ENSG00000172543 ENSG00000145649 ENSG00000197540 </span></span>
<span><span class="co">#&gt;         "CD244"         "CD247"          "CTSW"          "GZMA"          "GZMM" </span></span>
<span><span class="co">#&gt; ENSG00000100385 ENSG00000183542 ENSG00000150045 ENSG00000149294 ENSG00000189430 </span></span>
<span><span class="co">#&gt;         "IL2RB"         "KLRC4"         "KLRF1"         "NCAM1"          "NCR1" </span></span>
<span><span class="co">#&gt; ENSG00000168229 ENSG00000041353 ENSG00000198574 ENSG00000074966 ENSG00000115085 </span></span>
<span><span class="co">#&gt;         "PTGDR"        "RAB27B"        "SH2D1B"           "TXK"         "ZAP70" </span></span>
<span><span class="co">#&gt; ENSG00000134545 ENSG00000150637 </span></span>
<span><span class="co">#&gt;         "KLRC1"         "CD226"</span></span>
<span></span>
<span><span class="co">## PCA shows clear separation of NK cells</span></span>
<span><span class="co">## after intersection</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>                group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span>,</span>
<span>                n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"After intersection"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/markers%20for%20cell%20type-1.png" width="768"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## before intersection</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>                group_by <span class="op">=</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span>,</span>
<span>                n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Before intersection"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/markers%20for%20cell%20type-2.png" width="768"></p>
<p><strong><em>Note</em></strong>: There’s also a wrapper function
<code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> which can directly do the whole Step 2
in one step. The result would be the same as running the above workflow
step by step.</p>
<p><code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> can also be helpful when you have
multiple datasets (more details in Section @ref(multi-data)). It can
output final signature after aggregating signature lists from all
datasets.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run by wrapped function in one-step</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="remove-markers-with-background-noise-in-tissue-optional">Remove Markers with Background Noise in Tissue (optional)<a class="anchor" aria-label="anchor" href="#remove-markers-with-background-noise-in-tissue-optional"></a>
</h4>
<p>After screening signature for group specificity, we need to increase
the tissue specificity by removing background noise for the signature
now, what we need to do is:</p>
<ul>
<li>
<p>filter out cancer cell expressed genes</p>
<p>In order to remove background noise from the marker genes, we
utilized a signal-to-noise ratio (SNR) approach to filter out genes with
low SNR values. This was done to eliminate genes that have similar
expression patterns across the target group and background tissues,
which may be due to non-specific expression or technical artifacts.</p>
<p>The SNR is calculated as the difference between the mean expression
value of the group of interest and the mean expression value of the
background group, divided by the standard deviation of the background
group. Genes with low SNR values are considered to have low
discriminative power between the group of interest and the background
group, and are therefore filtered out.</p>
<p>By removing genes with low SNR values, we are able to improve the
specificity and accuracy of the marker genes, as well as reduce
potential confounding effects caused by non-specific expression or
technical artifacts. This approach is commonly used in gene expression
analysis to improve the reliability of differential expression analysis
and biomarker identification.</p>
<p>By doing this, we can guarantee the activity of our final signature
for infiltrating immune cells won’t be affected by tumor purity when
it’s applied on tumor samples.</p>
</li>
<li>
<p>optional</p>
<p>DE analysis to compare tumor infiltrating immune cells vs normal
immune cells (e.g. PBMC or normal tissue resident immune cells), add
DEGs into markers pool or combine it with the result of step 2. to
increase tissue/cancer specificity.</p>
</li>
</ul>
<p>mastR provides function <code><a href="../reference/remove_bg_snr.html">remove_bg_snr()</a></code> to remove genes
with background noise in cancer cells or tissues. Users need to provide
a data as signal data (e.g. immune cell data) and a data as background
noise data (e.g. cancer cells or tissue cells), then
<code><a href="../reference/remove_bg_snr.html">remove_bg_snr()</a></code> will compute SNR (signal to noise ratio)
for each gene in the given markers, and remove those genes with low SNR.
This can ensure our immune signature won’t be affected by tumor
purity.</p>
<p>Based on the <code>bg_data</code> input format, this function can be
used in 2 ways:</p>
<ul>
<li>use CCLE as <code>bg_data</code> input<br>
</li>
<li>use customized data as <code>bg_data</code> input</li>
</ul>
<div class="section level5">
<h5 id="use-ccle-as-bg_data">use CCLE as bg_data<a class="anchor" aria-label="anchor" href="#use-ccle-as-bg_data"></a>
</h5>
<p>The default setting of this <code><a href="../reference/remove_bg_snr.html">remove_bg_snr()</a></code> uses CCLE
from <a href="https://depmap.org/portal/download/" class="external-link">DepMap CCLE</a>,
containing RSEM quantified TPM data of 1392 cancer cell lines (until
2022.04).</p>
<p>The Cancer Cell Line Encyclopedia (CCLE) is a compilation of gene
expression, chromosomal copy number and massively parallel sequencing
data from more than 1,000 human cancer cell lines.</p>
<p>To specify cell subset markers for cancer type, use
<code><a href="../reference/remove_bg_snr.html">remove_bg_snr()</a></code> function to remove failing genes which are
also highly expressed by CRC adherent cell lines <em>per se</em>. This
can ensure our immune signature won’t be affected by tumor purity.</p>
<ul>
<li>
<code>bg_data</code> can be either “CCLE” or expression dataset,
e.g. matrix, eSet, DGEList, Seurat and so on.<br>
</li>
<li>
<code>sig_data</code> can be a matrix or DGEList, as signal data,
always use processed data after <code><a href="../reference/process_data.html">process_data()</a></code>.</li>
<li>
<code>b_group_col</code> param is the group vector of
<code>bg_data</code> samples, or the name of group column.<br>
</li>
<li>
<code>b_target_group</code> param is an regex pattern to specify the
samples of interest, e.g. choose ‘colorectal’ to select all colorectal
adherent cell lines from CCLE.<br>
</li>
<li>
<code>s_group_col</code> is a vector or character, to specify the
group of target_group in <code>sig_data</code>, or the name of group
column.</li>
<li>
<code>s_target_group</code> is a regex pattern, to specify the
target group of interest in <code>sig_data</code>.</li>
<li>
<code>markers</code> is the genes need to be filtered, should be
gene symbols.</li>
<li>
<code>snr</code> param is the the cutoff of SNR to screen markers
which are not or lowly expressed in bg_data.</li>
</ul>
<p><strong><em>Note</em></strong>: Both of <code>bg_data</code> and
<code>sig_data</code> should be log-normalized.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load CCLE data</span></span>
<span><span class="va">CCLE</span> <span class="op">&lt;-</span> <span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/depmap/man/TPM.html" class="external-link">depmap_TPM</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">CCLE_meta</span> <span class="op">&lt;-</span> <span class="fu">depmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/depmap/man/metadata.html" class="external-link">depmap_metadata</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## filter</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_bg_snr.html">remove_bg_snr</a></span><span class="op">(</span>bg_data <span class="op">=</span> <span class="st">"CCLE"</span>,</span>
<span>                       sig_data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>                       b_group_col <span class="op">=</span> <span class="st">"primary_disease"</span>,</span>
<span>                       b_target_group <span class="op">=</span> <span class="st">"colorectal"</span>,</span>
<span>                       s_group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>                       s_target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>                       markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>                       filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span><span class="op">)</span>, <span class="co">## needed when bg_data is 'CCLE'</span></span>
<span>                       gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>,</span>
<span>                       s_slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>                       ccle_tpm <span class="op">=</span> <span class="va">CCLE</span>, <span class="co">## only provided when data = 'CCLE'</span></span>
<span>                       ccle_meta <span class="op">=</span> <span class="va">CCLE_meta</span><span class="op">)</span></span>
<span><span class="co">## or without loading CCLE yourself</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_bg_snr.html">remove_bg_snr</a></span><span class="op">(</span>sig_data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>                       b_group_col <span class="op">=</span> <span class="st">"primary_disease"</span>,</span>
<span>                       b_target_group <span class="op">=</span> <span class="st">"colorectal"</span>,</span>
<span>                       s_group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>                       s_target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>                       markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>                       filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span><span class="op">)</span>,</span>
<span>                       gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>,</span>
<span>                       s_slot <span class="op">=</span> <span class="st">"voomE"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="use-customized-dataset-as-input">use customized dataset as input<a class="anchor" aria-label="anchor" href="#use-customized-dataset-as-input"></a>
</h5>
<p>Other customized data is also accepted as <code>bg_data</code> in
diverse formats: eSet, matrix, DGEList… But all input must be
log-transformed expression data.</p>
<p>In this demo, we use a small imported dataset <code>ccle_crc_5</code>
in mastR as a customized dataset, the expression of it is logTPM.</p>
<ul>
<li>
<code>ccle_crc_5</code> is a DGEList object only contains 5 CRC cell
line samples from CCLE. More details for
<code><a href="../reference/ccle_crc_5.html">help(ccle_crc_5)</a></code>.</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ccle_crc_5"</span><span class="op">)</span></span>
<span><span class="va">ccle_crc_5</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span>
<span><span class="co">#&gt; An object of class "DGEList"</span></span>
<span><span class="co">#&gt; $counts</span></span>
<span><span class="co">#&gt;          SNU283_LARGE_INTESTINE TGBC18TKB_LARGE_INTESTINE SW837_LARGE_INTESTINE</span></span>
<span><span class="co">#&gt; TSPAN6               4.91408610                  6.045923             5.2570106</span></span>
<span><span class="co">#&gt; TNMD                 0.17632277                  0.000000             0.3334237</span></span>
<span><span class="co">#&gt; DPM1                 6.94684781                  6.724105             6.9242187</span></span>
<span><span class="co">#&gt; SCYL3                2.57773093                  2.260026             2.9653225</span></span>
<span><span class="co">#&gt; C1orf112             3.85399565                  3.835924             4.7618171</span></span>
<span><span class="co">#&gt; FGR                  0.00000000                  0.000000             0.1890338</span></span>
<span><span class="co">#&gt; CFH                  0.08406426                  4.115200             0.2509616</span></span>
<span><span class="co">#&gt; FUCA2                4.85549144                  6.263034             6.9090530</span></span>
<span><span class="co">#&gt; GCLC                 4.93404465                  4.398487             5.8750428</span></span>
<span><span class="co">#&gt; NFYA                 3.68144927                  3.727920             4.0703893</span></span>
<span><span class="co">#&gt;          SNU1040_LARGE_INTESTINE HT29_LARGE_INTESTINE</span></span>
<span><span class="co">#&gt; TSPAN6                5.16188768           4.39985467</span></span>
<span><span class="co">#&gt; TNMD                  0.46466827           0.00000000</span></span>
<span><span class="co">#&gt; DPM1                  6.67849451           6.71836163</span></span>
<span><span class="co">#&gt; SCYL3                 2.48284828           2.78450398</span></span>
<span><span class="co">#&gt; C1orf112              3.32624970           3.46858332</span></span>
<span><span class="co">#&gt; FGR                   1.48542683           0.01435529</span></span>
<span><span class="co">#&gt; CFH                   0.05658353           4.08151007</span></span>
<span><span class="co">#&gt; FUCA2                 5.96809075           5.35508676</span></span>
<span><span class="co">#&gt; GCLC                  5.15583017           5.10097765</span></span>
<span><span class="co">#&gt; NFYA                  4.81249823           4.32481060</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $samples</span></span>
<span><span class="co">#&gt;                           group lib.size norm.factors cancer</span></span>
<span><span class="co">#&gt; SNU283_LARGE_INTESTINE        1 49889.02            1    CRC</span></span>
<span><span class="co">#&gt; TGBC18TKB_LARGE_INTESTINE     1 49489.64            1    CRC</span></span>
<span><span class="co">#&gt; SW837_LARGE_INTESTINE         1 56426.42            1    CRC</span></span>
<span><span class="co">#&gt; SNU1040_LARGE_INTESTINE       1 54366.12            1    CRC</span></span>
<span><span class="co">#&gt; HT29_LARGE_INTESTINE          1 49232.06            1    CRC</span></span>
<span></span>
<span><span class="co">## markers against for CRC cell lines</span></span>
<span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_bg_snr.html">remove_bg_snr</a></span><span class="op">(</span></span>
<span>  bg_data <span class="op">=</span> <span class="va">ccle_crc_5</span>,</span>
<span>  sig_data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  b_group_col <span class="op">=</span> <span class="st">"cancer"</span>,</span>
<span>  b_target_group <span class="op">=</span> <span class="st">"CRC"</span>,</span>
<span>  s_group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  s_target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, <span class="co">## as there're only 5 samples in ccle_crc_5</span></span>
<span>  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>, <span class="co">## different gene IDs in bg_ and sig_ data</span></span>
<span>  s_slot <span class="op">=</span> <span class="st">"voomE"</span> <span class="co">## use voomE of proc_data as input</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Warning in remove_bg_snr(bg_data = bg_data, sig_data = sig_data[[s_slot]], : Gene CCL4 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene CCL5 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL2 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL5A is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS2 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS5 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene KLRA1P is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene LOC653757 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC24103 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene TRDC is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC61571 is not in the sig_data! So remove it!</span></span>
<span><span class="co">#&gt; Gene PIK3R6 is not in the sig_data! So remove it!</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "IL32"   "RAB27B" "STYK1"  "RAB27A" "TXK"    "TTC38"</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="final-group-signature-for-tissue">Final Group Signature for Tissue<a class="anchor" aria-label="anchor" href="#final-group-signature-for-tissue"></a>
</h3>
<p>After above steps, now we need to integrate the results together.</p>
<p>As we want the final signature to be specific to the target cell
subset while less affected by the tissue/cancer purity, intersection is
preferred to select the final signature after both specificity
screening: cell type (subset) and cancer (tissue) type.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_NK_CRC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="va">sig_NK_CRC</span></span>
<span><span class="co">#&gt;  [1] "CD244"  "CD247"  "CTSW"   "GZMA"   "GZMM"   "IL2RB"  "KLRC4"  "KLRF1" </span></span>
<span><span class="co">#&gt;  [9] "NCAM1"  "NCR1"   "PTGDR"  "RAB27B" "SH2D1B" "TXK"    "ZAP70"  "KLRC1" </span></span>
<span><span class="co">#&gt; [17] "CD226"</span></span></code></pre></div>
<p>But here, if we directly use the <code>m_ct</code> instead of
<code>Markers</code> as input <code>markers</code> for function
<code><a href="../reference/remove_bg_snr.html">remove_bg_snr()</a></code>, then <code>m_ccl</code> is already the
intersection result of original <code>m_ccl</code> and
<code>m_ct</code>, so we can skip this.</p>
<p>Now we get the final signature specific for both cell type and cancer
type!</p>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="step-3--visualize-the-screening-effect">Step 3. Visualize The
Screening Effect<a class="anchor" aria-label="anchor" href="#step-3--visualize-the-screening-effect"></a>
</h2>
<hr>
<p>Now that we get our specific signature, we need to assess how well
this signature can discriminate our target group from others. To easily
realize this, some popular visualization functions are implemented in
mastR.</p>
<ul>
<li>
<p><code><a href="../reference/pca_matrix_plot.html">pca_matrix_plot()</a></code> for matrix of PCA plots with top n
PCs.</p>
<p>To show if the target group can be separated from other groups and
how well the separation is. Users can also know how much variance has
been explained by PC1 with the signature.</p>
</li>
<li>
<p><code><a href="../reference/sig_heatmap.html">sig_heatmap()</a></code> for signature heatmap across groups
and its comparsion with markers pool.</p>
<p>To validate if the signature shows clear and differential expression
pattern between the target group and others, and if the pattern is
clearer and cleaner after the screening when compared with un-screeed
markers pool.</p>
</li>
<li>
<p><code><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot()</a></code> for signature genes rank
density distribution across groups.</p>
<p>To see if the signature genes are highly expressed within each sample
of the target group while lowly expressed in others.</p>
</li>
<li>
<p><code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code> for signature expression or singscore
boxplot across groups.</p>
<p>To test the overall performance of the whole signature when applying
it for scoring, singscore is used to compute the rank score with the
whole signature. The boxplot across groups can be used to test if the
whole signature is powerful enough to distinguish the target group from
others on scoring level. Median expression of each signature gene is
also allowed to plot.</p>
</li>
<li>
<p><code><a href="../reference/sig_scatter_plot.html">sig_scatter_plot()</a></code> for signature scaled expression
scatter plot of the target group vs all other groups respectively.</p>
<p>To visualize the correlation of the signature between the target and
other groups, the row-scaled gene expression is computed to make the
scatter plot. Based on it, we can easily see the co-expression
correlation and signature’s specificity.</p>
</li>
<li>
<p><code><a href="../reference/sig_gseaplot.html">sig_gseaplot()</a></code> for gene-set enrichment analysis
result display.</p>
<p>To further validate if the signature is significantly enriched in our
target group compared with other groups,
<code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">clusterProfiler::GSEA()</a></code> analysis is done inside this
function. Users can choose to display ‘dotplot’ or gseaplot’ by
<code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">clusterProfiler::GSEA()</a></code>.</p>
</li>
</ul>
<div class="section level3">
<h3 id="pca-matrix-plot">PCA matrix plot<a class="anchor" aria-label="anchor" href="#pca-matrix-plot"></a>
</h3>
<p>To see if our curated NK cells signature has more distinguishable
expression pattern from other immune cell subsets, we can use PCA plot
again here to show it.</p>
<p>Loadings can be drawn to show which genes are relevant to NK cells,
and result shows most of signature genes are indicating NK cells.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## as proc_data shows better discrimination, use it for visualization</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  features <span class="op">=</span> <span class="va">sig_NK_CRC</span>, </span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/pca-1.png" width="768"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## loadings can be shown by setting loading = TRUE</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  features <span class="op">=</span> <span class="va">sig_NK_CRC</span>, </span>
<span>  loading <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 6 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/pca-2.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p><code><a href="../reference/sig_heatmap.html">sig_heatmap()</a></code> function can plot the original NK cells
markers and curated signatures expression pattern heatmaps in immune
datasets.</p>
<ul>
<li>
<code>sigs</code> param is the final curated signature symbols,
while <code>markers</code> param is the original markers pool.<br>
</li>
<li>
<code>scale</code> could specify the <code>column</code> or
<code>row</code> to be scaled or no scaling by <code>none</code>.<br>
</li>
<li>`rank_plot`` is used to decide if to gene rank instead of expression
for heatmap.</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_heatmap.html">sig_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>  show_column_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Gene CCL4 is not in data</span></span>
<span><span class="co">#&gt; Gene CCL5 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DL2 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DL5A is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DS2 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DS5 is not in data</span></span>
<span><span class="co">#&gt; Gene KLRA1P is not in data</span></span>
<span><span class="co">#&gt; Gene LOC653757 is not in data</span></span>
<span><span class="co">#&gt; Gene MGC24103 is not in data</span></span>
<span><span class="co">#&gt; Gene TRDC is not in data</span></span>
<span><span class="co">#&gt; Gene MGC61571 is not in data</span></span>
<span><span class="co">#&gt; Gene PIK3R6 is not in data</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/heatmap-1.png" width="960"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## for multiple datasets</span></span>
<span><span class="co"># sig_heatmap(</span></span>
<span><span class="co">#   data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#               proc_data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"), </span></span>
<span><span class="co">#   markers = geneIds(Markers),</span></span>
<span><span class="co">#   gene_id = "ENSEMBL"</span></span>
<span><span class="co"># ) + patchwork::plot_layout(guides = "collect", ncol = 1)</span></span></code></pre></div>
<p>We can see the expression pattern of the signature is more
distinguishable in NK cells.</p>
</div>
<div class="section level3">
<h3 id="rank-density-plot">Rank Density Plot<a class="anchor" aria-label="anchor" href="#rank-density-plot"></a>
</h3>
<p>Use <code><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot()</a></code> function to make rank density
plot for all cell subsets in the data.</p>
<p>By setting <code>aggregate = TRUE</code>, only one plot for each cell
type with mean expression would be shown.</p>
<p>We might want to see the rank density plot in more than 1 datasets
for comparison, this can be done by simply passing a list of datasets to
function.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ## visualization on both original and processed data</span></span>
<span><span class="co"># sig_rankdensity_plot(</span></span>
<span><span class="co">#   data = list(im_data_6 = im_data_6,</span></span>
<span><span class="co">#               process_Data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"), </span></span>
<span><span class="co">#   gene_id = "ENSEMBL",</span></span>
<span><span class="co">#   slot = "voomE"</span></span>
<span><span class="co"># )</span></span>
<span></span>
<span><span class="co">## visualization on aggregated data</span></span>
<span><span class="fu"><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  aggregate <span class="op">=</span>  <span class="cn">TRUE</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/rankdensity%20plot-1.png" width="768"></p>
<p>We can find, processed data shows more digstinguishable rankdensity
distribution of screened NK signature genes.</p>
</div>
<div class="section level3">
<h3 id="signature-score-boxplot">Signature Score Boxplot<a class="anchor" aria-label="anchor" href="#signature-score-boxplot"></a>
</h3>
<p>To see if the curated NK cells signatures have evident higher
abundance in NK cells than other immune cell subsets, use
<code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code> function to make boxplot of the NK scores for
all cell subsets in the data.</p>
<p>NK score is calculated by <code>singscore</code> package.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>, </span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/score%20boxplot-1.png" width="768"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## To make boxplots for more than 1 immune datasets at once:</span></span>
<span><span class="co"># sig_boxplot(</span></span>
<span><span class="co">#   data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#               proc_data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"),</span></span>
<span><span class="co">#   target_group = "NK",</span></span>
<span><span class="co">#   gene_id = "ENSEMBL"</span></span>
<span><span class="co"># ) * guides(col = "none")</span></span></code></pre></div>
<p>Obviously, our signature always shows significantly higher score in
NK cells.</p>
</div>
<div class="section level3">
<h3 id="signature-abundance-boxplot">Signature Abundance Boxplot<a class="anchor" aria-label="anchor" href="#signature-abundance-boxplot"></a>
</h3>
<p><code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code> function can also generate boxplot of the
NK signature abundance for all cell subsets in the data by setting
<code>type = "expression"</code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"expression"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20boxplot-1.png" width="768"></p>
<p>It’s clear the score can increase the difference between NK cells and
others.</p>
</div>
<div class="section level3">
<h3 id="signature-abundance-scatter-plot">Signature Abundance Scatter Plot<a class="anchor" aria-label="anchor" href="#signature-abundance-scatter-plot"></a>
</h3>
<p>To see if the curated NK cells signature shows higher specificity to
NK cells than other immune cell subsets. Use
<code><a href="../reference/sig_scatter_plot.html">sig_scatter_plot()</a></code> function to make scatter plot of the NK
signature abundance for all cell subsets in the data.</p>
<p>If the signature shows specificity to NK, then most of signature
genes should appear in left-top region (scaled expression &gt; 1 in NK,
&lt; 1 in other), while the gene present in right-top region should
represent the co-expression in the both groups.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot-1.png" width="768"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## To make scatter plot for more than 1 immune datasets at once:</span></span>
<span><span class="co"># sig_scatter_plot(</span></span>
<span><span class="co">#   data = list(data_6 = im_data_6,</span></span>
<span><span class="co">#               proc_data = proc_data), </span></span>
<span><span class="co">#   sigs = sig_NK_CRC,</span></span>
<span><span class="co">#   group_col = c("celltype:ch1", "celltype.ch1"),</span></span>
<span><span class="co">#   target_group = "NK", </span></span>
<span><span class="co">#   gene_id = "ENSEMBL")</span></span></code></pre></div>
<p>Based on scatter plot result, we can actually do further filtration
to improve signature specificity by removing signature genes present in
right bottom or left bottom region.</p>
</div>
<div class="section level3">
<h3 id="signature-gsea-plot">Signature GSEA plot<a class="anchor" aria-label="anchor" href="#signature-gsea-plot"></a>
</h3>
<p>To see if the curated signature is enriched in NK cells than other
immune cell subsets. Use <code><a href="../reference/sig_gseaplot.html">sig_gseaplot()</a></code> function to make
GSEAplot.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## dotplot</span></span>
<span><span class="fu"><a href="../reference/sig_gseaplot.html">sig_gseaplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">sig_NK_CRC</span>, markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned many:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gseaplot-1.png" width="960"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## gseaplot</span></span>
<span><span class="fu"><a href="../reference/sig_gseaplot.html">sig_gseaplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">sig_NK_CRC</span>, markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"gseaplot"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned many:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gseaplot-2.png" width="960"></p>
<p>It’s clear that our screened signature is enriched in all
comparisons.</p>
</div>
</div>
<div class="section level2">
<h2 id="refine-signature-specificity">Refine Signature Specificity<a class="anchor" aria-label="anchor" href="#refine-signature-specificity"></a>
</h2>
<hr>
<p>According to the visualization result, we can see our NK signature
shows good performance in separating NK cells with all other cells. But
CD8+ T cells still show similar abundance to NK cells in some signature
genes expression, which might impair the ability of signature to
distinguish NK cells from CD8+ T cells.</p>
<p>But we know those genes with similar abundance across NK and CD8+ T
are importance and distinguishable for separating NK cells from other
non-CD8+ T cells. So we don’t recommend to simply delete those genes
based on our abundance scatter plot.</p>
<p>Instead, here we implemented a better way to increase the difference
between the ambiguous cell types.</p>
<p>By setting <code>keep.top</code> and <code>keep.group</code>, we can
easily keep the top n DEGs of specified DE comparisons, DEGs are ranked
based on <code>Rank</code> param. For instance, here we want keep more
distinguishable genes between NK and CD8+ T cells, we can set
<code>keep.group = "CD8"</code> or <code>keep.group = "NK-CD8"</code>.
<code>keep.top</code> can specify how many top genes in that comparison
should be kept.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## keep top 200 DEGs in 'NK-CD8'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_refine</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  keep.top <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  keep.group <span class="op">=</span> <span class="st">"CD8"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             4012         3949       3146   2698   2155</span></span>
<span><span class="co">#&gt; NotSig           1492         2694       4429   5001   6201</span></span>
<span><span class="co">#&gt; Up               4945         3806       2874   2750   2093</span></span>
<span></span>
<span><span class="co">## get 8 more new signature genes</span></span>
<span><span class="va">sig_NK_CRC_refine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">m_ct_refine</span>, <span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">sig_NK_CRC_refine</span>, <span class="va">sig_NK_CRC</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "CD160"    "ELOVL6"   "GZMB"     "IL12RB2"  "NAALADL1" "NME8"     "TTC38"   </span></span>
<span><span class="co">#&gt; [8] "XCL2"     "HAVCR2"</span></span>
<span></span>
<span><span class="co">## most NK-CD8 top DEGs are not in the markers pool</span></span>
<span><span class="va">deg_tables</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_de_table.html">get_de_table</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>, </span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             4009         3944       3146   2694   2153</span></span>
<span><span class="co">#&gt; NotSig           1476         2678       4405   4985   6183</span></span>
<span><span class="co">#&gt; Up               4926         3789       2860   2732   2075</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu">mapIds</span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">deg_tables</span><span class="op">$</span><span class="va">`NK-CD8`</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span>,</span>
<span>                 <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span><span class="op">)</span>,</span>
<span>          <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span>  <span class="co">## only 19 out of top 200 DEGs are in the pool</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt;  [1] "NCAM1"    "KLRF1"    "SH2D1B"   "LTB"      "PTGDR"    "IL2RB"   </span></span>
<span><span class="co">#&gt;  [7] "KLRC1"    "NCR1"     "RAB27B"   "CD247"    "NAALADL1" "HAVCR2"  </span></span>
<span><span class="co">#&gt; [13] "CD244"</span></span></code></pre></div>
<p>We only get 17 more new genes when keeping top 200 DEGs in ‘NK-CD8’,
that’s because majority of those genes are not in our markers pool
<code>Markers</code> and can’t pass the screening step. So it would be
better to create a large enough pool for screening in case we miss any
useful DEGs.</p>
<div class="section level3">
<h3 id="visualization-after-refinement">Visualization after refinement<a class="anchor" aria-label="anchor" href="#visualization-after-refinement"></a>
</h3>
<p>After the refinement, we have 2 signature lists now. By comparing the
2 signatures performance, we can see the difference of NK vs CD4 and CD8
T cells increased, but the difference vs monocytes and neutrophils
dropped a bit.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## gsea dotplot</span></span>
<span><span class="fu"><a href="../reference/sig_gseaplot.html">sig_gseaplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>refined <span class="op">=</span> <span class="va">sig_NK_CRC_refine</span>,</span>
<span>              pre_ref <span class="op">=</span> <span class="va">sig_NK_CRC</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned many:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span>
<span><span class="co">#&gt; preparing geneSet collections...</span></span>
<span><span class="co">#&gt; GSEA analysis...</span></span>
<span><span class="co">#&gt; leading edge analysis...</span></span>
<span><span class="co">#&gt; done...</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gsea%20refine-1.png" width="960"></p>
<p>Abundance scatter plot also shows more specific genes to NK compared
with CD4/8 T cells.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## after refine</span></span>
<span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC_refine</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature after Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot%20refine-1.png" width="768"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## before refine</span></span>
<span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>, </span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>, </span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Signature before Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot%20refine-2.png" width="768"></p>
</div>
</div>
<div class="section level2">
<h2 id="screen-on-multiple-datasets">Screen on Multiple Datasets<a class="anchor" aria-label="anchor" href="#screen-on-multiple-datasets"></a>
</h2>
<hr>
<p>Till now, the whole workflow of mastR usage is complete.</p>
<p>But this demo is just a show case on single dataset, mastR can also
be applied on multiple datasets.</p>
<p>After the Step 2d (Section @ref(intersect-pool)), for:</p>
<ul>
<li><ol style="list-style-type: decimal">
<li>single dataset<br>
Return the intersection result as the signature.<br>
</li>
</ol></li>
<li><ol start="2" style="list-style-type: decimal">
<li>multiple datasets<br>
After Step 2d, choose one combination method (<code>union</code>
default) to aggregate signature lists generated from different datasets.
Robust Rank Aggregation (“RRA”) is another method to select in mastR,
which detects genes that are ranked consistently better than expected
under null hypothesis of uncorrelated inputs and assigns a significance
score for each gene.</li>
</ol></li>
</ul>
<p>Instead of integrating datasets into one larger dataset to fit into
one linear model, we screen signature in each dataset respectively and
users can choose to aggregate the ranked lists via
<code>RobustRankAggreg</code> method by setting
<code>comb = "RRA"</code> (as our output of signature by screening
function is ordered based on the given <code>Rank</code>).</p>
<p>The advantage is that we can avoid over-normalizing or mis-correcting
when carrying out data integration. And we can get more robust and
conserved signature across datasets. It’s better than
“RemoveBatchEffect” sometimes.</p>
<p>Of course, easy way such as “union” and “intersect” is also supported
besides “RRA” (Robust Rank Aggreg), by setting <code>comb = union</code>
or <code>comb = intersect</code>.</p>
<p>It’s better to use union when you get only a few genes screened for
each dataset, while “RRA” is better to pick up significant genes from
large DEG lists, “intersect” is proper for highly overlapping gene
lists.</p>
<p>Here, we will repeatedly use im_data_6 to show how it works on
multiple datasets.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## In the demo, we just repeatedly use im_data_6 as a show case</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## we will get exactly the same list</span></span>
<span><span class="co">## if we choose 'union' or 'intersect' as combination</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setequal</a></span><span class="op">(</span><span class="va">m_ct_m</span>, <span class="va">m_ct</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## but we will only get the genes appear at top rank across gene lists</span></span>
<span><span class="co">## if we choose 'RRA', s_thres is to determine the threshold for ranking score</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>, </span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  comb <span class="op">=</span> <span class="st">"RRA"</span>,  <span class="co">## change this to use different strategy, default is "union"</span></span>
<span>  s_thres <span class="op">=</span> <span class="fl">0.5</span>  <span class="co">## only work when comb = "RRA", set a threshold for ranking score</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## we can get only 8 signature genes this time</span></span>
<span><span class="va">m_ct_m</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="multi-data">Screen on scRNA data<a class="anchor" aria-label="anchor" href="#multi-data"></a>
</h2>
<hr>
<p>Although mastR is developed for bulk RNA data, to make use of
abundant scRNA resource, mastR also provides
<code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> function to help convert scRNA data into
pseudo-bulk data, which can then be used in the above workflow.</p>
<p><code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> can help aggregate cells into
pseudo-sample according to the given factor(s).</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a test scRNA object of 100 genes x 100 cells</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>,</span>
<span>  level <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, each <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">counts</span>, by <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pb</span>, group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span><span class="va">pb</span>, group_col <span class="op">=</span> <span class="st">"group"</span>, target_group <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Seurat or SCE object are also accepted</span></span>
<span><span class="co"># scRNA &lt;- Seurat::CreateSeuratObject(counts = counts, meta.data = meta)</span></span>
<span><span class="co"># pseudo_samples(scRNA, by = c("subset","level"))</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="application-on-simulated-data">Application on Simulated
data<a class="anchor" aria-label="anchor" href="#application-on-simulated-data"></a>
</h2>
<p>After getting the signature, we can apply it in multple ways. In this
short demo, we will just show 3 simple applications of it.</p>
<p>Here we will simulate a scRNA data using <code>splatter</code>. Then
randomly pseudo-bulking it to generate pseudo bulk data for scoring and
deconvolution.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">splatter</span><span class="op">)</span></span>
<span><span class="co">## set seed for reproduce as there's permutation inside</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim_params</span> <span class="op">&lt;-</span> <span class="fu">newSplatParams</span><span class="op">(</span></span>
<span>  nGenes <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  batchCells <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  group.prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.4</span>, length.out <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>  de.prob <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  <span class="co"># de.downProb = 0,  ## only set up-regulated genes for each group</span></span>
<span>  de.facLoc <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  de.facScale <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">splatSimulate</span><span class="op">(</span><span class="va">sim_params</span>, method <span class="op">=</span> <span class="st">"groups"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">markers_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"DEFacGroup"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                       \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## aggregate into pseudo-bulk samples</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>,</span>
<span>                     by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"Group"</span><span class="op">)</span>,</span>
<span>                     min.cells <span class="op">=</span> <span class="fl">50</span>, max.cells <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">dge</span>,</span>
<span>    markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    group_col <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>    target_group <span class="op">=</span> <span class="va">x</span>,</span>
<span>    lfc <span class="op">=</span> <span class="fl">0</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## venn plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, \<span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu">ggvenn</span><span class="fu">::</span><span class="fu">ggvenn</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">sig_ls</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                                          marker <span class="op">=</span> <span class="va">markers_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                     show_percentage <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## heatmap</span></span>
<span><span class="fu"><a href="../reference/sig_heatmap.html">sig_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="fu">cpm</span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sig_ls</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"TP53"</span><span class="op">)</span><span class="op">)</span>, <span class="co">## add a real gene to pass gene check</span></span>
<span>  group_col <span class="op">=</span> <span class="va">dge</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>  show_column_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_column_slices <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_row_slices <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">## randomly generate aggregating idx</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">rand_idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## aggregate into pseudo-bulk samples based on rand_idx</span></span>
<span><span class="va">pb_r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb_r</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## append cellular composition</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">@</span><span class="va">colData</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">rand_idx</span>, <span class="va">Group</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">Group</span>, values_from <span class="op">=</span> <span class="va">count</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>rand_idx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rand_idx</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span>, <span class="va">tmp</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span> <span class="op">=</span> <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## data process</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">filterByExpr</span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="va">dge_r</span><span class="op">[</span><span class="va">keep</span>,, keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">calcNormFactors</span><span class="op">(</span><span class="va">dge_r</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="score">Score<a class="anchor" aria-label="anchor" href="#score"></a>
</h3>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rank_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/rankGenes.html" class="external-link">rankGenes</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge_r</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score based on sig_ls</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span>, <span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>,<span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                    <span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Prop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prop</span>, y <span class="op">=</span> <span class="va">Score</span>, col <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score based on markers_list</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">markers_list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span>, <span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>,<span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                    <span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Prop"</span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"DEFac"</span>, <span class="va">tmp</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prop</span>, y <span class="op">=</span> <span class="va">Score</span>, col <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="deconvolution">Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h3>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_matrix</span> <span class="op">&lt;-</span> <span class="fu">DWLS</span><span class="fu">::</span><span class="fu">buildSignatureMatrixMAST</span><span class="op">(</span></span>
<span>  scdata <span class="op">=</span> <span class="va">data_sim</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>  id <span class="op">=</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span>,</span>
<span>  path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">DWLS</span><span class="fu">::</span><span class="fu">solveDampenedWLS</span><span class="op">(</span><span class="va">sig_matrix</span>, <span class="va">pb_r</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sig_matrix</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotation">Annotation<a class="anchor" aria-label="anchor" href="#annotation"></a>
</h3>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span><span class="co">## normalization</span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu">computePooledFactors</span><span class="op">(</span><span class="va">data_sim</span>, clusters <span class="op">=</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu">logNormCounts</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span></span>
<span><span class="co">## use singscore for annotation</span></span>
<span><span class="va">rank_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/rankGenes.html" class="external-link">rankGenes</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">@</span><span class="va">assays</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score using sig_ls</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="fl">2</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">==</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score using markers_list</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">markers_list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="fl">2</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">==</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GSEABase_1.60.0      graph_1.76.0         annotate_1.76.0     </span></span>
<span><span class="co">#&gt;  [4] XML_3.99-0.13        AnnotationDbi_1.60.2 IRanges_2.32.0      </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.36.2     Biobase_2.58.0       BiocGenerics_0.44.0 </span></span>
<span><span class="co">#&gt; [10] ggplot2_3.4.1        mastR_0.9.0         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] scattermore_0.8             SeuratObject_4.1.3         </span></span>
<span><span class="co">#&gt;   [3] ragg_1.2.5                  tidyr_1.3.0                </span></span>
<span><span class="co">#&gt;   [5] bit64_4.0.5                 knitr_1.42                 </span></span>
<span><span class="co">#&gt;   [7] irlba_2.3.5.1               DelayedArray_0.24.0        </span></span>
<span><span class="co">#&gt;   [9] data.table_1.14.8           KEGGREST_1.38.0            </span></span>
<span><span class="co">#&gt;  [11] RCurl_1.98-1.10             doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [13] generics_0.1.3              cowplot_1.1.1              </span></span>
<span><span class="co">#&gt;  [15] RSQLite_2.3.0               shadowtext_0.1.2           </span></span>
<span><span class="co">#&gt;  [17] RANN_2.6.1                  future_1.32.0              </span></span>
<span><span class="co">#&gt;  [19] bit_4.0.5                   enrichplot_1.18.3          </span></span>
<span><span class="co">#&gt;  [21] spatstat.data_3.0-1         httpuv_1.6.9               </span></span>
<span><span class="co">#&gt;  [23] SummarizedExperiment_1.28.0 viridis_0.6.2              </span></span>
<span><span class="co">#&gt;  [25] xfun_0.37                   jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [27] evaluate_0.20               promises_1.2.0.1           </span></span>
<span><span class="co">#&gt;  [29] fansi_1.0.4                 igraph_1.4.1               </span></span>
<span><span class="co">#&gt;  [31] DBI_1.1.3                   htmlwidgets_1.6.1          </span></span>
<span><span class="co">#&gt;  [33] spatstat.geom_3.1-0         purrr_1.0.1                </span></span>
<span><span class="co">#&gt;  [35] ellipsis_0.3.2              dplyr_1.1.0                </span></span>
<span><span class="co">#&gt;  [37] ggpubr_0.6.0                backports_1.4.1            </span></span>
<span><span class="co">#&gt;  [39] deldir_1.0-6                MatrixGenerics_1.10.0      </span></span>
<span><span class="co">#&gt;  [41] vctrs_0.6.0                 SingleCellExperiment_1.20.0</span></span>
<span><span class="co">#&gt;  [43] ROCR_1.0-11                 abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [45] cachem_1.0.7                withr_2.5.0                </span></span>
<span><span class="co">#&gt;  [47] ggforce_0.4.1               HDO.db_0.99.1              </span></span>
<span><span class="co">#&gt;  [49] progressr_0.13.0            sctransform_0.3.5          </span></span>
<span><span class="co">#&gt;  [51] treeio_1.22.0               goftest_1.2-3              </span></span>
<span><span class="co">#&gt;  [53] cluster_2.1.4               DOSE_3.24.2                </span></span>
<span><span class="co">#&gt;  [55] ape_5.7-1                   lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [57] crayon_1.5.2                spatstat.explore_3.1-0     </span></span>
<span><span class="co">#&gt;  [59] edgeR_3.40.2                pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [61] labeling_0.4.2              tweenr_2.0.2               </span></span>
<span><span class="co">#&gt;  [63] GenomeInfoDb_1.34.9         nlme_3.1-162               </span></span>
<span><span class="co">#&gt;  [65] rlang_1.1.0                 globals_0.16.2             </span></span>
<span><span class="co">#&gt;  [67] lifecycle_1.0.3             miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [69] downloader_0.4              rprojroot_2.0.3            </span></span>
<span><span class="co">#&gt;  [71] polyclip_1.10-4             matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt;  [73] lmtest_0.9-40               Matrix_1.5-3               </span></span>
<span><span class="co">#&gt;  [75] aplot_0.1.10                carData_3.0-5              </span></span>
<span><span class="co">#&gt;  [77] singscore_1.18.0            zoo_1.8-11                 </span></span>
<span><span class="co">#&gt;  [79] ggridges_0.5.4              GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt;  [81] png_0.1-8                   viridisLite_0.4.1          </span></span>
<span><span class="co">#&gt;  [83] rjson_0.2.21                bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [85] msigdb_1.6.0                gson_0.1.0                 </span></span>
<span><span class="co">#&gt;  [87] KernSmooth_2.23-20          Biostrings_2.66.0          </span></span>
<span><span class="co">#&gt;  [89] blob_1.2.3                  shape_1.4.6                </span></span>
<span><span class="co">#&gt;  [91] stringr_1.5.0               qvalue_2.30.0              </span></span>
<span><span class="co">#&gt;  [93] parallelly_1.34.0           spatstat.random_3.1-4      </span></span>
<span><span class="co">#&gt;  [95] gridGraphics_0.5-1          rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [97] ggsignif_0.6.4              scales_1.2.1               </span></span>
<span><span class="co">#&gt;  [99] memoise_2.0.1               magrittr_2.0.3             </span></span>
<span><span class="co">#&gt; [101] plyr_1.8.8                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt; [103] zlibbioc_1.44.0             scatterpie_0.1.8           </span></span>
<span><span class="co">#&gt; [105] compiler_4.2.2              RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [107] clue_0.3-64                 fitdistrplus_1.1-8         </span></span>
<span><span class="co">#&gt; [109] cli_3.6.0                   XVector_0.38.0             </span></span>
<span><span class="co">#&gt; [111] listenv_0.9.0               patchwork_1.1.2            </span></span>
<span><span class="co">#&gt; [113] pbapply_1.7-0               MASS_7.3-58.3              </span></span>
<span><span class="co">#&gt; [115] tidyselect_1.2.0            stringi_1.7.12             </span></span>
<span><span class="co">#&gt; [117] textshaping_0.3.6           highr_0.10                 </span></span>
<span><span class="co">#&gt; [119] yaml_2.3.7                  GOSemSim_2.24.0            </span></span>
<span><span class="co">#&gt; [121] locfit_1.5-9.7              ggrepel_0.9.3              </span></span>
<span><span class="co">#&gt; [123] grid_4.2.2                  sass_0.4.5                 </span></span>
<span><span class="co">#&gt; [125] fastmatch_1.1-3             tools_4.2.2                </span></span>
<span><span class="co">#&gt; [127] future.apply_1.10.0         parallel_4.2.2             </span></span>
<span><span class="co">#&gt; [129] circlize_0.4.15             foreach_1.5.2              </span></span>
<span><span class="co">#&gt; [131] gridExtra_2.3               farver_2.1.1               </span></span>
<span><span class="co">#&gt; [133] Rtsne_0.16                  ggraph_2.1.0               </span></span>
<span><span class="co">#&gt; [135] digest_0.6.31               shiny_1.7.4                </span></span>
<span><span class="co">#&gt; [137] Rcpp_1.0.10                 GenomicRanges_1.50.2       </span></span>
<span><span class="co">#&gt; [139] car_3.1-1                   broom_1.0.4                </span></span>
<span><span class="co">#&gt; [141] later_1.3.0                 RcppAnnoy_0.0.20           </span></span>
<span><span class="co">#&gt; [143] org.Hs.eg.db_3.16.0         httr_1.4.5                 </span></span>
<span><span class="co">#&gt; [145] ComplexHeatmap_2.14.0       colorspace_2.1-0           </span></span>
<span><span class="co">#&gt; [147] fs_1.6.1                    tensor_1.5                 </span></span>
<span><span class="co">#&gt; [149] reticulate_1.28             splines_4.2.2              </span></span>
<span><span class="co">#&gt; [151] yulab.utils_0.0.6           uwot_0.1.14                </span></span>
<span><span class="co">#&gt; [153] tidytree_0.4.2              spatstat.utils_3.0-2       </span></span>
<span><span class="co">#&gt; [155] pkgdown_2.0.7               graphlayouts_0.8.4         </span></span>
<span><span class="co">#&gt; [157] sp_1.6-0                    ggplotify_0.1.0            </span></span>
<span><span class="co">#&gt; [159] plotly_4.10.1               systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt; [161] xtable_1.8-4                ggtree_3.6.2               </span></span>
<span><span class="co">#&gt; [163] jsonlite_1.8.4              tidygraph_1.2.3            </span></span>
<span><span class="co">#&gt; [165] UpSetR_1.4.0                ggfun_0.0.9                </span></span>
<span><span class="co">#&gt; [167] R6_2.5.1                    pillar_1.8.1               </span></span>
<span><span class="co">#&gt; [169] htmltools_0.5.4             mime_0.12                  </span></span>
<span><span class="co">#&gt; [171] glue_1.6.2                  fastmap_1.1.1              </span></span>
<span><span class="co">#&gt; [173] clusterProfiler_4.6.2       BiocParallel_1.32.5        </span></span>
<span><span class="co">#&gt; [175] codetools_0.2-19            fgsea_1.24.0               </span></span>
<span><span class="co">#&gt; [177] utf8_1.2.3                  lattice_0.20-45            </span></span>
<span><span class="co">#&gt; [179] bslib_0.4.2                 spatstat.sparse_3.0-1      </span></span>
<span><span class="co">#&gt; [181] tibble_3.2.0                leiden_0.4.3               </span></span>
<span><span class="co">#&gt; [183] GO.db_3.16.0                survival_3.5-5             </span></span>
<span><span class="co">#&gt; [185] limma_3.54.2                rmarkdown_2.20             </span></span>
<span><span class="co">#&gt; [187] desc_1.4.2                  munsell_0.5.0              </span></span>
<span><span class="co">#&gt; [189] GetoptLong_1.0.5            GenomeInfoDbData_1.2.9     </span></span>
<span><span class="co">#&gt; [191] iterators_1.0.14            reshape2_1.4.4             </span></span>
<span><span class="co">#&gt; [193] gtable_0.3.1                Seurat_4.3.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<hr>
<blockquote>
<p>Cursons J, Souza-Fonseca-Guimaraes F, Foroutan M, Anderson A,
Hollande F, Hediyeh-Zadeh S, Behren A, Huntington ND, Davis MJ. A Gene
Signature Predicting Natural Killer Cell Infiltration and Improved
Survival in Melanoma Patients. Cancer Immunol Res. 2019
Jul;7(7):1162-1174. doi: 10.1158/2326-6066.CIR-18-0500. Epub 2019 May
14. PMID: 31088844.</p>
</blockquote>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jinjin Chen.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
