<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mastR: Markers Automated Screening Tool in R • mastR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mastR: Markers Automated Screening Tool in R">
<meta property="og:description" content="mastR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mastR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/mastR_Demo.html">mastR: Markers Automated Screening Tool in R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/DavisLaboratory/mastR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mastR: Markers Automated Screening Tool in
R</h1>
                        <h4 data-toc-skip class="author">Jinjin
Chen</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052, AustraliaDepartment of Medical
Biology, University of Melbourne, Parkville, VIC 3010,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:chen.j@wehi.edu.au" class="email">chen.j@wehi.edu.au</a>
      </address>
                              <h4 data-toc-skip class="author">Ahmed
Mohamed</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:mohamed.a@wehi.edu.au" class="email">mohamed.a@wehi.edu.au</a>
      </address>
                              <h4 data-toc-skip class="author">Chin Wee
Tan</h4>
            <address class="author_afil">
      Bioinformatics Division, Walter and Eliza Hall Institute of
Medical Research, Parkville, VIC 3052,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:cwtan@wehi.edu.au" class="email">cwtan@wehi.edu.au</a>
      </address>
                  
            <h4 data-toc-skip class="date">11 Apr 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/mastR/blob/HEAD/vignettes/mastR_Demo.Rmd" class="external-link"><code>vignettes/mastR_Demo.Rmd</code></a></small>
      <div class="hidden name"><code>mastR_Demo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr>
<p><strong>Why do we need group markers</strong></p>
<p>In biological and clinical research, the identification of biomarkers
specific to a disease, tissue, or cell type is a critical step in
advancing our understanding and application of this knowledge.</p>
<p>Specific biomarkers can be identified through various methods, such
as transcriptomics, proteomics, or metabolomics, which can provide a
global view of the molecular landscape of the system being studied.</p>
<p>To denote these biomarkers, we use the term “group markers” in a
generic sense, referring to any molecular feature that is specific to a
particular group or condition of interest.</p>
<p>Such markers can provide insight into disease diagnosis, prognosis,
and treatment options, and can help to differentiate between diseased
and healthy states. Thus the identification of the group markers is
crucial in various biological and medical applications, as it allows us
to distinguish between different cell types or disease states.</p>
<p>For example, in cancer research, identifying marker genes that are
differentially expressed between cancer cells and normal cells can help
diagnose and monitor the progression of the disease, as well as identify
potential therapeutic targets. Similarly, in developmental biology,
identifying marker genes that are specific to certain cell types or
stages can help us understand the underlying mechanisms of
differentiation and development.</p>
<p>Furthermore, marker genes can also be used in diagnostic assays to
detect specific diseases or monitor treatment responses. For instance,
the presence of certain marker genes in a patient’s blood or tissue
sample can indicate the presence or progression of a disease.</p>
<p><strong>Importance of immune cells signature in tumor
micro-environment (TME)</strong></p>
<p>In this demo, we will be using the example of immune cell signature
in the context of the TME as a practical case study to showcase the
application of our approach.</p>
<p>TME is made up of a diverse range of cell types (fibroblasts,
epithelial cells, endothelial cells, and immune cells) as well as
various extracellular components (collagens, growth factors, hormones,
cytokines, etc.). Tumor immune micro-environment (TIME) is reported to
be highly associated with prognosis and various treatment response to
many kinds of cancers.</p>
<p>Recent studies have highlighted the role of immune components in the
TME in modulating tumor progression, making them attractive therapeutic
targets. These components make up TIME, which is a subset of the TME.
Subsequently, it’s proved that tumor infiltrating lymphocytes (TILs)
play an essential role in tumor progression, metastasis and treatment
response.</p>
<p>This drives TILs to be a strong prognostic indicator for better
precision therapy of cancer patients. The identification of these TILs
are the subject of extensive research interest due to the roles of
specific subset of immune cells acting on different tissues types.</p>
<p>Identification of these cell types are based on flow cytometry in the
past, which has limited granularity. But now with the advance of single
cell RNA-seq and spatial transcriptomics technologies, more detailed and
novel cell types or subtypes can be identified. One of the key
advantages of scRNA-seq and spatial technologies is the ability to
investigate cell types in the TIME and understand cellular heterogeneity
in the TME.</p>
<p>To quantify TILs infiltration (in bulk), estimate cellular
composition (in bulk), annotate single cell types (in scRNA) or identify
sample states/activities, many computational methods like cell
deconvolution (CIBERSORTx), marker based annotation (CelliD) and sample
scoring (singscore) are developed. But to distinguish between closely
related cell types, estimate cell composition and states, a refined
signature is required which typically requires manual curation by domain
expert. With the increasing large amount of data and cell types, this is
gradually getting to be untenable and will require a different approach
to automatically screen such signatures.</p>
<p>mastR is a software package specifically designed to automatically
screen signatures of interest for particular research questions. This
automated process saves significant time and effort that would otherwise
be required for manual labor.</p>
<p><strong>What this package does</strong></p>
<p><strong>mastR</strong>, <strong>M</strong>arkers
<strong>A</strong>utomated <strong>S</strong>creening
<strong>T</strong>ool in <strong>R</strong>, is a package to
automatically derive signature for specific group of interest in
specific tissue.</p>
<p>With mastR, users can simply input their expression data containing
the groups of interest, along with group labels, to obtain a list of
marker genes (signature) for the target group.</p>
<p>While there are many tools available for generating cell
type-specific markers, they are primarily designed for scRNA-seq data
and often rely on machine learning algorithms to select relevant
features for classification. However, these methods may lack robustness
and consistency across datasets when compared to using statistical
methods like empirical Bayesian in <code>limma</code>. Furthermore, some
of these tools may return a signature even when the data does not
contain any, leading to potential inaccuracies.</p>
<p>Although differential expression (DE) analysis can also be done on
scRNA data, like <code><a href="https://satijalab.org/seurat/reference/FindMarkers.html" class="external-link">Seurat::FindMarkers()</a></code>, it’s reported that
DE analysis done on pseudo-bulk scRNA data is more robust and reliable
than directly done on scRNA data.</p>
<p>Thus, mastR is designed to generate a more refined list of signature
genes from multiple group comparisons based on the results from
<code>edgeR</code> and <code>limma</code> DE analysis workflow. The
final signature is selected by rank product score test for picking up
genes with high ranks in the most of comparisons. The rank can be
ordered by any gene statistic generated by limma analysis. Signature can
be further refined by keeping the top n DEGs in the specified
comparison(s), which can help to improve the discrimination between
fairly similar cell types.</p>
<p>Another unique advantage of mastR is that it takes into account the
background expression of tissue-specificity, which is often ignored by
other marker generation tools. Unlike most tools that only consider the
genes’ contribution to the classification, mastR also offers a
background expression removal function. This function is designed to
remove parts of the marker genes that are highly expressed in specific
tissues or cancer cells. These signals from the background, regarded as
background expression, can cause potential ambiguity when applying the
markers to specific tissues due to background or sample purity issues.
This feature makes mastR apart and enhances the accuracy and specificity
of the generated markers.</p>
<p>Furthermore, mastR allows users to build a markers pool before
signature screening, which might contain the potential markers of
interest to the users. The final signature will be integrated with this
pool (by intersection), which can help to constrain the final signature
within the interested pathway-related genes or functional gene-sets.
People can borrow the published knowledge to build this.</p>
<p>The motivation of this package arises from the importance and
necessity of identifying specific genes that are differentially
expressed in different groups or tissues, as these genes can serve as
biomarkers for diagnosis, prognosis, and therapeutic targeting. However,
identifying marker genes can be a challenging and time-consuming task,
and the presence of background expression can lead to erroneous results.
Our package simplifies and streamlines this process, allowing
researchers to focus on their analyses and interpretations without the
burden of manual marker gene selection and background expression
removal.</p>
<p>This report demonstrates the main functions and applications of mastR
0.99.4.</p>
<p>This report demonstrates the signature screening workflow of NK cells
in colorectal cancer (CRC), assessing the results by using in-built
visualization functions.</p>
<p><strong>mastR screen the signature using the following 3 key
steps:</strong></p>
<p>step 1. build markers pool<br>
step 2. identify signature from the pool<br>
step 3. refine signature by background expression<br>
step 4. visualize signature performance</p>
<p><strong>Applications</strong></p>
<ul>
<li>score samples</li>
<li>estimate cellular proportion</li>
<li>single cell annotation</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<hr>
<p><code>mastR</code> R package can be installed from Bioconductor or <a href="https://github.com/Gene233/mastR" class="external-link">GitHub</a>.</p>
<p>The most updated version of <code>mastR</code> is hosted on GitHub
and can be installed using <code><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">devtools::install_github()</a></code>
function provided by <a href="https://cran.r-project.org/package=devtools" class="external-link">devtools</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!requireNamespace("devtools", quietly = TRUE)) {</span></span>
<span><span class="co">#   install.packages("devtools")</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># if (!requireNamespace("mastR", quietly = TRUE)) {</span></span>
<span><span class="co">#   devtools::install_github("Gene233/mastR")</span></span>
<span><span class="co"># }</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"mastR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"mastR"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/mastR" class="external-link">mastR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.3.0</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'GSEABase' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'BiocGenerics' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'Biobase' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'annotate' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'AnnotationDbi' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'IRanges' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'S4Vectors' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'XML' was built under R version 4.3.0</span></span>
<span><span class="co">#&gt; Warning: package 'graph' was built under R version 4.3.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1--build-markers-pool">Step 1. Build Markers Pool<a class="anchor" aria-label="anchor" href="#step-1--build-markers-pool"></a>
</h2>
<hr>
<p>The first step is to define the original markers pool this analysis
will be based on.</p>
<p>The final signature will only be the intersected genes with this
markers pool. The whole gene list of the data will be regarded as the
markers pool if no preliminary result or interested genes are
provided.</p>
<p><strong><em>Note:</em></strong> <code>markers = NULL</code> won’t
keep any special genes if they fail the filtration by edgeR.</p>
<p>If users have any preliminary knowledge about the target group type
(cell type), they can build the markers pool of interest from the
available datasets or build from MSigDB, PanglaoDB or LM7/LM22 using our
in-built functions. All genes in the pool will be reserved for DE
analysis even even if they are filtered out during the filtration of
edgeR.</p>
<p>The standard pool building process involves the following:</p>
<ol style="list-style-type: decimal">
<li>generate from sources
<ol style="list-style-type: lower-roman">
<li>LM7/22 signature matrix for CIBERSORT<br>
</li>
<li>MSigDB<br>
</li>
<li>PanglaoDB<br>
</li>
<li>Customized gene list<br>
</li>
</ol>
</li>
<li>merge gene-sets together</li>
</ol>
<p>mastR allows the following markers to be conveniently loaded:</p>
<ul>
<li><p><code>lm7</code> and <code>lm22</code>: CIBERSORT (LM22) and
Tosolini et al. (LM7) gene sets for leukocytes, get more details using
<code><a href="../reference/lm7.html">?mastR::lm7</a></code> or <code><a href="../reference/lm22.html">?mastR::lm22</a></code>.</p></li>
<li><p><code>msigdb_gobp_nk</code>: a ‘GeneSetCollection’ object,
containing gene sets with gene-set name matched to ‘NATURAL_KILLER’ in
<a href="GO:BP" class="uri">GO:BP</a> MSigDB v7.4 database. More details
<code><a href="../reference/msigdb_gobp_nk.html">?mastR::msigdb_gobp_nk</a></code>.</p></li>
<li><p><code>nk_markers</code>: a combination of CIBERSORT LM7, LM22 and
Huntington gene list with 114 genes. (<a href="https://cancerimmunolres.aacrjournals.org/content/7/7/1162.long" class="external-link">Cursons
et al. 2019</a>).</p></li>
</ul>
<p>All markers generation functions will return ‘GeneSet’ or
‘GeneSetCollection’ object.</p>
<div class="section level3">
<h3 id="generate-markers-from-sources">Generate Markers from Sources<a class="anchor" aria-label="anchor" href="#generate-markers-from-sources"></a>
</h3>
<p>To generate the immune cell signatures, firstly users need to define
a pool of markers of interest. mastR allows users to build markers pool
from multiple resources.</p>
<p>In this demo, we will load some some publicly available example
datasets.</p>
<div class="section level4">
<h4 id="i-leukocyte-gene-signature-matrix-lm">i) Leukocyte gene signature Matrix (LM)<a class="anchor" aria-label="anchor" href="#i-leukocyte-gene-signature-matrix-lm"></a>
</h4>
<p><code>lm7/lm22</code> are immune cells signature matrices from
CIBERSORT, contains 7 or 22 immune cell types.</p>
<p>Users can use function <code><a href="../reference/get_lm_sig.html">get_lm_sig()</a></code> to generate immune
cells markers from LM7 and/or lm22.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"lm7"</span>, <span class="st">"lm22"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## collect both LM7 and LM22</span></span>
<span><span class="va">LM</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_lm_sig.html">get_lm_sig</a></span><span class="op">(</span>lm7.pattern <span class="op">=</span> <span class="st">"^NK"</span>, lm22.pattern <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span><span class="va">LM</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: LM7, LM22 (2 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: CD244, FASLG, ..., ZNF135 (92 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
<p>Function <code><a href="../reference/gsc_plot.html">gsc_plot()</a></code> allows visualization of an UpSetR
plot across all gene-sets.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show upset diagram</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">LM</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gsc%20plot-1.png" width="768"></p>
</div>
<div class="section level4">
<h4 id="ii-msigdb">ii) MSigDB<a class="anchor" aria-label="anchor" href="#ii-msigdb"></a>
</h4>
<p>The Molecular Signatures Database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/" class="external-link"><strong>MSigDB</strong></a>)
is a database of annotated gene sets, typically used for pathway
analysis.</p>
<p>Users can use <code><a href="../reference/get_gsc_sig.html">get_gsc_sig()</a></code> to generate a collection of
gene sets for the biological subjects of interest.</p>
<p>In this case, NK cell relevant gene-sets from
<code>msigdb_gobp_nk</code> are collected.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"msigdb_gobp_nk"</span><span class="op">)</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gsc_sig.html">get_gsc_sig</a></span><span class="op">(</span></span>
<span>  gsc <span class="op">=</span> <span class="va">msigdb_gobp_nk</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"NATURAL_KILLER_CELL_MEDIATED"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">MSig</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY, GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY_DIRECTED_AGAINST_TUMOR_CELL_TARGET, ..., GOBP_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY (18 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: AP1G1, ARRB2, ..., KLRC4-KLRK1 (67 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: SymbolIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: BroadCollection (1 total)</span></span></code></pre></div>
<p><strong><em>Note</em></strong>: <code>data = "msigdb"</code> allows
searching of MsigDB without loading into the environment.
<code>species</code> or <code>version</code> are optional
parameters.</p>
<p>Similarly, using <code><a href="../reference/gsc_plot.html">gsc_plot()</a></code> users can visualize the
overlapping gene sets.</p>
<p>As the gene-set names are too long, for better visualization the
gene-set names are replaced with letters and shown below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## cut gene set name within 11 characters</span></span>
<span><span class="va">gsn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span>, <span class="va">LETTERS</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">setName</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="va">i</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co">## show upset diagram of collected gene-sets</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/show%20overlap-1.png" width="768"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsn</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"M"</span>, <span class="st">"D"</span><span class="op">)</span><span class="op">]</span> <span class="co">## show gene-set names of top 3</span></span>
<span><span class="co">#&gt;                                                          A </span></span>
<span><span class="co">#&gt;               "GOBP_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY" </span></span>
<span><span class="co">#&gt;                                                          M </span></span>
<span><span class="co">#&gt;           "GOBP_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY" </span></span>
<span><span class="co">#&gt;                                                          D </span></span>
<span><span class="co">#&gt; "GOBP_REGULATION_OF_NATURAL_KILLER_CELL_MEDIATED_IMMUNITY"</span></span></code></pre></div>
<p>There are 18 gene-sets in MSig, which is too many for visualization.
Thus, we use function <code><a href="../reference/merge_markers.html">merge_markers()</a></code> to merge all
gene-sets into one ‘GeneSet’ object. (<em>Note</em>: Users can directly
use the un-merged object for subsequent analyses.)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## merge all gene sets into one</span></span>
<span><span class="va">MSig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">setName</a></span><span class="op">(</span><span class="va">MSig</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"MSigDB"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="iii-panglaodb">iii) PanglaoDB<a class="anchor" aria-label="anchor" href="#iii-panglaodb"></a>
</h4>
<p><a href="https://panglaodb.se/" class="external-link">PanglaoDB</a> is a database of single
cell RNA sequencing experiments with cell type markers.</p>
<p>Users can use <code><a href="../reference/get_panglao_sig.html">get_panglao_sig()</a></code> to generate markers
based on the required organs and cell types.</p>
<p>Functions <code><a href="../reference/list_panglao_organs.html">list_panglao_organs()</a></code> and
<code><a href="../reference/list_panglao_types.html">list_panglao_types()</a></code> show all available organs and cell
types in PanglaoDB.</p>
<p><strong><em>Note:</em></strong> This requires real-time connection to
PanglaoDB, thus not run in this demo.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show available organs on PanglaoDB</span></span>
<span><span class="fu"><a href="../reference/list_panglao_organs.html">list_panglao_organs</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## show available cell types of interest organ on PanglaoDB</span></span>
<span><span class="fu"><a href="../reference/list_panglao_types.html">list_panglao_types</a></span><span class="op">(</span>organ <span class="op">=</span> <span class="st">"Immune system"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## collect all "NK cells" markers from PanglaoDB website</span></span>
<span><span class="va">Panglao</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_panglao_sig.html">get_panglao_sig</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"NK cells"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Panglao</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="iv-customized-gene-list">iv) Customized gene list<a class="anchor" aria-label="anchor" href="#iv-customized-gene-list"></a>
</h4>
<p>Customized markers can be imported as a ‘GeneSet’ object.</p>
<p>Here we demonstrate this by loading the in-built
<code>nk_markers</code> markers dataset and convert it into ‘GeneSet’
object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## show what nk_markers looks like:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"nk_markers"</span><span class="op">)</span></span>
<span><span class="va">nk_markers</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 114 × 4</span></span></span>
<span><span class="co">#&gt;    HGNC_Symbol LM22  LM7   Huntington</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> APOBEC3G    TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> APOL6       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> AZU1        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> BPI         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CAMP        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CCL4        TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CCL5        TRUE  -     TRUE      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CCND2       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CD160       TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CD2         TRUE  -     -         </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 104 more rows</span></span></span>
<span></span>
<span><span class="co">## convert NK markers into 'GeneSet' object</span></span>
<span><span class="va">nk_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-methods.html" class="external-link">GeneSet</a></span><span class="op">(</span><span class="va">nk_markers</span><span class="op">$</span><span class="va">HGNC_Symbol</span>,</span>
<span>  geneIdType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneIdentifierType-constructors.html" class="external-link">SymbolIdentifier</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  setName <span class="op">=</span> <span class="st">"NK_markers"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="pool-markers-from-sources">Pool Markers from Sources<a class="anchor" aria-label="anchor" href="#pool-markers-from-sources"></a>
</h3>
<p>With multiple lists of markers from different sources, use
<code><a href="../reference/merge_markers.html">merge_markers()</a></code> to merge them into one ‘GeneSet’
object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSetCollection-methods.html" class="external-link">GeneSetCollection</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">nk_m</span>, <span class="va">LM</span>, <span class="va">MSig</span><span class="op">)</span><span class="op">)</span> <span class="co">## add Panglao if you run it</span></span>
<span><span class="va">Markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_markers.html">merge_markers</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## upset plot</span></span>
<span><span class="fu"><a href="../reference/gsc_plot.html">gsc_plot</a></span><span class="op">(</span><span class="va">gsc</span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/integrate%20markers-1.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Markers</span></span>
<span><span class="co">#&gt; setName: merged_markers_pool </span></span>
<span><span class="co">#&gt; geneIds: APOBEC3G, APOL6, ..., KLRC4-KLRK1 (total: 167)</span></span>
<span><span class="co">#&gt; geneIdType: Symbol</span></span>
<span><span class="co">#&gt; collectionType: Computed </span></span>
<span><span class="co">#&gt; details: use 'details(object)'</span></span></code></pre></div>
<p>The summary of the merged list is saved as ‘longDescription’ in the
output.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## to show the table summary of merged list</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/jsonlite/man/fromJSON.html" class="external-link">fromJSON</a></span><span class="op">(</span><span class="fu">GSEABase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">longDescription</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       Gene NK_markers LM7 LM22 MSigDB</span></span>
<span><span class="co">#&gt; 1    AP1G1          -   -    -   TRUE</span></span>
<span><span class="co">#&gt; 2 APOBEC3G       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 3    APOL6       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 4    ARRB2          -   -    -   TRUE</span></span>
<span><span class="co">#&gt; 5     AZU1       TRUE   - TRUE      -</span></span>
<span><span class="co">#&gt; 6      BPI       TRUE   - TRUE      -</span></span></code></pre></div>
<p>Now with a merged markers pool, we refine the signature genes for
group specificity and tissue background removal.</p>
</div>
</div>
<div class="section level2">
<h2 id="step-2--signature-identification-for-target-group">Step 2. Signature Identification for Target Group<a class="anchor" aria-label="anchor" href="#step-2--signature-identification-for-target-group"></a>
</h2>
<hr>
<p>For group specificity, there are 3 main steps:</p>
<ol style="list-style-type: lower-alpha">
<li>differential expression (DE) analysis: filtration, normalization,
sample weighting and linear model fit;<br>
</li>
<li>feature selection: select differentially expressed genes based on
rank product score;<br>
</li>
<li>constrain selected genes within the markers pool.</li>
</ol>
<p><strong><em>Note</em></strong>: External data is accepted in
different formats (e.g. DGEList, eSet, matrix). Input data must be raw
counts or log-transformed expression data.</p>
<p>In this demo, we use the imported example data <code>im_data_6</code>
from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60424" class="external-link">GSE60424</a>
(Download using <code>GEOquery::getGEO()</code>), consisting of immune
cells from healthy individuals.</p>
<p><code>im_data_6</code> is a eSet object, containing RNA-seq TMM
normalized counts data of 6 sorted immune cell types each with 4
samples. More details in <code><a href="../reference/im_data_6.html">?mastR::im_data_6</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"im_data_6"</span><span class="op">)</span></span>
<span><span class="va">im_data_6</span></span>
<span><span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">#&gt; assayData: 50045 features, 24 samples </span></span>
<span><span class="co">#&gt;   element names: exprs </span></span>
<span><span class="co">#&gt; protocolData: none</span></span>
<span><span class="co">#&gt; phenoData</span></span>
<span><span class="co">#&gt;   sampleNames: GSM1479438 GSM1479439 ... GSM1479525 (24 total)</span></span>
<span><span class="co">#&gt;   varLabels: title geo_accession ... years since diagnosis:ch1 (66</span></span>
<span><span class="co">#&gt;     total)</span></span>
<span><span class="co">#&gt;   varMetadata: labelDescription</span></span>
<span><span class="co">#&gt; featureData: none</span></span>
<span><span class="co">#&gt; experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">#&gt;   pubMedIds: 25314013 </span></span>
<span><span class="co">#&gt; Annotation: GPL15456</span></span></code></pre></div>
<div class="section level3">
<h3 id="process">a) data processing<a class="anchor" aria-label="anchor" href="#process"></a>
</h3>
<p>To screen signature for the group specificity, the data needs to be
pre-processed.</p>
<p><code><a href="../reference/process_data.html">process_data()</a></code> in mastR does the End-to-End differential
expression analysis using <code>edgeR</code> and <code>limma</code>
pipeline as a single function call.</p>
<p>Processes under the hood:</p>
<ul>
<li><p>Filter data by the given cutoff, genes with low expression will
be removed by edgeR.</p></li>
<li><p>If data is raw counts (<code>normalize = TRUE</code>), normalize
data by ‘TMM’ and fit it using <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code>. Otherwise
apply <code>trend</code> of limma on normalized data.</p></li>
<li><p>Fit linear model.</p></li>
<li><p>Compute gene statistic for differential expression analysis using
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code>.</p></li>
</ul>
<p><code>im_data_6</code> has 6 immune cell types:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">im_data_6</span><span class="op">$</span><span class="va">`celltype:ch1`</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     B-cells         CD4         CD8   Monocytes Neutrophils          NK </span></span>
<span><span class="co">#&gt;           4           4           4           4           4           4</span></span></code></pre></div>
<p><code><a href="../reference/process_data.html">process_data()</a></code> requires an expression matrix and group
labels of the samples. This returns a <code>DGEList</code> object with
processed data.</p>
<p>To keep consistent with <code>Markers</code>, use param
<code>gene_id</code> to convert ENSEMBL IDs of <code>im_data_6</code> to
SYMBOLs for intersection.</p>
<p>Refer to <code>help(proc_data)</code> for optional parameters.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">proc_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/process_data.html">process_data</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">im_data_6</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span> <span class="co">## rownames of im_data_6 is ENSEMBL ID</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt;        NK-Neutrophils NK-Monocytes NK-B.cells NK-CD4 NK-CD8</span></span>
<span><span class="co">#&gt; Down             4012         3949       3146   2698   2155</span></span>
<span><span class="co">#&gt; NotSig           1492         2694       4429   5001   6201</span></span>
<span><span class="co">#&gt; Up               4945         3806       2874   2750   2093</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">proc_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; $names</span></span>
<span><span class="co">#&gt; [1] "counts"          "samples"         "original_counts" "vfit"           </span></span>
<span><span class="co">#&gt; [5] "tfit"           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $class</span></span>
<span><span class="co">#&gt; [1] "DGEList"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "edgeR"</span></span></code></pre></div>
<p>For ease of process later in this demo, add the expression matrix ‘E’
fitted by <code><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">limma::voom()</a></code> as ‘voomE’ into
<code>proc_data</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## add voom fitted expression as a new list of proc_data for use</span></span>
<span><span class="va">proc_data</span><span class="op">$</span><span class="va">voomE</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">vfit</span><span class="op">$</span><span class="va">E</span></span></code></pre></div>
<p><strong><em>Note:</em></strong></p>
<p>mastR provides visualization functions to compare the data before and
after <code><a href="../reference/process_data.html">process_data()</a></code> and assess the quality control (QC) by
using <code><a href="../reference/plot_diagnostics.html">plot_diagnostics()</a></code> and
<code><a href="../reference/plot_mean_var.html">plot_mean_var()</a></code>.</p>
<p>To assess the removal of the low quality genes, use
<code><a href="../reference/plot_diagnostics.html">plot_diagnostics()</a></code> to show the expression distribution, RLE
and MDS plots.</p>
</div>
<div class="section level3">
<h3 id="selection">b) signature selection based on differential expression<a class="anchor" aria-label="anchor" href="#selection"></a>
</h3>
<p>For selection of group specific signature, pass
<code>proc_data</code> to <code><a href="../reference/select_sig.html">select_sig()</a></code> function. Genes with
high rank product score in DE results are selected.</p>
<p>mastR automatically determines if feature selection is required, the
default approach (<code>feature_selection = "auto"</code>) performs rank
product scoring to select genes. But if the numbers of the result are
&lt; 5, no feature selection will be conducted (switch to
<code>feature_selection = "none"</code>).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## get the same result as there's permutation test for rank product</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_sig.html">select_sig</a></span><span class="op">(</span><span class="va">proc_data</span>, feature_selection <span class="op">=</span> <span class="st">"auto"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sig_ct</span><span class="op">)</span></span>
<span><span class="co">#&gt; GeneSetCollection</span></span>
<span><span class="co">#&gt;   names: UP, DOWN (2 total)</span></span>
<span><span class="co">#&gt;   unique identifiers: ENSG00000149294, ENSG00000173068, ..., ENSG00000162591 (98 total)</span></span>
<span><span class="co">#&gt;   types in collection:</span></span>
<span><span class="co">#&gt;     geneIdType: NullIdentifier (1 total)</span></span>
<span><span class="co">#&gt;     collectionType: NullCollection (1 total)</span></span></code></pre></div>
<p><strong><em>Note:</em></strong></p>
<p>mastR also implements a feature to further optimize the
discriminative power of the signature between fairly similar groups by
using params <code>keep.top</code> and <code>keep.group</code>. More
details in <code><a href="../reference/select_sig.html">help(select_sig)</a></code>.</p>
<p><strong><em>Tips:</em></strong></p>
<p>All above steps from @ref(process) to @ref(selection) for refining
group signature can be done using an integrated function
<code><a href="../reference/get_degs.html">get_degs()</a></code>.</p>
<p>It will return a list of a processed data <code>proc_data</code> and
a ‘GeneSetCollection’ <code>DEGs</code> with ‘UP’ and ‘DOWN’ regulated
genes.</p>
<p>QC plots can also be shown by setting optional param
<code>plot = TRUE</code>.</p>
</div>
<div class="section level3">
<h3 id="intersect-pool">c) constrain signature within markers pool<a class="anchor" aria-label="anchor" href="#intersect-pool"></a>
</h3>
<p>To constrain the final signature within the interested gene list,
intersect the signature genes <code>sig_ct</code> with the markers pool
<code>Markers</code>. In this demo, only the ‘UP’ regulated genes are
kept.</p>
<p>For consistency with <code>Markers</code> type, convert ENSEMBL IDs
of <code>im_data_6</code> to gene SYMBOLs.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## convert ensembl IDs into symbols to match markers pool</span></span>
<span><span class="va">deg_up</span> <span class="op">&lt;-</span> <span class="fu">mapIds</span><span class="op">(</span></span>
<span>  <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">sig_ct</span><span class="op">[[</span><span class="st">"UP"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  <span class="st">"SYMBOL"</span>, <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="va">deg_up</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span></span>
<span><span class="co">## markers specific for NK cells</span></span>
<span><span class="va">m_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>, <span class="va">deg_up</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">deg_up</span><span class="op">)</span><span class="op">]</span> <span class="co">## set ensembl ID as names for downstream visualization</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m_ct</span><span class="op">)</span></span>
<span><span class="co">#&gt; ENSG00000122223 ENSG00000198821 ENSG00000172543 ENSG00000145649 ENSG00000197540 </span></span>
<span><span class="co">#&gt;         "CD244"         "CD247"          "CTSW"          "GZMA"          "GZMM" </span></span>
<span><span class="co">#&gt; ENSG00000100385 </span></span>
<span><span class="co">#&gt;         "IL2RB"</span></span></code></pre></div>
<p>Users can use <code><a href="../reference/pca_matrix_plot.html">pca_matrix_plot()</a></code> to assess the group
separation.</p>
<p>Looking at the variance of PC1 shown below, it’s clear the
intersected genes explain more variance in PC1 compared with all ‘UP’
DEGs.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## PCA shows clear separation of NK cells</span></span>
<span><span class="co">## after intersection</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span>,</span>
<span>  features <span class="op">=</span> <span class="va">m_ct</span>,</span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html" class="external-link">plot_annotation</a></span><span class="op">(</span><span class="st">"Intersected UP DEGs"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/PCA%20on%20sig-1.png" width="768"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## before intersection</span></span>
<span><span class="fu"><a href="../reference/pca_matrix_plot.html">pca_matrix_plot</a></span><span class="op">(</span><span class="va">proc_data</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">deg_up</span><span class="op">)</span>,</span>
<span>  group_by <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html" class="external-link">plot_annotation</a></span><span class="op">(</span><span class="st">"All UP DEGs"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/PCA%20on%20sig-2.png" width="768"></p>
<p><strong><em>Note</em></strong>:</p>
<p>Wrapper function <code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> can complete all
processes in Step 2 using a single function. The result would be the
same.</p>
<p><code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> is helpful when users have multiple
datasets (more details in Section @ref(multi-data)). It can output the
final signature after aggregating signature lists from all datasets.</p>
</div>
</div>
<div class="section level2">
<h2 id="step-3--signature-refinement-by-background-expression-in-tissue">Step 3. Signature Refinement by Background Expression in Tissue<a class="anchor" aria-label="anchor" href="#step-3--signature-refinement-by-background-expression-in-tissue"></a>
</h2>
<hr>
<p>Further refinement can be achieved by eliminating genes with strong
signal in the cancer cells or tissues, regarded as background
expression.</p>
<p>mastR utilizes a signal-to-noise ratio (SNR) approach to filter out
genes with low SNR values that have low discriminative power between the
group of interest and the background. This removes tumor purity as a
factor in the identified signature markers.</p>
<p>Here the signal is the expression of the signature in signal data and
the noise is the expression of the signature in background data.</p>
<p>Two input datasets are required:</p>
<ul>
<li>signal data (<code>proc_data</code> from Step 2)</li>
<li>background data (target context)</li>
</ul>
<p><strong><em>Note</em></strong>: Both of signal data and background
data must be log-normalized.</p>
<p>The signal data should be the result generated by
<code><a href="../reference/process_data.html">process_data()</a></code> function.</p>
<p>The refinement consists of:</p>
<ol style="list-style-type: upper-roman">
<li>data subset</li>
<li>data filtration</li>
<li>markers removal</li>
<li>combination with the signature</li>
</ol>
<div class="section level3">
<h3 id="i-data-subsetting">I) data subsetting<a class="anchor" aria-label="anchor" href="#i-data-subsetting"></a>
</h3>
<p>To remove genes with background expression, subset the samples based
on the context. In this demo, we are looking to remove CRC-specific
signal.</p>
<p>Here we choose to load in-built <code>ccle_crc_5</code> as the
background data. The CRC adherent cell lines data is extracted
(<code>bg_mat</code>).</p>
<p>The signal data is in ‘voomE’ of <code>proc_data</code> and NK cells
data is extracted (<code>sig_mat</code>).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ccle_crc_5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## subset CRC cell lines of bg data</span></span>
<span><span class="va">bg_mat</span> <span class="op">&lt;-</span> <span class="va">ccle_crc_5</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span>, <span class="va">ccle_crc_5</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">cancer</span> <span class="op">==</span> <span class="st">"CRC"</span><span class="op">]</span></span>
<span><span class="co">## subset all NK cells of sig data</span></span>
<span><span class="va">sig_mat</span> <span class="op">&lt;-</span> <span class="va">proc_data</span><span class="op">$</span><span class="va">voomE</span><span class="op">[</span>, <span class="va">proc_data</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">celltype.ch1</span> <span class="op">==</span> <span class="st">"NK"</span><span class="op">]</span></span></code></pre></div>
<p><strong><em>Note:</em></strong></p>
<p><code>ccle_crc_5</code> is a DGEList object contains 5 CRC cell line
samples from <a href="https://depmap.org/portal/download/" class="external-link">CCLE</a>.
More details for <code><a href="../reference/ccle_crc_5.html">help(ccle_crc_5)</a></code>.</p>
<p><a href="https://depmap.org/portal/download/" class="external-link">DepMap CCLE</a> is a
large database containing RSEM quantified TPM data of more than 1,000
cancer cell lines.</p>
</div>
<div class="section level3">
<h3 id="ii-data-filtration">II) data filtration<a class="anchor" aria-label="anchor" href="#ii-data-filtration"></a>
</h3>
<p>Because SNR is computed using scaled expression across genes,
low-expression genes must be removed.</p>
<p>In this case, <code>sig_mat</code> from <code>proc_data</code> has
already been filtered, therefore background data (<code>bg_mat</code>)
needs to be filtered.</p>
<p><code>bg_mat</code> has only 5 samples, genes with logTPM &gt; 1
within more than 2 samples are kept.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">bg_mat</span> <span class="op">&gt;</span> <span class="fl">1</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span></span>
<span><span class="va">bg_mat</span> <span class="op">&lt;-</span> <span class="va">bg_mat</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="iii-markers-pool-refinement">III) markers pool refinement<a class="anchor" aria-label="anchor" href="#iii-markers-pool-refinement"></a>
</h3>
<p>To refine the markers pool, use <code><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat()</a></code> to
keep genes from <code>Markers</code> with high expression in
<code>sig_mat</code> and low expression in <code>bg_mat</code>.</p>
<p>This ensures the specificity of the markers minimizing the context
effect.</p>
<p>Similarly, use <code>gene_id</code> to convert gene IDs of input to
the same type (SYMBOL). As <code>sig_mat</code> uses ENSEMBL IDs,
<code>gene_id</code> types have to be defined in the order:
<code>sig_mat</code>, <code>bg_mat</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat</a></span><span class="op">(</span></span>
<span>  sig_mat <span class="op">=</span> <span class="va">sig_mat</span>,</span>
<span>  bg_mat <span class="op">=</span> <span class="va">bg_mat</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Warning in remove_bg_exp_mat(sig_mat = sig_mat, bg_mat = bg_mat, markers = geneIds(Markers), : Gene CCL4 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene CCL5 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL2 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL5A is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS2 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS5 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KLRA1P is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene LOC653757 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC24103 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene TRDC is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC61571 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene PIK3R6 is not in the sig_mat! So remove it!</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "IL32"   "RAB27B" "STYK1"  "RAB27A" "TXK"    "TTC38"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="use-ccle-as-background-data-optional">use CCLE as background data (optional)<a class="anchor" aria-label="anchor" href="#use-ccle-as-background-data-optional"></a>
</h3>
<p>Users can also use <a href="https://depmap.org/portal/download/" class="external-link">CCLE</a> downloaded by
<code><a href="https://rdrr.io/pkg/depmap/man/TPM.html" class="external-link">depmap::depmap_TPM()</a></code> for the refinement.</p>
<p>The entire CCLE data is quite large, thus use <code>ccle_crc_5</code>
to construct a small pseudo-CCLE data <code>ccle</code> with similar
format.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">ccle_crc_5</span><span class="op">$</span><span class="va">counts</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ccle_crc_5</span><span class="op">)</span>,</span>
<span>  primary_disease <span class="op">=</span> <span class="st">"CRC"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gene_name</span>, <span class="va">primary_disease</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"depmap_id"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"rna_expression"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">ccle</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 95,885 × 4</span></span></span>
<span><span class="co">#&gt;    gene_name primary_disease depmap_id                 rna_expression</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TSPAN6    CRC             SNU283_LARGE_INTESTINE             4.91 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TSPAN6    CRC             TGBC18TKB_LARGE_INTESTINE          6.05 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TSPAN6    CRC             SW837_LARGE_INTESTINE              5.26 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TSPAN6    CRC             SNU1040_LARGE_INTESTINE            5.16 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TSPAN6    CRC             HT29_LARGE_INTESTINE               4.40 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TNMD      CRC             SNU283_LARGE_INTESTINE             0.176</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TNMD      CRC             TGBC18TKB_LARGE_INTESTINE          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TNMD      CRC             SW837_LARGE_INTESTINE              0.333</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TNMD      CRC             SNU1040_LARGE_INTESTINE            0.465</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TNMD      CRC             HT29_LARGE_INTESTINE               0    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 95,875 more rows</span></span></span></code></pre></div>
<p>Because <code>ccle</code> is a long data, subsetting needs to be done
before converting it into matrix.</p>
<p>In this demo, we still extract CRC adherent cell lines data from
it.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## subset CRC cell lines of bg data</span></span>
<span><span class="va">ccle</span> <span class="op">&lt;-</span> <span class="va">ccle</span><span class="op">[</span><span class="va">ccle</span><span class="op">$</span><span class="va">primary_disease</span> <span class="op">==</span> <span class="st">"CRC"</span>, <span class="op">]</span></span></code></pre></div>
<p>As <code><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat()</a></code> can only accept matrix in wide
format, users can use <code><a href="../reference/ccle_2_wide.html">ccle_2_wide()</a></code> to convert
<code>ccle</code> into wide matrix.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccle</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ccle_2_wide.html">ccle_2_wide</a></span><span class="op">(</span>ccle <span class="op">=</span> <span class="va">ccle</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ccle</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;        SNU283_LARGE_INTESTINE TGBC18TKB_LARGE_INTESTINE SW837_LARGE_INTESTINE</span></span>
<span><span class="co">#&gt; TSPAN6              4.9140861                  6.045923             5.2570106</span></span>
<span><span class="co">#&gt; TNMD                0.1763228                  0.000000             0.3334237</span></span>
<span><span class="co">#&gt; DPM1                6.9468478                  6.724105             6.9242187</span></span></code></pre></div>
<p>The same filtration cutoff used in <code>bg_mat</code> above is used
for <code>ccle</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">ccle</span> <span class="op">&gt;</span> <span class="fl">1</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span></span>
<span><span class="va">ccle</span> <span class="op">&lt;-</span> <span class="va">ccle</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span></span></code></pre></div>
<p>Use <code><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat()</a></code> to refine the markers pool based
on <code>ccle</code>. As we use <code>ccle_crc_5</code> to construct
<code>ccle</code>, the result should be the same.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m_ccl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat</a></span><span class="op">(</span></span>
<span>  sig_mat <span class="op">=</span> <span class="va">sig_mat</span>,</span>
<span>  bg_mat <span class="op">=</span> <span class="va">ccle</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Warning in remove_bg_exp_mat(sig_mat = sig_mat, bg_mat = ccle, markers = geneIds(Markers), : Gene CCL4 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene CCL5 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL2 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DL5A is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS2 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KIR2DS5 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene KLRA1P is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene LOC653757 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC24103 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene TRDC is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene MGC61571 is not in the sig_mat! So remove it!</span></span>
<span><span class="co">#&gt; Gene PIK3R6 is not in the sig_mat! So remove it!</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "IL32"   "RAB27B" "STYK1"  "RAB27A" "TXK"    "TTC38"</span></span></code></pre></div>
<p><strong><em>Note:</em></strong></p>
<p>Wrapper function <code><a href="../reference/remove_bg_exp.html">remove_bg_exp()</a></code> can complete Step 3. I)
II) III) all using one single function.</p>
<p>Users can use the entire CCLE data from DepMap without loading it by
using param <code>bg_data = 'CCLE'</code>. More details in
<code><a href="../reference/remove_bg_exp.html">help(remove_bg_exp)</a></code>.</p>
</div>
<div class="section level3">
<h3 id="iv-combination-with-signature">IV) combination with Signature<a class="anchor" aria-label="anchor" href="#iv-combination-with-signature"></a>
</h3>
<p>To get the final signature with high group specificity and low
background expression, intersect the result <code>m_ccl</code> with the
signature <code>m_ct</code> we get from Step 2.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_NK_CRC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">m_ct</span>, <span class="va">m_ccl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sig_NK_CRC</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "CD244" "CD247" "CTSW"  "GZMA"  "GZMM"  "IL2RB"</span></span></code></pre></div>
<p>Here, if we directly use <code>m_ct</code> instead of
<code>Markers</code> as input <code>markers</code> for
<code><a href="../reference/remove_bg_exp_mat.html">remove_bg_exp_mat()</a></code> or <code><a href="../reference/remove_bg_exp.html">remove_bg_exp()</a></code>,
<code>m_ccl</code> will already be the intersection result of the
original <code>m_ccl</code> and <code>m_ct</code>, then we can skip
this.</p>
</div>
</div>
<div class="section level2">
<h2 id="step-4--visualization-of-final-results">Step 4. Visualization of Final Results<a class="anchor" aria-label="anchor" href="#step-4--visualization-of-final-results"></a>
</h2>
<hr>
<p>To assess how well the refined signature can discriminate our target
group from others, mastR provides the following visualization functions
for different purposes:</p>
<ul>
<li><p><code><a href="../reference/pca_matrix_plot.html">pca_matrix_plot()</a></code>: matrix of PCA plots with top n
PCs. Show if the target group can be separated from other groups and how
well the separation is. Users can also know how much variance has been
explained by PC1 with the signature.</p></li>
<li><p><code><a href="../reference/sig_heatmap.html">sig_heatmap()</a></code>: signature heatmap across groups and
its comparison with un-refined markers pool.<br>
Validate if the signature shows clear and differential expression
pattern between the target group and others. Show if the pattern is
clearer and cleaner after the refinement compared with un-refined
markers pool.</p></li>
<li><p><code><a href="../reference/sig_rankdensity_plot.html">sig_rankdensity_plot()</a></code>: rank density distribution of
signature genes across groups.<br>
Show if the signature genes are highly expressed within each sample of
the target group while lowly expressed in others.</p></li>
<li><p><code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code>: boxplot of signature expression or
singscore across groups.<br>
Test the overall performance of the whole signature when applying it for
scoring, singscore is used to compute the rank score using the whole
signature. The boxplot across groups can be used to test if the whole
signature is powerful enough to distinguish the target group from
others. Median expression of each signature gene can also be
plotted.</p></li>
<li><p><code><a href="../reference/sig_scatter_plot.html">sig_scatter_plot()</a></code>: scatter plot of signature scaled
expression of the target group vs all other groups respectively.<br>
Visualize the correlation of the signature between the target and other
groups, the row-scaled gene expression is computed to make the scatter
plot. Based on it, the co-expression correlation and signature’s
specificity can be assessed.</p></li>
<li><p><code><a href="../reference/sig_gseaplot.html">sig_gseaplot()</a></code>: gene set enrichment analysis (GSEA)
result display.<br>
Validate the enrichment significance of signature in our target group
compared with other groups, using <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html" class="external-link">clusterProfiler::GSEA()</a></code>
analysis. Users can display ‘dotplot’ or gseaplot’.</p></li>
</ul>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>Use <code><a href="../reference/sig_heatmap.html">sig_heatmap()</a></code> function to compare the expression
pattern between the original markers pool and the final signature in
<code>proc_data</code>.</p>
<ul>
<li>
<code>sigs</code>: final signature (or list of multiple
signatures).</li>
<li>
<code>markers</code>: original markers pool (optional).</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_heatmap.html">sig_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>  show_column_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_names <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="co">#&gt; Gene CCL4 is not in data</span></span>
<span><span class="co">#&gt; Gene CCL5 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DL2 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DL5A is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DS2 is not in data</span></span>
<span><span class="co">#&gt; Gene KIR2DS5 is not in data</span></span>
<span><span class="co">#&gt; Gene KLRA1P is not in data</span></span>
<span><span class="co">#&gt; Gene LOC653757 is not in data</span></span>
<span><span class="co">#&gt; Gene MGC24103 is not in data</span></span>
<span><span class="co">#&gt; Gene TRDC is not in data</span></span>
<span><span class="co">#&gt; Gene MGC61571 is not in data</span></span>
<span><span class="co">#&gt; Gene PIK3R6 is not in data</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/heatmap-1.png" width="960"></p>
<p>The expression pattern of the signature is more distinguishable in NK
cells.</p>
</div>
<div class="section level3">
<h3 id="signature-score-boxplot">Signature Score Boxplot<a class="anchor" aria-label="anchor" href="#signature-score-boxplot"></a>
</h3>
<p>To see if the final signature can distinguish NK cells from other
immune cells, use <code><a href="../reference/sig_boxplot.html">sig_boxplot()</a></code> function to make a boxplot
of NK scores using <code>sig_NK_CRC</code>. Scores are calculated by
<code>singscore</code> package.</p>
<p><strong><em>Note</em></strong>:</p>
<p>Boxplot of median expression of each gene can be plotted using param
<code>type = "expression"</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sig_boxplot.html">sig_boxplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/score%20boxplot-1.png" width="768"></p>
<p>It’s clear that the signature shows significantly higher score level
in NK cells.</p>
</div>
<div class="section level3">
<h3 id="signature-abundance-scatter-plot">Signature Abundance Scatter Plot<a class="anchor" aria-label="anchor" href="#signature-abundance-scatter-plot"></a>
</h3>
<p>To validate the specificity of the final signature to NK cells, use
<code><a href="../reference/sig_scatter_plot.html">sig_scatter_plot()</a></code> function to make the scatter plot of the
signature z-scored expression in <code>proc_data</code>.</p>
<p>If the signature is highly specific to NK cells, most of the
signature genes should appear in top left region (scaled expression &gt;
1 in NK, &lt; 1 in others). The genes in top right region should
represent the co-expression in both groups.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## before refinement</span></span>
<span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Before Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot-1.png" width="768"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## after refinement</span></span>
<span><span class="fu"><a href="../reference/sig_scatter_plot.html">sig_scatter_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="va">sig_NK_CRC</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"After Refinement"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/abundance%20scatter%20plot-2.png" width="768"></p>
<p>Almost all of non-specific genes are filtered by mastR.</p>
<p>Further optimization can be performed by removing signature genes
present in the bottom right or bottom left region based on the scatter
plot.</p>
</div>
<div class="section level3">
<h3 id="signature-gsea-plot">Signature GSEA plot<a class="anchor" aria-label="anchor" href="#signature-gsea-plot"></a>
</h3>
<p>To assess the enrichment significance of the final signature in NK
cells, use <code><a href="../reference/sig_gseaplot.html">sig_gseaplot()</a></code> function to display GSEA result.
Dotplot can be plotted by using <code>method = "dotplot"</code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## gseaplot</span></span>
<span><span class="fu"><a href="../reference/sig_gseaplot.html">sig_gseaplot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">proc_data</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>sig <span class="op">=</span> <span class="va">sig_NK_CRC</span>, markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype.ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"voomE"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"gseaplot"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="mastR_Demo_files/figure-html/gseaplot-1.png" width="960"></p>
<p>The refined signature is enriched in all comparisons and mastR
successfully removed the trailing tail from the markers pool.</p>
</div>
</div>
<div class="section level2">
<h2 id="working-with-extension-data-input">Working with Extension Data Input<a class="anchor" aria-label="anchor" href="#working-with-extension-data-input"></a>
</h2>
<div class="section level3">
<h3 id="multi-data">Multiple Datasets<a class="anchor" aria-label="anchor" href="#multi-data"></a>
</h3>
<hr>
<p>The standard workflow of mastR usage is as described above.</p>
<p>The above is a show case of single dataset, mastR can also be applied
to multiple datasets.</p>
<p>After the Step 2c (Section @ref(intersect-pool)), for:</p>
<ol style="list-style-type: decimal">
<li><p>single dataset<br>
Returns the final result as the signature.</p></li>
<li><p>multiple datasets<br>
After Step 2c, choose one combination method (<code>union</code>
default) to aggregate signature lists generated from different datasets.
The method Robust Rank Aggregation (“RRA”) can also be selected in
mastR, which detects genes that are ranked consistently better than
expected under null hypothesis of uncorrelated inputs and assigns a
significance score for each gene.</p></li>
</ol>
<p>Instead of integrating datasets into one larger dataset and fit it
into one linear model, mastR refines signature in each dataset
independently and the aggregation method can be defined using param
<code>comb</code> (e.g. <code>comb = "RRA"</code>).</p>
<p>The advantage of this is that it avoids over-normalizing or
mis-correcting when carrying out data integration. More robust and
conserved signature across datasets can be obtained. In some cases, it’s
better than using “Remove Batch Effect”.</p>
<p>Simple combination methods like “union” and “intersect” are also
supported (e.g. <code>comb = union</code>).</p>
<p>It’s recommended to use “union” when only a few genes are refined for
each dataset, while “RRA” is better for robust gene selection from large
DEG lists, “intersect” is proper for highly overlapping gene lists.</p>
<p><strong><em>Note:</em></strong></p>
<p>All signature visualization functions in mastR support multiple
datasets. More details in <code><a href="https://rdrr.io/r/utils/help.html" class="external-link">help()</a></code>.</p>
<p>To demonstrate the usage on multiple datasets, <code>im_data_6</code>
is used as an example.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## In the demo, we just repeatedly use im_data_6 as a show case</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  feature_selection <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  dir <span class="op">=</span> <span class="st">"UP"</span>, <span class="co">## specify to keep "UP" or "DOWN" regulated genes</span></span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  comb <span class="op">=</span> <span class="va">union</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The signature <code>m_ct_m</code> is the same as <code>m_ct</code> as
“union” is set as combination method.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## we will get exactly the same list</span></span>
<span><span class="co">## if we choose 'union' or 'intersect' as combination</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setequal</a></span><span class="op">(</span><span class="va">m_ct_m</span>, <span class="va">m_ct</span><span class="op">)</span></span></code></pre></div>
<p>By using <code>comb = "RRA"</code>, only the genes at the top rank
across signature lists are kept. The cutoff of “RRA” can be set with
param <code>s_thres</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## but we will only get the genes appear at top rank across gene lists</span></span>
<span><span class="co">## if we choose 'RRA', s_thres is to determine the threshold for ranking score</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">m_ct_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>A <span class="op">=</span> <span class="va">im_data_6</span>, B <span class="op">=</span> <span class="va">im_data_6</span><span class="op">)</span>,</span>
<span>  markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GSEABase/man/GeneSet-class.html" class="external-link">geneIds</a></span><span class="op">(</span><span class="va">Markers</span><span class="op">)</span>,</span>
<span>  group_col <span class="op">=</span> <span class="st">"celltype:ch1"</span>,</span>
<span>  target_group <span class="op">=</span> <span class="st">"NK"</span>,</span>
<span>  feature_selection <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  dir <span class="op">=</span> <span class="st">"UP"</span>, <span class="co">## specify to keep "UP" or "DOWN" regulated genes</span></span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>  comb <span class="op">=</span> <span class="st">"RRA"</span>, <span class="co">## change this to use different strategy, default is "union"</span></span>
<span>  s_thres <span class="op">=</span> <span class="fl">0.5</span> <span class="co">## only work when comb = "RRA", set a threshold for ranking score</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## fewer signature genes this time</span></span>
<span><span class="va">m_ct_m</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="single-cell-rna-sequencing-data">Single Cell RNA-sequencing Data<a class="anchor" aria-label="anchor" href="#single-cell-rna-sequencing-data"></a>
</h3>
<hr>
<p>Although mastR is designed for bulk RNA-seq data, in order to support
usage of abundant scRNA resources, mastR provides
<code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> function to help convert scRNA-seq data
into pseudo-bulk data, which can then be used in mastR workflow.</p>
<p><code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> aggregates cells into pseudo-bulk
samples according to the factor(s) provided using param <code>by</code>.
Then use <code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> to get the signature from
scRNA-seq data.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## create a test scRNA object of 100 genes x 100 cells</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">10000</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>,</span>
<span>  level <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, each <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">100</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">counts</span>, by <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pb</span>, group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span><span class="va">pb</span>, group_col <span class="op">=</span> <span class="st">"group"</span>, target_group <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Seurat or SCE object are also accepted</span></span>
<span><span class="co"># scRNA &lt;- SeuratObject::CreateSeuratObject(counts = counts, meta.data = meta)</span></span>
<span><span class="co"># pseudo_samples(scRNA, by = c("subset","level"))</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 class="tabset tabset-fade tabset-pills" id="applications">Applications<a class="anchor" aria-label="anchor" href="#applications"></a>
</h2>
<p>Users can apply the signature in multiple ways. In this short demo,
we will show 3 simple applications.</p>
<p><strong><em>Note:</em></strong></p>
<p>These examples are not run in this demo.</p>
<p>Firstly, simulate a scRNA data using <code>splatter</code>
package.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">splatter</span><span class="op">)</span></span>
<span><span class="co">## set seed for reproduce as there's permutation inside</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim_params</span> <span class="op">&lt;-</span> <span class="fu">newSplatParams</span><span class="op">(</span></span>
<span>  nGenes <span class="op">=</span> <span class="fl">1e4</span>,</span>
<span>  batchCells <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  group.prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.4</span>, length.out <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>  de.prob <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  <span class="co"># de.downProb = 0,  ## only set up-regulated genes for each group</span></span>
<span>  de.facLoc <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  de.facScale <span class="op">=</span> <span class="fl">0.4</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">splatSimulate</span><span class="op">(</span><span class="va">sim_params</span>, method <span class="op">=</span> <span class="st">"groups"</span><span class="op">)</span></span></code></pre></div>
<p>Markers list for each group can be obtained from the simulated scRNA
data.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="fu">rowData</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"DEFacGroup"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Use <code><a href="../reference/pseudo_samples.html">pseudo_samples()</a></code> to get pseudo-bulk data and create
a DGEList.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## aggregate into pseudo-bulk samples</span></span>
<span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>,</span>
<span>  by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"Group"</span><span class="op">)</span>,</span>
<span>  min.cells <span class="op">=</span> <span class="fl">50</span>, max.cells <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Use mastR wrapper function <code><a href="../reference/filter_subset_sig.html">filter_subset_sig()</a></code> to get
the signature for each group. In this case, there are no markers pool of
interest and no background.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">sig_ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/filter_subset_sig.html">filter_subset_sig</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">dge</span>,</span>
<span>    markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    group_col <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>    target_group <span class="op">=</span> <span class="va">x</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p>Show the overlap of the signature lists generated by mastR and the
markers lists from the simulation.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## venn plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, \<span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="fu">ggvenn</span><span class="fu">::</span><span class="fu">ggvenn</span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    sig <span class="op">=</span> <span class="va">sig_ls</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    marker <span class="op">=</span> <span class="va">markers_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="op">)</span>,</span>
<span>  show_percentage <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p>Use <code><a href="../reference/sig_heatmap.html">sig_heatmap()</a></code> to compare the expression of
signatures and markers lists.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## heatmap</span></span>
<span><span class="fu"><a href="../reference/sig_heatmap.html">sig_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="fu">cpm</span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  sigs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sig_ls</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"TP53"</span><span class="op">)</span><span class="op">)</span>, <span class="co">## add a real gene to pass gene check</span></span>
<span>  group_col <span class="op">=</span> <span class="va">dge</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">group</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>  show_column_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_row_den <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_column_slices <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_row_slices <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Randomly aggregate scRNA data to generate pseudo bulk data for
scoring and deconvolution.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">## randomly generate aggregating idx</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">rand_idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## aggregate into pseudo-bulk samples based on rand_idx</span></span>
<span><span class="va">pb_r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudo_samples.html">pseudo_samples</a></span><span class="op">(</span><span class="va">data_sim</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch"</span>, <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">DGEList</span><span class="op">(</span></span>
<span>  counts <span class="op">=</span> <span class="va">pb_r</span>,</span>
<span>  samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*\\.(.*)_.*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    Batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\..*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_.*"</span>, <span class="st">"\\1"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pb_r</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Cellular proportion can be added into the random pseudo bulk
data.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## append cellular composition</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">rand_idx</span>, <span class="va">Group</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">Group</span>, values_from <span class="op">=</span> <span class="va">count</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>rand_idx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rand_idx</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span>, <span class="va">tmp</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group"</span> <span class="op">=</span> <span class="st">"rand_idx"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## data process</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">filterByExpr</span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="va">dge_r</span><span class="op">[</span><span class="va">keep</span>, , keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="va">dge_r</span> <span class="op">&lt;-</span> <span class="fu">calcNormFactors</span><span class="op">(</span><span class="va">dge_r</span>, method <span class="op">=</span> <span class="st">"TMM"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="score">Score<a class="anchor" aria-label="anchor" href="#score"></a>
</h3>
<p>Multiple scores are calculated by <code>singscore</code> package
using signatures and markers lists. Make scatter plot to compare the
correlation between the score and the actual cellular proportion.</p>
<p>For signatures generated by mastR:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rank_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/rankGenes.html" class="external-link">rankGenes</a></span><span class="op">(</span><span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge_r</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score based on sig_ls</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span>, <span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  <span class="op">-</span><span class="va">Sample</span>,</span>
<span>  names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Prop"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prop</span>, y <span class="op">=</span> <span class="va">Score</span>, col <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For markers list from simulation:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## score based on markers_list</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">markers_list</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dge_r</span><span class="op">)</span>, <span class="va">dge_r</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  <span class="op">-</span><span class="va">Sample</span>,</span>
<span>  names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Prop"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">$</span><span class="va">Group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"DEFac"</span>, <span class="va">tmp</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">Sample</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"Score"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">tmp</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Prop</span>, y <span class="op">=</span> <span class="va">Score</span>, col <span class="op">=</span> <span class="va">Group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="deconvolution">Deconvolution<a class="anchor" aria-label="anchor" href="#deconvolution"></a>
</h3>
<p>Use the <code>DWLS</code> package for cellular deconvolution.</p>
<p>Subset the scRNA reference data using the generated signatures and
build its signature profile. Cellular proportion can be estimated from
this.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_matrix</span> <span class="op">&lt;-</span> <span class="fu">DWLS</span><span class="fu">::</span><span class="fu">buildSignatureMatrixMAST</span><span class="op">(</span></span>
<span>  scdata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>  id <span class="op">=</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span>,</span>
<span>  path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">DWLS</span><span class="fu">::</span><span class="fu">solveDampenedWLS</span><span class="op">(</span><span class="va">sig_matrix</span>, <span class="va">pb_r</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sig_matrix</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotation">Annotation<a class="anchor" aria-label="anchor" href="#annotation"></a>
</h3>
<p>Use <code>singscore</code> as a marker-based annotation method.
Calculate the scores based on the signatures and markers lists, then
assign each cell with the cell type having the highest score. Compare
the results.</p>
<p>Using signatures:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://davislaboratory.github.io/singscore" class="external-link">singscore</a></span><span class="op">)</span></span>
<span><span class="co">## normalization</span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu">computePooledFactors</span><span class="op">(</span><span class="va">data_sim</span>, clusters <span class="op">=</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span>
<span><span class="va">data_sim</span> <span class="op">&lt;-</span> <span class="fu">scuttle</span><span class="fu">::</span><span class="fu">logNormCounts</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span></span>
<span><span class="co">## use singscore for annotation</span></span>
<span><span class="va">rank_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/rankGenes.html" class="external-link">rankGenes</a></span><span class="op">(</span><span class="fu">logcounts</span><span class="op">(</span><span class="va">data_sim</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## score using sig_ls</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">sig_ls</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="fl">2</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">==</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span></code></pre></div>
<p>Using markers lists:</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## score using markers_list</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://DavisLaboratory.github.io/singscore/reference/multiScore.html" class="external-link">multiScore</a></span><span class="op">(</span><span class="va">rank_data</span>, upSetColc <span class="op">=</span> <span class="fu"><a href="../reference/gls2gsc.html">gls2gsc</a></span><span class="op">(</span><span class="va">markers_list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">Scores</span>, <span class="fl">2</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">data_sim</span><span class="op">$</span><span class="va">Pred</span> <span class="op">==</span> <span class="va">data_sim</span><span class="op">$</span><span class="va">Group</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<hr>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GSEABase_1.61.1      graph_1.77.3         annotate_1.77.0     </span></span>
<span><span class="co">#&gt;  [4] XML_3.99-0.14        AnnotationDbi_1.61.2 IRanges_2.33.0      </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.37.4     Biobase_2.59.0       BiocGenerics_0.45.3 </span></span>
<span><span class="co">#&gt; [10] ggplot2_3.4.2        mastR_0.99.4        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] shadowtext_0.1.2            backports_1.4.1            </span></span>
<span><span class="co">#&gt;   [3] circlize_0.4.15             fastmatch_1.1-3            </span></span>
<span><span class="co">#&gt;   [5] systemfonts_1.0.4           igraph_1.4.1               </span></span>
<span><span class="co">#&gt;   [7] plyr_1.8.8                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;   [9] sp_1.6-0                    splines_4.2.3              </span></span>
<span><span class="co">#&gt;  [11] BiocParallel_1.33.12        listenv_0.9.0              </span></span>
<span><span class="co">#&gt;  [13] GenomeInfoDb_1.35.16        msigdb_1.7.0               </span></span>
<span><span class="co">#&gt;  [15] digest_0.6.31               yulab.utils_0.0.6          </span></span>
<span><span class="co">#&gt;  [17] foreach_1.5.2               htmltools_0.5.5            </span></span>
<span><span class="co">#&gt;  [19] GOSemSim_2.25.0             viridis_0.6.2              </span></span>
<span><span class="co">#&gt;  [21] GO.db_3.17.0                fansi_1.0.4                </span></span>
<span><span class="co">#&gt;  [23] magrittr_2.0.3              memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [25] cluster_2.1.4               doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [27] limma_3.55.6                graphlayouts_0.8.4         </span></span>
<span><span class="co">#&gt;  [29] ComplexHeatmap_2.15.3       globals_0.16.2             </span></span>
<span><span class="co">#&gt;  [31] Biostrings_2.67.2           matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt;  [33] pkgdown_2.0.7               enrichplot_1.19.2          </span></span>
<span><span class="co">#&gt;  [35] colorspace_2.1-0            ggrepel_0.9.3              </span></span>
<span><span class="co">#&gt;  [37] blob_1.2.4                  textshaping_0.3.6          </span></span>
<span><span class="co">#&gt;  [39] xfun_0.38                   dplyr_1.1.1                </span></span>
<span><span class="co">#&gt;  [41] crayon_1.5.2                RCurl_1.98-1.12            </span></span>
<span><span class="co">#&gt;  [43] jsonlite_1.8.4              scatterpie_0.1.8           </span></span>
<span><span class="co">#&gt;  [45] progressr_0.13.0            ape_5.7-1                  </span></span>
<span><span class="co">#&gt;  [47] iterators_1.0.14            glue_1.6.2                 </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-4             gtable_0.3.3               </span></span>
<span><span class="co">#&gt;  [51] zlibbioc_1.45.0             XVector_0.39.0             </span></span>
<span><span class="co">#&gt;  [53] UpSetR_1.4.0                GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;  [55] DelayedArray_0.25.0         car_3.1-2                  </span></span>
<span><span class="co">#&gt;  [57] future.apply_1.10.0         shape_1.4.6                </span></span>
<span><span class="co">#&gt;  [59] SingleCellExperiment_1.21.1 abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [61] scales_1.2.1                DOSE_3.25.0                </span></span>
<span><span class="co">#&gt;  [63] DBI_1.1.3                   edgeR_3.41.6               </span></span>
<span><span class="co">#&gt;  [65] rstatix_0.7.2               Rcpp_1.0.10                </span></span>
<span><span class="co">#&gt;  [67] viridisLite_0.4.1           xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [69] clue_0.3-64                 tidytree_0.4.2             </span></span>
<span><span class="co">#&gt;  [71] gridGraphics_0.5-1          bit_4.0.5                  </span></span>
<span><span class="co">#&gt;  [73] httr_1.4.5                  fgsea_1.25.0               </span></span>
<span><span class="co">#&gt;  [75] RColorBrewer_1.1-3          pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [77] farver_2.1.1                sass_0.4.5                 </span></span>
<span><span class="co">#&gt;  [79] locfit_1.5-9.7              utf8_1.2.3                 </span></span>
<span><span class="co">#&gt;  [81] ggplotify_0.1.0             tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;  [83] labeling_0.4.2              rlang_1.1.0                </span></span>
<span><span class="co">#&gt;  [85] reshape2_1.4.4              munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [87] tools_4.2.3                 cachem_1.0.7               </span></span>
<span><span class="co">#&gt;  [89] downloader_0.4              cli_3.6.1                  </span></span>
<span><span class="co">#&gt;  [91] generics_0.1.3              RSQLite_2.3.1              </span></span>
<span><span class="co">#&gt;  [93] gson_0.1.0                  broom_1.0.4                </span></span>
<span><span class="co">#&gt;  [95] evaluate_0.20               stringr_1.5.0              </span></span>
<span><span class="co">#&gt;  [97] fastmap_1.1.1               yaml_2.3.7                 </span></span>
<span><span class="co">#&gt;  [99] ragg_1.2.5                  ggtree_3.7.1               </span></span>
<span><span class="co">#&gt; [101] org.Hs.eg.db_3.17.0         knitr_1.42                 </span></span>
<span><span class="co">#&gt; [103] bit64_4.0.5                 fs_1.6.1                   </span></span>
<span><span class="co">#&gt; [105] tidygraph_1.2.3             purrr_1.0.1                </span></span>
<span><span class="co">#&gt; [107] ggraph_2.1.0                KEGGREST_1.39.0            </span></span>
<span><span class="co">#&gt; [109] nlme_3.1-162                future_1.32.0              </span></span>
<span><span class="co">#&gt; [111] aplot_0.1.10                compiler_4.2.3             </span></span>
<span><span class="co">#&gt; [113] png_0.1-8                   ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [115] treeio_1.23.1               tweenr_2.0.2               </span></span>
<span><span class="co">#&gt; [117] tibble_3.2.1                bslib_0.4.2                </span></span>
<span><span class="co">#&gt; [119] stringi_1.7.12              highr_0.10                 </span></span>
<span><span class="co">#&gt; [121] desc_1.4.2                  lattice_0.21-8             </span></span>
<span><span class="co">#&gt; [123] Matrix_1.5-4                vctrs_0.6.1                </span></span>
<span><span class="co">#&gt; [125] pillar_1.9.0                lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt; [127] jquerylib_0.1.4             GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [129] data.table_1.14.8           cowplot_1.1.1              </span></span>
<span><span class="co">#&gt; [131] bitops_1.0-7                patchwork_1.1.2            </span></span>
<span><span class="co">#&gt; [133] GenomicRanges_1.51.4        qvalue_2.31.1              </span></span>
<span><span class="co">#&gt; [135] R6_2.5.1                    gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [137] parallelly_1.35.0           codetools_0.2-19           </span></span>
<span><span class="co">#&gt; [139] MASS_7.3-58.3               SummarizedExperiment_1.29.1</span></span>
<span><span class="co">#&gt; [141] rprojroot_2.0.3             rjson_0.2.21               </span></span>
<span><span class="co">#&gt; [143] withr_2.5.0                 SeuratObject_4.1.3         </span></span>
<span><span class="co">#&gt; [145] GenomeInfoDbData_1.2.10     parallel_4.2.3             </span></span>
<span><span class="co">#&gt; [147] clusterProfiler_4.7.1       ggfun_0.0.9                </span></span>
<span><span class="co">#&gt; [149] grid_4.2.3                  tidyr_1.3.0                </span></span>
<span><span class="co">#&gt; [151] HDO.db_0.99.1               rmarkdown_2.21             </span></span>
<span><span class="co">#&gt; [153] MatrixGenerics_1.11.1       carData_3.0-5              </span></span>
<span><span class="co">#&gt; [155] singscore_1.19.1            ggpubr_0.6.0               </span></span>
<span><span class="co">#&gt; [157] ggforce_0.4.1</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jinjin Chen, Ahmed Mohamed.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
